<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mireia Ramos-Rodríguez" />


<title>#5 IREs and T1D risk</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">β-cells and T1D</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-α
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_ifna.html">Description</a>
    </li>
    <li>
      <a href="ifn_01_accessibility.html">#1 Chromatin remodeling induced by IFN-α</a>
    </li>
    <li>
      <a href="ifn_02_transcription.html">#2 Chromatin remodeling correlates with transcription changes</a>
    </li>
    <li>
      <a href="ifn_03_footprint.html">#3 IRF1, STAT1 and STAT2 are key regulators of IFN-α signaling</a>
    </li>
    <li>
      <a href="ifn_04_modules.html">#4 mRNA and protein modules regulated by IFN-α</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-γ + IL-1β
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_cyt.html">Description</a>
    </li>
    <li>
      <a href="cyt_01_remodeling.html">#1 Chromatin remodeling induced by cytokines</a>
    </li>
    <li>
      <a href="cyt_02_transcription.html">#2 Changes in transcription and translation</a>
    </li>
    <li>
      <a href="cyt_03_subclassification.html">#3 Subclassification of IREs</a>
    </li>
    <li>
      <a href="cyt_04_contacts.html">#4 Changes in 3D chormatin structure</a>
    </li>
    <li>
      <a href="cyt_05_t1d.html">#5 IREs and T1D risk</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IRB
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_irb.html">Description</a>
    </li>
    <li>
      <a href="https://mireia-bioinfo.github.io/plotRegulome">plotRegulome R package</a>
    </li>
    <li>
      <a href="https://github.com/mireia-bioinfo/isletregulome_shiny">IRB Shiny App code</a>
    </li>
    <li>
      <a href="http://www.isletregulome.org">Islet Regulome Browser</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mireia-bioinfo/phdthesis_code">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">#5 IREs and T1D risk</h1>
<h3 class="subtitle">Islet induced regulotory elements are implicated in T1D genetic susceptibility</h3>
<h4 class="author">Mireia Ramos-Rodríguez</h4>

</div>


<hr />
<div id="details" class="section level1">
<h1>Details</h1>
<ul>
<li><strong>Original publication</strong>:</li>
</ul>
<blockquote>
<p><strong>Ramos-Rodríguez, M.</strong>, Raurell-Vila, H., Colli, M.L. et al. The impact of proinflammatory cytokines on the β-cell regulatory landscape provides insights into the genetics of type 1 diabetes. <em>Nat Genet.</em> <strong>51</strong>, 1588–1595 (2019) <a href="https://doi.org/10.1038/s41588-019-0524-6" class="uri">https://doi.org/10.1038/s41588-019-0524-6</a></p>
</blockquote>
<ul>
<li><p><strong>Contents</strong>: Analyses and figures contained in this document correspond to the following figures/sections of the original publication:</p>
<ul>
<li><strong>Results</strong>: “Islet cytokine enhancers are implicated in T1D genetic susceptibility”.</li>
<li><strong>Figure 4</strong>: “Cytokine-induced islet regulatory elements map to T1D-associated regions and guide the identification of functional risk variants”. Panels <strong>a</strong>, <strong>b</strong>, <strong>d</strong>, <strong>e</strong>, <strong>f</strong> and <strong>h</strong>.</li>
<li><strong>Extended Data Figure 6</strong>: “Cytokine-induced islet regulatory elements are enriched in T1D associated variants”. Panels <strong>a</strong> to <strong>f</strong>.</li>
</ul></li>
</ul>
<hr />
</div>
<div id="methods" class="section level1">
<h1>Methods</h1>
<ul>
<li>Data was downloaded from the <a href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a> by selecting trait <strong>Type i diabetes mellitus</strong> with child traits. Date: 2019-04-18</li>
<li>Only SNPs with <code>DISEASE.TRAIT=="Type 1 diabetes"</code> were kept.</li>
<li>Proxies were obtained using <code>proxysnps</code> are package, which obtains proxies using <em>1,000 Genomes phase 3</em>, <em>EUR</em> population and <code>window_size=2.5e5</code>. Only proxies with LD &gt;= 0.5 were kept.</li>
<li>Locus were determined creating a window from the most upstream to the most downstream SNP. Overlapping loci were merged.</li>
</ul>
<pre><code>Rscript code/CYT_create_LD0.8.R
Rscript code/CYT_create_T1D_loci_public.R
Rscript code/CYT_create_T2D_loci_public.R
Rscript code/CYT_rm_common_T1D_T2D.R</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/T1D_GWAS/T1D_LD0.8_risk_loci_granges.rda&quot;)

df &lt;- data.frame(loci)
df$seqnames &lt;- factor(df$seqnames, levels=rev(paste0(&quot;chr&quot;, c(1:22, &quot;X&quot;, &quot;Y&quot;))))

width &lt;- 
  ggplot(df,
       aes(width)) +
  geom_density() +
  scale_x_continuous(labels=scales::comma,
                     name=&quot;Locus width (bp)&quot;)

counts &lt;- 
  ggplot(df,
       aes(seqnames)) +
  geom_dotplot(aes(fill=seqnames)) +
  coord_flip() +
  xlab(&quot;Chromosome&quot;) + 
  ylab(&quot;Proportion of loci&quot;) +
  theme(legend.position=&quot;none&quot;)

plot_grid(counts, width, ncol=2, rel_widths=c(.35, .65))</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-1-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/986b505fa6799362ed6a8a0f743a6821e97d5f98/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-1-1.png" target="_blank">986b505</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="variant-set-enrichment-analysis-vse" class="section level1">
<h1>Variant Set Enrichment Analysis (VSE)</h1>
<div id="t1d-gwas-data-minus-loci-common-with-t2d" class="section level2">
<h2>T1D GWAS data minus loci common with T2D</h2>
<pre class="r"><code>re &lt;- list.files(&quot;../data/CYT/bedfiles&quot;,
                 pattern=&quot;*.bed&quot;,
                 full.names=T)
re &lt;- re[!grepl(&quot;distal&quot;, re)]
re &lt;- re[grepl(&quot;padj0.05&quot;, re) &amp;
           grepl(&quot;hi&quot;, re)]

samples &lt;- data.frame(X=1:length(re),
                      SampleID=pipelineNGS::getNameFromPath(re, suffix=&quot;.bed&quot;),
                      Peaks=re)
samples$SampleID &lt;- gsub(&quot;_fc1_padj0.05&quot;, &quot;&quot;, samples$SampleID)

load(&quot;../data/CYT/T1D_GWAS/T1Drm_VSE_MRVS_500.rda&quot;)
load(&quot;../data/CYT/T1D_GWAS/T1Drm_VSE_AVS.rda&quot;)
seqlevelsStyle(t1d.avs) &lt;- &quot;Ensembl&quot;

t1d.vse &lt;- variantSetEnrichment.mod(t1d.avs, t1d.mrvs.500, samples)
save(t1d.vse, file=file.path(out_dir, &quot;T1Drm_hi_vse_result.rda&quot;))</code></pre>
<pre class="r"><code>load(file.path(out_dir, &quot;T1Drm_hi_vse_result.rda&quot;))

t1d.vse.res &lt;- VSESummary.mod(t1d.vse,
                                method=&quot;bonferroni&quot;)
knitr::kable(t1d.vse.res)</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
region
</th>
<th style="text-align:right;">
avs
</th>
<th style="text-align:right;">
enrichment
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
p.adjusted
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
IREs_hi
</td>
<td style="text-align:left;">
IREs_hi
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
4.617614
</td>
<td style="text-align:right;">
0.0000015
</td>
<td style="text-align:right;">
0.0000030
</td>
</tr>
<tr>
<td style="text-align:left;">
SREs_hi
</td>
<td style="text-align:left;">
SREs_hi
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
1.139202
</td>
<td style="text-align:right;">
0.1290808
</td>
<td style="text-align:right;">
0.2581615
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>VSEplot.mod(t1d.vse, 
            method=&quot;bonferroni&quot;, 
            pch = 20, 
            cex = 1, 
            padj = 0.05, 
            main = &quot;T1D in HI&quot;)</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-3-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/986b505fa6799362ed6a8a0f743a6821e97d5f98/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-3-1.png" target="_blank">986b505</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>re &lt;- list.files(&quot;../data/CYT/bedfiles&quot;,
                 pattern=&quot;*.bed&quot;,
                 full.names=T)
re &lt;- re[!grepl(&quot;distal&quot;, re) &amp; !grepl(&quot;k270.8&quot;, re)]
re &lt;- re[grepl(&quot;padj0.05&quot;, re) &amp;
           grepl(&quot;endoc&quot;, re)]

samples &lt;- data.frame(X=1:length(re),
                      SampleID=pipelineNGS::getNameFromPath(re, suffix=&quot;.bed&quot;),
                      Peaks=re)
samples$SampleID &lt;- gsub(&quot;_fc1_padj0.05&quot;, &quot;&quot;, samples$SampleID)

load(&quot;../data/CYT/T1D_GWAS/T1Drm_VSE_MRVS_500.rda&quot;)
load(&quot;../data/CYT/T1D_GWAS/T1Drm_VSE_AVS.rda&quot;)
seqlevelsStyle(t1d.avs) &lt;- &quot;Ensembl&quot;

t1d.vse &lt;- variantSetEnrichment.mod(t1d.avs, t1d.mrvs.500, samples)
save(t1d.vse, file=file.path(out_dir, &quot;T1Drm_endoc_vse_result.rda&quot;))</code></pre>
<pre class="r"><code>load(file.path(out_dir, &quot;T1Drm_endoc_vse_result.rda&quot;))

t1d.vse.res &lt;- VSESummary.mod(t1d.vse,
                                method=&quot;bonferroni&quot;)
knitr::kable(t1d.vse.res)</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
region
</th>
<th style="text-align:right;">
avs
</th>
<th style="text-align:right;">
enrichment
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
p.adjusted
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
IREs_endoc
</td>
<td style="text-align:left;">
IREs_endoc
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
3.140784
</td>
<td style="text-align:right;">
0.0007120
</td>
<td style="text-align:right;">
0.0014240
</td>
</tr>
<tr>
<td style="text-align:left;">
SREs_endoc
</td>
<td style="text-align:left;">
SREs_endoc
</td>
<td style="text-align:right;">
33
</td>
<td style="text-align:right;">
1.810616
</td>
<td style="text-align:right;">
0.0342564
</td>
<td style="text-align:right;">
0.0685129
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>VSEplot.mod(t1d.vse, 
            method=&quot;bonferroni&quot;, 
            pch = 20, 
            cex = 1, 
            padj = 0.05, 
            main = &quot;T1D in EndoC&quot;)</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-5-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/986b505fa6799362ed6a8a0f743a6821e97d5f98/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-5-1.png" target="_blank">986b505</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="t2d-gwas-data-minus-loci-common-with-t1d" class="section level2">
<h2>T2D GWAS data minus loci common with T1D</h2>
<pre class="r"><code>re &lt;- list.files(&quot;../data/CYT/bedfiles&quot;,
                 pattern=&quot;*.bed&quot;,
                 full.names=T)
re &lt;- re[!grepl(&quot;distal&quot;, re)]
re &lt;- re[grepl(&quot;padj0.05&quot;, re) &amp;
           grepl(&quot;hi&quot;, re)]

samples &lt;- data.frame(X=1:length(re),
                      SampleID=pipelineNGS::getNameFromPath(re, suffix=&quot;.bed&quot;),
                      Peaks=re)
samples$SampleID &lt;- gsub(&quot;_fc1_padj0.05&quot;, &quot;&quot;, samples$SampleID)

load(&quot;../data/CYT/T1D_GWAS/T2Drm_VSE_MRVS_500.rda&quot;)
load(&quot;../data/CYT/T1D_GWAS/T2Drm_VSE_AVS.rda&quot;)
seqlevelsStyle(t2d.avs) &lt;- &quot;Ensembl&quot;

t2d.vse &lt;- variantSetEnrichment.mod(t2d.avs, t2d.mrvs.500, samples)
save(t2d.vse, file=file.path(out_dir, &quot;T2Drm_hi_vse_result.rda&quot;))</code></pre>
<pre class="r"><code>load(file.path(out_dir, &quot;T2Drm_hi_vse_result.rda&quot;))

t2d.vse.res &lt;- VSESummary.mod(t2d.vse,
                                method=&quot;bonferroni&quot;)
knitr::kable(t2d.vse.res)</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
region
</th>
<th style="text-align:right;">
avs
</th>
<th style="text-align:right;">
enrichment
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
p.adjusted
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
IREs_hi
</td>
<td style="text-align:left;">
IREs_hi
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-0.3737613
</td>
<td style="text-align:right;">
0.6490075
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
SREs_hi
</td>
<td style="text-align:left;">
SREs_hi
</td>
<td style="text-align:right;">
216
</td>
<td style="text-align:right;">
5.4742895
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>VSEplot.mod(t2d.vse, 
            method=&quot;bonferroni&quot;, 
            pch = 20, 
            cex = 1, 
            padj = 0.05, 
            main = &quot;T2D in HI&quot;)</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-7-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/986b505fa6799362ed6a8a0f743a6821e97d5f98/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-7-1.png" target="_blank">986b505</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>re &lt;- list.files(&quot;../data/CYT/bedfiles&quot;,
                 pattern=&quot;*.bed&quot;,
                 full.names=T)
re &lt;- re[!grepl(&quot;distal&quot;, re) &amp; !grepl(&quot;k270.8&quot;, re)]
re &lt;- re[grepl(&quot;padj0.05&quot;, re) &amp;
           grepl(&quot;endoc&quot;, re)]

samples &lt;- data.frame(X=1:length(re),
                      SampleID=pipelineNGS::getNameFromPath(re, suffix=&quot;.bed&quot;),
                      Peaks=re)
samples$SampleID &lt;- gsub(&quot;_fc1_padj0.05&quot;, &quot;&quot;, samples$SampleID)

load(&quot;../data/CYT/T1D_GWAS/T2Drm_VSE_MRVS_500.rda&quot;)
load(&quot;../data/CYT/T1D_GWAS/T2Drm_VSE_AVS.rda&quot;)
seqlevelsStyle(t2d.avs) &lt;- &quot;Ensembl&quot;

t2d.vse &lt;- variantSetEnrichment.mod(t2d.avs, t2d.mrvs.500, samples)
save(t2d.vse, file=file.path(out_dir, &quot;T2Drm_endoc_vse_result.rda&quot;))</code></pre>
<pre class="r"><code>load(file.path(out_dir, &quot;T2Drm_endoc_vse_result.rda&quot;))

t2d.vse.res &lt;- VSESummary.mod(t2d.vse,
                                method=&quot;bonferroni&quot;)
knitr::kable(t2d.vse.res)</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
region
</th>
<th style="text-align:right;">
avs
</th>
<th style="text-align:right;">
enrichment
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
p.adjusted
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
IREs_endoc
</td>
<td style="text-align:left;">
IREs_endoc
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
0.6608974
</td>
<td style="text-align:right;">
0.2572423
</td>
<td style="text-align:right;">
0.5144847
</td>
</tr>
<tr>
<td style="text-align:left;">
SREs_endoc
</td>
<td style="text-align:left;">
SREs_endoc
</td>
<td style="text-align:right;">
174
</td>
<td style="text-align:right;">
4.5283477
</td>
<td style="text-align:right;">
0.0000029
</td>
<td style="text-align:right;">
0.0000059
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>VSEplot.mod(t2d.vse, 
            method=&quot;bonferroni&quot;, 
            pch = 20, 
            cex = 1, 
            padj = 0.05, 
            main = &quot;T2D in EndoC&quot;)</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-9-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/986b505fa6799362ed6a8a0f743a6821e97d5f98/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-9-1.png" target="_blank">986b505</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="views-of-example-t1d-loci-with-variants-overlapping-ires" class="section level1">
<h1>Views of example T1D loci with variants overlapping IREs</h1>
<div id="il7r-locus" class="section level2">
<h2>IL7R locus</h2>
<pre class="r"><code>## SNPs ----------------
rsids &lt;- c(&quot;rs11954020&quot;, &quot;rs6897932&quot;)

load(&quot;../data/CYT/T1D_GWAS/T1Drm_LD0.8_VSE_AVS.rda&quot;)
load(&quot;../data/CYT/T1D_GWAS/T1Drm_risk_proxies_granges.rda&quot;)
t1d.dat &lt;- data.frame(mcols(proxies))[,c(1,5)]
t1d.dat &lt;- t1d.dat[t1d.dat$R.squared&gt;=0.8,]

snp &lt;- unlist(t1d.avs)
snp &lt;- data.frame(snp[snp$idTag %in% rsids,])

snp$type &lt;- &quot;proxy&quot;
snp$type[snp$idTag==snp$idLd] &lt;- &quot;leading&quot;
snp$lab &lt;- &quot;&quot;
snp$lab[snp$idLd %in% c(rsids, &quot;rs34143578&quot;, &quot;rs6890853&quot;, &quot;rs10214273&quot;)] &lt;- snp$idLd[snp$idLd %in% c(rsids, &quot;rs34143578&quot;, &quot;rs6890853&quot;, &quot;rs10214273&quot;)]

snp &lt;- dplyr::left_join(snp,
                        t1d.dat,
                        by=c(&quot;idLd&quot;=&quot;ID&quot;))

snp$type &lt;- factor(snp$type, levels=c(&quot;proxy&quot;, &quot;leading&quot;))
snp &lt;- snp[order(snp$type),]

win &lt;- GRanges(paste0(unique(snp$seqnames), &quot;:&quot;, min(snp$start), &quot;-&quot;, max(snp$start)))
xlims &lt;- c(min(snp$start), max(snp$start))

## RE --------------------
load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_subgroup.rda&quot;)
re.sel &lt;- subsetByOverlaps(re, win)
re.sel &lt;- resize(re.sel, width=1e3, fix=&quot;center&quot;)
re.sel &lt;- data.frame(re.sel)

## Genes -------------------
load(&quot;../data/CYT/RNA/diffAnalysis/RNA_hi_GRangesBatch.rda&quot;)
res.gr$type[res.gr$baseMean&lt;=1] &lt;- &quot;not-expressed&quot;
res.gr &lt;- res.gr[res.gr$gene_biotype==&quot;protein_coding&quot;,]

col.df &lt;- data.frame(&quot;type&quot;=c(names(pals$differential), &quot;not-expressed&quot;),
                     &quot;color&quot;=c(pals$differential, &quot;black&quot;),
                     stringsAsFactors = FALSE)
col.df$color[grep(&quot;grey&quot;, col.df$color)] &lt;- &quot;grey39&quot;
mcols(res.gr) &lt;- dplyr::left_join(data.frame(mcols(res.gr)[,c(1:2,10)]),
                                      col.df)
colnames(mcols(res.gr))[1] &lt;- &quot;ensembl_gene_id&quot;

mcol.color=4
mcol.name=2
mcol.ensembl=1</code></pre>
<pre class="r"><code>## Plot snps ------------------
snp.plot &lt;- 
  ggplot(snp,
       aes(start, R.squared)) +
  geom_vline(xintercept=snp$start[snp$type==&quot;proxy&quot; &amp; snp$lab!=&quot;&quot;],
             lty=2, color=&quot;grey&quot;) +
  geom_point(aes(shape=type, size=R.squared, fill=idTag)) +
  scale_shape_manual(values=c(21, 23), name=&quot;Type of variant&quot;) +
  scale_size_continuous(guide=F) +
  scale_fill_discrete(name=&quot;Leading variant&quot;) +
  ggrepel::geom_text_repel(aes(label=lab)) +
  scale_y_continuous(name=expression(R^2),
                     limits=c(0.8,1),
                     expand=c(0,0)) +
  xlim(xlims) +
  themeXblank()

## Plot REs ---------------------
re.plot &lt;- 
  ggplot(re.sel) +
  geom_rect(aes(xmin=start, xmax=end,
                ymin=0, ymax=1, fill=h3k27ac.log2FoldChange)) +
  scale_fill_gradient2(low=&quot;dark grey&quot;,
                         mid=&quot;grey&quot;,
                        high=&quot;dark green&quot;,
                        name=expression(&quot;H3K27ac &quot;*log[2]*&quot; FC&quot;),
                        breaks=scales::pretty_breaks(n=3),
                        midpoint=0,
                        guide = guide_colorbar(direction = &quot;horizontal&quot;,
                                               title.position=&quot;top&quot;)) +
  xlim(xlims) +
  themeYblank() +
  themeXblank()

## Plot genes ------------------
plot.genes &lt;- plotGenes(genes=res.gr[res.gr$type!=&quot;not-expressed&quot;,],
                          which=win,
                          mcol.color=mcol.color,
                          mcol.name=mcol.name,
                          mcol.ensembl=mcol.ensembl)

g.plots &lt;-
    ggplot() +
    plot.genes +
    scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(win), &quot;(Mb)&quot;),
                       breaks=scales::pretty_breaks(),
                       limits=xlims,
                       labels=function(x) round(x/1e6, 2),
                       expand=c(0,0)) +
    scale_fill_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    scale_color_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    themeYblank()

## Create legend --------------
snps.leg &lt;- get_legend(snp.plot)
re.leg &lt;- get_legend(re.plot)
gene.leg &lt;- get_legend(g.plots)
  
legends &lt;- plot_grid(snps.leg, re.leg, gene.leg, ncol=1)

## Plot all -------------------
main &lt;- 
  plot_grid(snp.plot + theme(legend.position=&quot;none&quot;,
                             plot.margin=margin(1,0,0,0, &quot;cm&quot;)), 
          re.plot + theme(legend.position=&quot;none&quot;), 
          g.plots + theme(legend.position=&quot;none&quot;),
          ncol=1, align=&quot;v&quot;, rel_heights=c(0.65, 0.1, 0.25))

plot_grid(main, legends, rel_widths=c(0.7, 0.3))</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-11-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/986b505fa6799362ed6a8a0f743a6821e97d5f98/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-11-1.png" target="_blank">986b505</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="fgf2" class="section level2">
<h2>FGF2</h2>
<pre class="r"><code>## SNPs ----------------
rsids &lt;- c(&quot;rs3024505&quot;)

load(&quot;../data/CYT/T1D_GWAS/T1Drm_LD0.8_VSE_AVS.rda&quot;)
load(&quot;../data/CYT/T1D_GWAS/T1Drm_risk_proxies_granges.rda&quot;)
t1d.dat &lt;- data.frame(mcols(proxies))[,c(1,5)]
t1d.dat &lt;- t1d.dat[t1d.dat$R.squared&gt;=0.8,]

snp &lt;- unlist(t1d.avs)
snp &lt;- data.frame(snp[snp$idTag %in% rsids,])

snp$type &lt;- &quot;proxy&quot;
snp$type[snp$idTag==snp$idLd] &lt;- &quot;leading&quot;
snp$lab &lt;- &quot;&quot;
snp$lab[snp$idLd %in% c(rsids, &quot;rs3024493&quot;)] &lt;- snp$idLd[snp$idLd %in% c(rsids, &quot;rs3024493&quot;)]

snp &lt;- dplyr::left_join(snp,
                        t1d.dat,
                        by=c(&quot;idLd&quot;=&quot;ID&quot;))

snp$type &lt;- factor(snp$type, levels=c(&quot;proxy&quot;, &quot;leading&quot;))
snp &lt;- snp[order(snp$type),]

win &lt;- GRanges(paste0(unique(snp$seqnames), &quot;:&quot;, min(snp$start), &quot;-&quot;, max(snp$start)))
win &lt;- resize(win, width=81e3, fix=&quot;start&quot;)
xlims &lt;- c(start(win), end(win))

## RE --------------------
load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_subgroup.rda&quot;)
re.sel &lt;- subsetByOverlaps(re, win)
re.sel &lt;- resize(re.sel, width=1e3, fix=&quot;center&quot;)
re.sel &lt;- data.frame(re.sel)

## Genes -------------------
load(&quot;../data/CYT/RNA/diffAnalysis/RNA_hi_GRangesBatch.rda&quot;)
res.gr$type[res.gr$baseMean&lt;=1] &lt;- &quot;not-expressed&quot;
res.gr &lt;- res.gr[res.gr$gene_biotype==&quot;protein_coding&quot;,]

col.df &lt;- data.frame(&quot;type&quot;=c(names(pals$differential), &quot;not-expressed&quot;),
                     &quot;color&quot;=c(pals$differential, &quot;black&quot;),
                     stringsAsFactors = FALSE)
col.df$color[grep(&quot;grey&quot;, col.df$color)] &lt;- &quot;grey39&quot;
mcols(res.gr) &lt;- dplyr::left_join(data.frame(mcols(res.gr)[,c(1:2,10)]),
                                      col.df)
colnames(mcols(res.gr))[1] &lt;- &quot;ensembl_gene_id&quot;

mcol.color=4
mcol.name=2
mcol.ensembl=1</code></pre>
<pre class="r"><code>## Plot snps ------------------
snp.plot &lt;- 
  ggplot(snp,
       aes(start, R.squared)) +
  geom_vline(xintercept=snp$start[snp$type==&quot;proxy&quot; &amp; snp$lab!=&quot;&quot;],
             lty=2, color=&quot;grey&quot;) +
  geom_point(aes(shape=type, size=R.squared, fill=idTag)) +
  scale_shape_manual(values=c(21, 23), name=&quot;Type of variant&quot;) +
  scale_size_continuous(guide=F) +
  scale_fill_discrete(name=&quot;Leading variant&quot;) +
  ggrepel::geom_text_repel(aes(label=lab)) +
  scale_y_continuous(name=expression(R^2),
                     limits=c(0.8,1),
                     expand=c(0,0)) +
  xlim(xlims) +
  themeXblank()

## Plot REs ---------------------
re.plot &lt;- 
  ggplot(re.sel) +
  geom_rect(aes(xmin=start, xmax=end,
                ymin=0, ymax=1, fill=h3k27ac.log2FoldChange)) +
  scale_fill_gradient2(low=&quot;dark grey&quot;,
                         mid=&quot;grey&quot;,
                        high=&quot;dark green&quot;,
                        name=expression(&quot;H3K27ac &quot;*log[2]*&quot; FC&quot;),
                        breaks=scales::pretty_breaks(n=3),
                        midpoint=0,
                        guide = guide_colorbar(direction = &quot;horizontal&quot;,
                                               title.position=&quot;top&quot;)) +
  xlim(xlims) +
  themeYblank() +
  themeXblank()

## Plot genes ------------------
plot.genes &lt;- plotGenes(genes=res.gr[res.gr$type!=&quot;not-expressed&quot;,],
                          which=win,
                          mcol.color=mcol.color,
                          mcol.name=mcol.name,
                          mcol.ensembl=mcol.ensembl)

g.plots &lt;-
    ggplot() +
    plot.genes +
    scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(win), &quot;(Mb)&quot;),
                       breaks=scales::pretty_breaks(),
                       limits=xlims,
                       labels=function(x) round(x/1e6, 2),
                       expand=c(0,0)) +
    scale_fill_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    scale_color_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    themeYblank()

## Create legend --------------
snps.leg &lt;- get_legend(snp.plot)
re.leg &lt;- get_legend(re.plot)
gene.leg &lt;- get_legend(g.plots)
  
legends &lt;- plot_grid(snps.leg, re.leg, gene.leg, ncol=1)

## Plot all -------------------
main &lt;- 
  plot_grid(snp.plot + theme(legend.position=&quot;none&quot;,
                             plot.margin=margin(1,0,0,0, &quot;cm&quot;)), 
          re.plot + theme(legend.position=&quot;none&quot;), 
          g.plots + theme(legend.position=&quot;none&quot;),
          ncol=1, align=&quot;v&quot;, rel_heights=c(0.65, 0.1, 0.25))

plot_grid(main, legends, rel_widths=c(0.7, 0.3))</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-13-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/986b505fa6799362ed6a8a0f743a6821e97d5f98/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-13-1.png" target="_blank">986b505</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="ptpn2" class="section level1">
<h1>PTPN2</h1>
<pre class="r"><code>## SNPs ----------------
rsids &lt;- c(&quot;rs12971201&quot;)

load(&quot;../data/CYT/T1D_GWAS/T1Drm_LD0.8_VSE_AVS.rda&quot;)
load(&quot;../data/CYT/T1D_GWAS/T1Drm_risk_proxies_granges.rda&quot;)
t1d.dat &lt;- data.frame(mcols(proxies))[,c(1,5)]
t1d.dat &lt;- t1d.dat[t1d.dat$R.squared&gt;=0.8,]

snp &lt;- unlist(t1d.avs)
snp &lt;- data.frame(snp[snp$idTag %in% rsids,])

snp$type &lt;- &quot;proxy&quot;
snp$type[snp$idTag==snp$idLd] &lt;- &quot;leading&quot;
snp$lab &lt;- &quot;&quot;
snp$lab[snp$idLd %in% c(rsids, &quot;rs2847281&quot;)] &lt;- snp$idLd[snp$idLd %in% c(rsids, &quot;rs2847281&quot;)]

snp &lt;- dplyr::left_join(snp,
                        t1d.dat,
                        by=c(&quot;idLd&quot;=&quot;ID&quot;))

snp$type &lt;- factor(snp$type, levels=c(&quot;proxy&quot;, &quot;leading&quot;))
snp &lt;- snp[order(snp$type),]

win &lt;- GRanges(paste0(unique(snp$seqnames), &quot;:&quot;, min(snp$start), &quot;-&quot;, max(snp$start)))
win &lt;- resize(win, width=150e3, fix=&quot;center&quot;)
xlims &lt;- c(start(win), end(win))

## RE --------------------
load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_subgroup.rda&quot;)
re.sel &lt;- subsetByOverlaps(re, win)
re.sel &lt;- resize(re.sel, width=1e3, fix=&quot;center&quot;)
re.sel &lt;- data.frame(re.sel)

## Genes -------------------
load(&quot;../data/CYT/RNA/diffAnalysis/RNA_hi_GRangesBatch.rda&quot;)
res.gr$type[res.gr$baseMean&lt;=1] &lt;- &quot;not-expressed&quot;
res.gr &lt;- res.gr[res.gr$gene_biotype==&quot;protein_coding&quot;,]

col.df &lt;- data.frame(&quot;type&quot;=c(names(pals$differential), &quot;not-expressed&quot;),
                     &quot;color&quot;=c(pals$differential, &quot;black&quot;),
                     stringsAsFactors = FALSE)
col.df$color[grep(&quot;grey&quot;, col.df$color)] &lt;- &quot;grey39&quot;
mcols(res.gr) &lt;- dplyr::left_join(data.frame(mcols(res.gr)[,c(1:2,10)]),
                                      col.df)
colnames(mcols(res.gr))[1] &lt;- &quot;ensembl_gene_id&quot;

mcol.color=4
mcol.name=2
mcol.ensembl=1</code></pre>
<pre class="r"><code>## Plot snps ------------------
snp.plot &lt;- 
  ggplot(snp,
       aes(start, R.squared)) +
  geom_vline(xintercept=snp$start[snp$type==&quot;proxy&quot; &amp; snp$lab!=&quot;&quot;],
             lty=2, color=&quot;grey&quot;) +
  geom_point(aes(shape=type, size=R.squared, fill=idTag)) +
  scale_shape_manual(values=c(21, 23), name=&quot;Type of variant&quot;) +
  scale_size_continuous(guide=F) +
  scale_fill_discrete(name=&quot;Leading variant&quot;) +
  ggrepel::geom_text_repel(aes(label=lab)) +
  scale_y_continuous(name=expression(R^2),
                     limits=c(0.8,1),
                     expand=c(0,0)) +
  xlim(xlims) +
  themeXblank()

## Plot REs ---------------------
re.plot &lt;- 
  ggplot(re.sel) +
  geom_rect(aes(xmin=start, xmax=end,
                ymin=0, ymax=1, fill=h3k27ac.log2FoldChange)) +
  scale_fill_gradient2(low=&quot;dark grey&quot;,
                         mid=&quot;grey&quot;,
                        high=&quot;dark green&quot;,
                        name=expression(&quot;H3K27ac &quot;*log[2]*&quot; FC&quot;),
                        breaks=scales::pretty_breaks(n=3),
                        midpoint=0,
                        guide = guide_colorbar(direction = &quot;horizontal&quot;,
                                               title.position=&quot;top&quot;)) +
  xlim(xlims) +
  themeYblank() +
  themeXblank()

## Plot genes ------------------
plot.genes &lt;- plotGenes(genes=res.gr[res.gr$type!=&quot;not-expressed&quot;,],
                          which=win,
                          mcol.color=mcol.color,
                          mcol.name=mcol.name,
                          mcol.ensembl=mcol.ensembl)

g.plots &lt;-
    ggplot() +
    plot.genes +
    scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(win), &quot;(Mb)&quot;),
                       breaks=scales::pretty_breaks(),
                       limits=xlims,
                       labels=function(x) round(x/1e6, 2),
                       expand=c(0,0)) +
    scale_fill_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    scale_color_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    themeYblank()

## Create legend --------------
snps.leg &lt;- get_legend(snp.plot)
re.leg &lt;- get_legend(re.plot)
gene.leg &lt;- get_legend(g.plots)
  
legends &lt;- plot_grid(snps.leg, re.leg, gene.leg, ncol=1)

## Plot all -------------------
main &lt;- 
  plot_grid(snp.plot + theme(legend.position=&quot;none&quot;,
                             plot.margin=margin(1,0,0,0, &quot;cm&quot;)), 
          re.plot + theme(legend.position=&quot;none&quot;), 
          g.plots + theme(legend.position=&quot;none&quot;),
          ncol=1, align=&quot;v&quot;, rel_heights=c(0.65, 0.1, 0.25))

plot_grid(main, legends, rel_widths=c(0.7, 0.3))</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-15-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/39f446f931b1ad9d7ce6bc97a9c85b70886b5343/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-15-1.png" target="_blank">39f446f</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="nupr1" class="section level2">
<h2>NUPR1</h2>
<pre class="r"><code>## SNPs ----------------
rsids &lt;- c(&quot;rs4788084&quot;)

load(&quot;../data/CYT/T1D_GWAS/T1Drm_LD0.8_VSE_AVS.rda&quot;)
load(&quot;../data/CYT/T1D_GWAS/T1Drm_risk_proxies_granges.rda&quot;)
t1d.dat &lt;- data.frame(mcols(proxies))[,c(1,5)]
t1d.dat &lt;- t1d.dat[t1d.dat$R.squared&gt;=0.8,]

snp &lt;- unlist(t1d.avs)
snp &lt;- data.frame(snp[snp$idTag %in% rsids,])

snp$type &lt;- &quot;proxy&quot;
snp$type[snp$idTag==snp$idLd] &lt;- &quot;leading&quot;
snp$lab &lt;- &quot;&quot;
snp$lab[snp$idLd %in% c(rsids, &quot;rs12446550&quot;)] &lt;- snp$idLd[snp$idLd %in% c(rsids, &quot;rs12446550&quot;)]

snp &lt;- dplyr::left_join(snp,
                        t1d.dat,
                        by=c(&quot;idLd&quot;=&quot;ID&quot;))

snp$type &lt;- factor(snp$type, levels=c(&quot;proxy&quot;, &quot;leading&quot;))
snp &lt;- snp[order(snp$type),]

win &lt;- GRanges(paste0(unique(snp$seqnames), &quot;:&quot;, min(snp$start), &quot;-&quot;, max(snp$start)))
win &lt;- resize(win, width=80e3, fix=&quot;start&quot;)
xlims &lt;- c(start(win), end(win))

## RE --------------------
load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_subgroup.rda&quot;)
re.sel &lt;- subsetByOverlaps(re, win)
re.sel &lt;- resize(re.sel, width=1e3, fix=&quot;center&quot;)
re.sel &lt;- data.frame(re.sel)

## Genes -------------------
load(&quot;../data/CYT/RNA/diffAnalysis/RNA_hi_GRangesBatch.rda&quot;)
res.gr$type[res.gr$baseMean&lt;=1] &lt;- &quot;not-expressed&quot;
res.gr &lt;- res.gr[res.gr$gene_biotype==&quot;protein_coding&quot;,]

col.df &lt;- data.frame(&quot;type&quot;=c(names(pals$differential), &quot;not-expressed&quot;),
                     &quot;color&quot;=c(pals$differential, &quot;black&quot;),
                     stringsAsFactors = FALSE)
col.df$color[grep(&quot;grey&quot;, col.df$color)] &lt;- &quot;grey39&quot;
mcols(res.gr) &lt;- dplyr::left_join(data.frame(mcols(res.gr)[,c(1:2,10)]),
                                      col.df)
colnames(mcols(res.gr))[1] &lt;- &quot;ensembl_gene_id&quot;

mcol.color=4
mcol.name=2
mcol.ensembl=1</code></pre>
<pre class="r"><code>## Plot snps ------------------
snp.plot &lt;- 
  ggplot(snp,
       aes(start, R.squared)) +
  geom_vline(xintercept=snp$start[snp$type==&quot;proxy&quot; &amp; snp$lab!=&quot;&quot;],
             lty=2, color=&quot;grey&quot;) +
  geom_point(aes(shape=type, size=R.squared, fill=idTag)) +
  scale_shape_manual(values=c(21, 23), name=&quot;Type of variant&quot;) +
  scale_size_continuous(guide=F) +
  scale_fill_discrete(name=&quot;Leading variant&quot;) +
  ggrepel::geom_text_repel(aes(label=lab)) +
  scale_y_continuous(name=expression(R^2),
                     limits=c(0.8,1),
                     expand=c(0,0)) +
  xlim(xlims) +
  themeXblank()

## Plot REs ---------------------
re.plot &lt;- 
  ggplot(re.sel) +
  geom_rect(aes(xmin=start, xmax=end,
                ymin=0, ymax=1, fill=h3k27ac.log2FoldChange)) +
  scale_fill_gradient2(low=&quot;dark grey&quot;,
                         mid=&quot;grey&quot;,
                        high=&quot;dark green&quot;,
                        name=expression(&quot;H3K27ac &quot;*log[2]*&quot; FC&quot;),
                        breaks=scales::pretty_breaks(n=3),
                        midpoint=0,
                        guide = guide_colorbar(direction = &quot;horizontal&quot;,
                                               title.position=&quot;top&quot;)) +
  xlim(xlims) +
  themeYblank() +
  themeXblank()

## Plot genes ------------------
plot.genes &lt;- plotGenes(genes=res.gr[res.gr$type!=&quot;not-expressed&quot;,],
                          which=win,
                          mcol.color=mcol.color,
                          mcol.name=mcol.name,
                          mcol.ensembl=mcol.ensembl)

g.plots &lt;-
    ggplot() +
    plot.genes +
    scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(win), &quot;(Mb)&quot;),
                       breaks=scales::pretty_breaks(),
                       limits=xlims,
                       labels=function(x) round(x/1e6, 2),
                       expand=c(0,0)) +
    scale_fill_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    scale_color_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    themeYblank()

## Create legend --------------
snps.leg &lt;- get_legend(snp.plot)
re.leg &lt;- get_legend(re.plot)
gene.leg &lt;- get_legend(g.plots)
  
legends &lt;- plot_grid(snps.leg, re.leg, gene.leg, ncol=1)

## Plot all -------------------
main &lt;- 
  plot_grid(snp.plot + theme(legend.position=&quot;none&quot;,
                             plot.margin=margin(1,0,0,0, &quot;cm&quot;)), 
          re.plot + theme(legend.position=&quot;none&quot;), 
          g.plots + theme(legend.position=&quot;none&quot;),
          ncol=1, align=&quot;v&quot;, rel_heights=c(0.65, 0.1, 0.25))

plot_grid(main, legends, rel_widths=c(0.7, 0.3))</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-17-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/39f446f931b1ad9d7ce6bc97a9c85b70886b5343/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-17-1.png" target="_blank">39f446f</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gsdmb" class="section level2">
<h2>GSDMB</h2>
<pre class="r"><code>## SNPs ----------------
rsids &lt;- c(&quot;rs12453507&quot;)

load(&quot;../data/CYT/T1D_GWAS/T1Drm_LD0.8_VSE_AVS.rda&quot;)
load(&quot;../data/CYT/T1D_GWAS/T1Drm_risk_proxies_granges.rda&quot;)
t1d.dat &lt;- data.frame(mcols(proxies))[proxies$topSNP %in% rsids,c(1,5)]
t1d.dat &lt;- t1d.dat[t1d.dat$R.squared&gt;=0.8,]

snp &lt;- unlist(t1d.avs)
snp &lt;- data.frame(snp[snp$idTag %in% rsids,])
row.gr &lt;- proxies[proxies$ID %in% rsids &amp; proxies$CHOSEN,]
row &lt;- data.frame(seqnames=as.character(seqnames(row.gr)),
                  start=start(row.gr), 
                  end=end(row.gr),
                  width=1,
                  strand=&quot;*&quot;,
                  idLd=row.gr$ID,
                  idTag=row.gr$ID)
snp &lt;- rbind(snp, row)

snp$type &lt;- &quot;proxy&quot;
snp$type[snp$idTag==snp$idLd] &lt;- &quot;leading&quot;
snp$lab &lt;- &quot;&quot;
snp$lab[snp$idLd %in% c(rsids, &quot;rs7219923&quot;, &quot;rs201389301&quot;)] &lt;- snp$idLd[snp$idLd %in% c(rsids, &quot;rs7219923&quot;, &quot;rs201389301&quot;)]

snp &lt;- dplyr::left_join(snp,
                        t1d.dat,
                        by=c(&quot;idLd&quot;=&quot;ID&quot;))

snp$type &lt;- factor(snp$type, levels=c(&quot;proxy&quot;, &quot;leading&quot;))
snp &lt;- snp[order(snp$type),]

win &lt;- GRanges(paste0(unique(snp$seqnames), &quot;:&quot;, min(snp$start), &quot;-&quot;, max(snp$start)))
win &lt;- resize(win, width=30e3, fix=&quot;end&quot;)
xlims &lt;- c(start(win), end(win))

## RE --------------------
load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_subgroup.rda&quot;)
re.sel &lt;- subsetByOverlaps(re, win)
re.sel &lt;- resize(re.sel, width=1e3, fix=&quot;center&quot;)
re.sel &lt;- data.frame(re.sel)

## Genes -------------------
load(&quot;../data/CYT/RNA/diffAnalysis/RNA_hi_GRangesBatch.rda&quot;)
res.gr$type[res.gr$baseMean&lt;=1] &lt;- &quot;not-expressed&quot;
res.gr &lt;- res.gr[res.gr$gene_biotype==&quot;protein_coding&quot;,]

col.df &lt;- data.frame(&quot;type&quot;=c(names(pals$differential), &quot;not-expressed&quot;),
                     &quot;color&quot;=c(pals$differential, &quot;black&quot;),
                     stringsAsFactors = FALSE)
col.df$color[grep(&quot;grey&quot;, col.df$color)] &lt;- &quot;grey39&quot;
mcols(res.gr) &lt;- dplyr::left_join(data.frame(mcols(res.gr)[,c(1:2,10)]),
                                      col.df)
colnames(mcols(res.gr))[1] &lt;- &quot;ensembl_gene_id&quot;

mcol.color=4
mcol.name=2
mcol.ensembl=1</code></pre>
<pre class="r"><code>## Plot snps ------------------
snp.plot &lt;- 
  ggplot(snp,
       aes(start, R.squared)) +
  geom_vline(xintercept=snp$start[snp$type==&quot;proxy&quot; &amp; snp$lab!=&quot;&quot;],
             lty=2, color=&quot;grey&quot;) +
  geom_point(aes(shape=type, size=R.squared, fill=idTag)) +
  scale_shape_manual(values=c(21, 23), name=&quot;Type of variant&quot;) +
  scale_size_continuous(guide=F) +
  scale_fill_discrete(name=&quot;Leading variant&quot;) +
  ggrepel::geom_text_repel(aes(label=lab)) +
  scale_y_continuous(name=expression(R^2),
                     limits=c(0.8,1),
                     expand=c(0,0)) +
  xlim(xlims) +
  themeXblank()

## Plot REs ---------------------
re.plot &lt;- 
  ggplot(re.sel) +
  geom_rect(aes(xmin=start, xmax=end,
                ymin=0, ymax=1, fill=h3k27ac.log2FoldChange)) +
  scale_fill_gradient2(low=&quot;dark grey&quot;,
                         mid=&quot;grey&quot;,
                        high=&quot;dark green&quot;,
                        name=expression(&quot;H3K27ac &quot;*log[2]*&quot; FC&quot;),
                        breaks=scales::pretty_breaks(n=3),
                        midpoint=0,
                        limits=c(0,2.5),
                        guide = guide_colorbar(direction = &quot;horizontal&quot;,
                                               title.position=&quot;top&quot;)) +
  xlim(xlims) +
  themeYblank() +
  themeXblank()

## Plot genes ------------------
plot.genes &lt;- plotGenes(genes=res.gr[res.gr$type!=&quot;not-expressed&quot;,],
                          which=win,
                          mcol.color=mcol.color,
                          mcol.name=mcol.name,
                          mcol.ensembl=mcol.ensembl)

g.plots &lt;-
    ggplot() +
    plot.genes +
    scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(win), &quot;(Mb)&quot;),
                       breaks=scales::pretty_breaks(),
                       limits=xlims,
                       labels=function(x) round(x/1e6, 2),
                       expand=c(0,0)) +
    scale_fill_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    scale_color_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    themeYblank()

## Create legend --------------
snps.leg &lt;- get_legend(snp.plot)
re.leg &lt;- get_legend(re.plot)
gene.leg &lt;- get_legend(g.plots)
  
legends &lt;- plot_grid(snps.leg, re.leg, gene.leg, ncol=1)

## Plot all -------------------
main &lt;- 
  plot_grid(snp.plot + theme(legend.position=&quot;none&quot;,
                             plot.margin=margin(1,0,0,0, &quot;cm&quot;)), 
          re.plot + theme(legend.position=&quot;none&quot;), 
          g.plots + theme(legend.position=&quot;none&quot;),
          ncol=1, align=&quot;v&quot;, rel_heights=c(0.65, 0.1, 0.25))

plot_grid(main, legends, rel_widths=c(0.7, 0.3), scale=c(1,0.5))</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-19-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/39f446f931b1ad9d7ce6bc97a9c85b70886b5343/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-19-1.png" target="_blank">39f446f</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/986b505fa6799362ed6a8a0f743a6821e97d5f98/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-19-1.png" target="_blank">986b505</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="validated-snp-views-credible-sets" class="section level1">
<h1>Validated SNP views &amp; Credible Sets</h1>
<pre><code>Rscript CYT_getCredibleSets.R</code></pre>
<pre class="r"><code>df &lt;- read.delim(&quot;../data/CYT/UMI4C/UMI4C_promoters_views.tsv&quot;, stringsAsFactors=T, header=T)

## RE ---------------------------------------------------------------
load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_subgroup.rda&quot;)

## Genes ------------------------------------------------------------
load(&quot;../data/CYT/RNA/diffAnalysis/RNA_hi_GRangesBatch.rda&quot;)
res.gr$type[res.gr$baseMean&lt;=1] &lt;- &quot;Not-expressed&quot;
res.gr &lt;- res.gr[res.gr$gene_biotype==&quot;protein_coding&quot;,]

col.df &lt;- data.frame(&quot;type&quot;=c(names(pals$differential), &quot;Not-expressed&quot;),
                     &quot;color&quot;=c(pals$differential, &quot;black&quot;),
                     stringsAsFactors = FALSE)
pals$differential &lt;- c(pals$differential, &quot;Not-expressed&quot;=&quot;black&quot;)
pals[&quot;stable&quot;] &lt;- &quot;grey39&quot;
col.df$color[grep(&quot;grey&quot;, col.df$color)] &lt;- &quot;grey39&quot;
mcols(res.gr) &lt;- unique(dplyr::left_join(data.frame(mcols(res.gr)[,c(1:2,10)]),
                                      col.df,
                                  by=c(&quot;type&quot;=&quot;type&quot;)))
colnames(mcols(res.gr))[1] &lt;- &quot;ensembl_gene_id&quot;

mcol.color=4
mcol.name=2
mcol.ensembl=1</code></pre>
<div id="rs78037977" class="section level2">
<h2>rs78037977</h2>
<pre class="r"><code>i &lt;- grep(&quot;rs78037977&quot;, df$bait)

load(&quot;../output/cyt_04_contacts/UMI4C_norm_results_rs78037977.rda&quot;)
  
xlim &lt;- c(df$start[i], df$end[i])
region &lt;- GRanges(seqnames=paste0(seqnames(res$bait)),
                  ranges=IRanges(start=xlim[1],
                                 end=xlim[2]))

## CTCF --------------------------------------------------------------
sm &lt;- 200

ctcf &lt;- rtracklayer::import(&quot;~/data/TFsIslet_hg19/visualization/CTCF_HI32_hg19-extended_compressed_RPKM.bw&quot;,
                            which=region)
score(ctcf) &lt;- zoo::rollmean(score(ctcf), sm,
                                fill=c(NA, NA, NA))

plot.ctcf &lt;-
  ggplot(data.frame(ctcf)) +
  geom_area(aes(x=start,
                y=score),
            fill=&quot;purple3&quot;) +
  scale_y_continuous(name=&quot;CTCF&quot;,
                     limits=c(0,20),
                     expand=c(0,0),
                     breaks=c(0,20)) +
  scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(region), &quot;(Mb)&quot;),
                     breaks=scales::pretty_breaks(),
                     limits=xlim,
                     labels=function(x) round(x/1e6, 2),
                     expand=c(0,0)) +
  themeXblank() +
  theme(plot.margin=margin(0.2,0,0.2,0, &quot;cm&quot;))
  
### Genes ------------------
plot.genes &lt;- plotGenes(genes=res.gr[res.gr$type!=&quot;not-expressed&quot;,],
                        which=region,
                        mcol.color=mcol.color,
                        mcol.name=mcol.name,
                        mcol.ensembl=mcol.ensembl)
  
g.plots &lt;-
  ggplot() +
  plot.genes +
  scale_fill_manual(values=pals$differential,
                     name=&quot;Gene expression&quot;,
                     labels=c(gained=&quot;Up-regulated&quot;,
                              lost=&quot;Down-regulated&quot;,
                              stable=&quot;Equal-regulated&quot;,
                              `Not-expressed`=&quot;Not-expressed&quot;)) + 
  scale_color_manual(values=pals$differential,
                     name=&quot;Gene expression&quot;,
                     labels=c(gained=&quot;Up-regulated&quot;,
                              lost=&quot;Down-regulated&quot;,
                              stable=&quot;Equal-regulated&quot;,
                              `Not-expressed`=&quot;Not-expressed&quot;)) + 
  themeYblank()  +
  scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(region), &quot;(Mb)&quot;),
                     breaks=scales::pretty_breaks(),
                     limits=xlim,
                     labels=function(x) round(x/1e6, 2),
                     expand=c(0,0))
  
### UMI-4C ------------------
umi &lt;-
  ggplot(res$norm_trend,
         aes(start, trend)) +
  geom_ribbon(aes(ymin=devM, ymax=devP, group=interaction(group, sample)),
              color=NA, fill=&quot;light grey&quot;) +
  geom_line(aes(color=sample, group=interaction(group, sample)),
            lwd=0.7) +
  annotate(&quot;point&quot;, x=start(res$bait), y=df$ymax[i], pch=25, fill=&quot;black&quot;,
           size=3) +
  annotate(&quot;text&quot;, x=start(res$bait), y=df$ymax[i]-0.2, label=df$bait[i],
           size=3) +
  scale_color_manual(values=c(ctrl=&quot;#1f78b4&quot;, treat=&quot;#d95f02&quot;),
                     labels=c(&quot;CYT&quot;, &quot;CTRL&quot;), name=&quot;Treatment&quot;) +
  scale_y_continuous(name=&quot;# UMI-4C contacts&quot;,
                     limits=c(0, df$ymax[i]),
                     breaks=scales::pretty_breaks(),
                     expand=c(0,0)) +
  scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(region), &quot;(Mb)&quot;),
                     breaks=scales::pretty_breaks(),
                     limits=xlim,
                     labels=function(x) round(x/1e6, 2),
                     expand=c(0,0)) +
  theme(legend.position=&quot;right&quot;)

### RE ------------------
re.sel &lt;- as.data.frame(subsetByOverlaps(resize(re, 3e3, fix=&quot;center&quot;),
                                         region))

ire &lt;-
  ggplot(re.sel) +
  geom_rect(aes(xmin=start, xmax=end, ymin=0, ymax=1, fill=h3k27ac.log2FoldChange)) +
  scale_fill_gradient2(low=&quot;dark grey&quot;,
                       mid=&quot;grey&quot;,
                      high=&quot;dark green&quot;,
                      name=expression(&quot;H3K27ac &quot;*log[2]*&quot; FC&quot;),
                      breaks=scales::pretty_breaks(n=3),
                      midpoint=0,
                      guide = guide_colorbar(direction = &quot;horizontal&quot;,
                                             title.position=&quot;top&quot;)) +
  scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(region), &quot;(Mb)&quot;),
                     breaks=scales::pretty_breaks(),
                     limits=xlim,
                     labels=function(x) round(x/1e6, 2),
                     expand=c(0,0))

  
### Get legends -----------
gene.leg &lt;- get_legend(g.plots)
umi.leg &lt;- get_legend(umi)
ire.leg &lt;- get_legend(ire)

  
legends &lt;- plot_grid(umi.leg, ire.leg, gene.leg, ncol=1)
### Grid ------------------
p &lt;- 
  plot_grid(umi + themeXblank() + theme(legend.position = &quot;none&quot;,
                                        plot.margin=margin(1,0,0,0, &quot;cm&quot;)), 
          ire + themeYblank() + themeXblank() + theme(legend.position = &quot;none&quot;),
          plot.ctcf,
          g.plots + theme(legend.position = &quot;none&quot;),
          ncol=1, 
          rel_heights = c(0.5, 0.08, 0.12, 0.3),
          align=&quot;v&quot;)
  
plot &lt;- 
    plot_grid(p, 
              legends,
              ncol=2, rel_widths=c(0.7, 0.3))

plot</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="zoom-ins" class="section level3">
<h3>Zoom ins</h3>
<pre class="r"><code>win &lt;- 7e3
snp &lt;- GRanges(&quot;chr1:172715702&quot;)
snp &lt;- resize(snp, width=win, fix=&quot;center&quot;)

##-------------------------------
## Load tracks
##-------------------------------
## ATAC-seq
sm.at &lt;- 5

ctrl.at &lt;- rtracklayer::import(&quot;../data/CYT/ATAC/Visualization/ATAC_hi_ctrl.bw&quot;,
                               which=snp)
score(ctrl.at) &lt;- zoo::rollmean(score(ctrl.at), sm.at,
                                fill=c(NA, NA, NA))

cyt.at &lt;- rtracklayer::import(&quot;../data/CYT/ATAC/Visualization/ATAC_hi_cyt.bw&quot;,
                               which=snp)
score(cyt.at) &lt;- zoo::rollmean(score(cyt.at), sm.at,
                               fill=c(NA, NA, NA))

## H3K27ac
sm.ac &lt;- 5

ctrl.ac &lt;- rtracklayer::import(&quot;../data/CYT/H3K27ac/Visualization/H3K27ac_hi_ctrl.bw&quot;,
                               which=snp)
score(ctrl.ac) &lt;- zoo::rollmean(score(ctrl.ac), sm.ac,
                                fill=c(NA, NA, NA))

cyt.ac &lt;- rtracklayer::import(&quot;../data/CYT/H3K27ac/Visualization/H3K27ac_hi_cyt.bw&quot;,
                               which=snp)
score(cyt.ac) &lt;- zoo::rollmean(score(cyt.ac), sm.ac,
                               fill=c(NA, NA, NA))

## Islet TFS
files &lt;- list.files(&quot;~/data/TFsIslet_hg19/visualization&quot;,
                    pattern=&quot;.bw&quot;, full.names=TRUE)[-1]
islet &lt;- lapply(files, rtracklayer::import,
                            which=snp)
for (i in 1:length(islet)) {
  score(islet[[i]]) &lt;- zoo::rollmean(score(islet[[i]]), 10,
                             fill=c(NA, NA, NA))
}

names &lt;- pipelineNGS::getNameFromPath(files, suffix=&quot;.bw&quot;)
len &lt;- sapply(islet, length)

islet &lt;- GRangesList(islet)
islet &lt;- unlist(islet)
islet$TF &lt;- unlist(mapply(rep, names, each=len))

## RE
load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_subgroup.rda&quot;)
re.sel &lt;- subsetByOverlaps(re, snp)

## Conservation
cons &lt;- rtracklayer::import(&quot;~/data/phastCons_46_placentalMammals/placental_mammals.bw&quot;,
                            which=snp)

load(&quot;../data/CYT/T1D_credibleSet/uk_credibleSet_fullr2.rda&quot;)
uk.full &lt;- GRanges(uk.full)
uk &lt;- subsetByOverlaps(uk.full,snp)
uk$lab &lt;- &quot;&quot;
uk$lab[uk$ID==&quot;rs78037977&quot;] &lt;- uk$ID[uk$ID==&quot;rs78037977&quot;]</code></pre>
<pre class="r"><code>##-------------------------------
## Plot tracks
##-------------------------------
xlims &lt;- c(start(ranges(snp)),
           end(ranges(snp)))

cons.p &lt;- 
  ggplot(data.frame(cons)) +
  geom_area(aes(x=start,
                y=score),
            fill=&quot;maroon4&quot;) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,1),
                     expand=c(0,0),
                     breaks=c(0,1)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))

ctrl.at.p &lt;-
  ggplot(data.frame(ctrl.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,60),
                     expand=c(0,0),
                     breaks=c(0,60)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.at.p &lt;-
  ggplot(data.frame(cyt.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,60),
                     expand=c(0,0),
                     breaks=c(0,60)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
  

ctrl.ac.p &lt;-
  ggplot(data.frame(ctrl.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,20),
                     expand=c(0,0),
                     breaks=c(0,20)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.ac.p &lt;-
  ggplot(data.frame(cyt.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,20),
                     expand=c(0,0),
                     breaks=c(0,20)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))

islet.plot &lt;-
  ggplot(data.frame(islet)) +
  geom_area(aes(x=start,
                y=score, 
                fill=TF),
            alpha=0.6, position=&quot;identity&quot;) +
  scale_y_continuous(name=&quot;&quot;,
                     limits=c(0,10),
                     expand=c(0,0),
                     breaks=c(0,10)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;),
        legend.position=&quot;none&quot;)

re.plot &lt;- 
  ggplot(data.frame(re.sel)) +
  geom_rect(aes(xmin=start, xmax=end, 
                ymin=0, ymax=1, fill=h3k27ac.log2FoldChange),
            color=&quot;black&quot;) +
  scale_fill_gradient(low=pals$differential[&quot;stable&quot;],
                      high=pals$differential[&quot;gained&quot;],
                      limits=c(0,3)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeYblank() +
  themeXblank() +
  theme(legend.position=&quot;none&quot;)


snp.plot &lt;- 
  ggplot(data.frame(uk)) +
  geom_vline(xintercept=172715702, lty=2) +
  geom_point(aes(start, -log10(pmeta),
                 pch=inCredible, 
                 size=r2, fill=r2), 
             color=&quot;black&quot;) +
  scale_shape_manual(values=c(21,23),
                     guide=FALSE) +
  geom_text(aes(start, -log10(pmeta),
                                label=lab),
                            alpha=0.8) +
  scale_fill_gradient2(low=&quot;white&quot;,
                       high=&quot;goldenrod3&quot;,
                       guide=FALSE) +
  scale_size_continuous(range=c(0.05, 4), 
                        guide=FALSE) +
  scale_y_continuous(name=expression(-Log[10]*&quot; P&quot;),
                     breaks=scales::pretty_breaks(n=4),
                     expand=c(0,0),
                     limits=c(0,NA)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))

plot_grid(cons.p,
          islet.plot,
          ctrl.at.p,
          cyt.at.p,
          ctrl.ac.p,
          cyt.ac.p,
          snp.plot,
          re.plot,
          ncol=1,
          rel_heights=c(rep(0.12, 6), 0.24, 0.04),
          align=&quot;v&quot;)</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-23-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/39f446f931b1ad9d7ce6bc97a9c85b70886b5343/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-23-1.png" target="_blank">39f446f</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>load(&quot;../data/CYT/RNA/diffAnalysis/RNA_hi_GRangesBatch.rda&quot;)
gene &lt;- res.gr[res.gr$external_gene_name==&quot;TNFSF18&quot;,]
gene &lt;- resize(gene, width=2e4, fix=&quot;center&quot;)
gene$color &lt;- pals$differential[&quot;gained&quot;]
colnames(mcols(gene))[1] &lt;- &quot;ensembl_gene_id&quot;

##-------------------------------
## Load tracks
##-------------------------------
## ATAC-seq
sm.at &lt;- 20

ctrl.at &lt;- rtracklayer::import(&quot;../data/CYT/ATAC/Visualization/ATAC_hi_ctrl.bw&quot;,
                               which=gene)
score(ctrl.at) &lt;- zoo::rollmean(score(ctrl.at), sm.at,
                                fill=c(NA, NA, NA))

cyt.at &lt;- rtracklayer::import(&quot;../data/CYT/ATAC/Visualization/ATAC_hi_cyt.bw&quot;,
                               which=gene)
score(cyt.at) &lt;- zoo::rollmean(score(cyt.at), sm.at,
                               fill=c(NA, NA, NA))

## H3K27ac
sm.ac &lt;- 20

ctrl.ac &lt;- rtracklayer::import(&quot;../data/CYT/H3K27ac/Visualization/H3K27ac_hi_ctrl.bw&quot;,
                               which=gene)
score(ctrl.ac) &lt;- zoo::rollmean(score(ctrl.ac), sm.ac,
                                fill=c(NA, NA, NA))

cyt.ac &lt;- rtracklayer::import(&quot;../data/CYT/H3K27ac/Visualization/H3K27ac_hi_cyt.bw&quot;,
                               which=gene)
score(cyt.ac) &lt;- zoo::rollmean(score(cyt.ac), sm.ac,
                               fill=c(NA, NA, NA))

## RNA-seq
sm.rna &lt;- 20

ctrl.rna &lt;- rtracklayer::import(&quot;../data/CYT/RNA/Visualization/RNA_hi_ctrl.bw&quot;,
                               which=gene)
score(ctrl.rna) &lt;- zoo::rollmean(score(ctrl.rna), sm.rna,
                                fill=c(NA, NA, NA))

cyt.rna &lt;- rtracklayer::import(&quot;../data/CYT/RNA/Visualization/RNA_hi_cyt.bw&quot;,
                               which=gene)
score(cyt.rna) &lt;- zoo::rollmean(score(cyt.rna), sm.rna,
                               fill=c(NA, NA, NA))


## RE
load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_subgroup.rda&quot;)
re.sel &lt;- subsetByOverlaps(re, gene)</code></pre>
<pre class="r"><code>##-------------------------------
## Plot tracks
##-------------------------------
xlims &lt;- c(start(ranges(gene)),
           end(ranges(gene)))


ctrl.at.p &lt;-
  ggplot(data.frame(ctrl.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,30),
                     expand=c(0,0),
                     breaks=c(0,30)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.at.p &lt;-
  ggplot(data.frame(cyt.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,30),
                     expand=c(0,0),
                     breaks=c(0,30)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
  

ctrl.ac.p &lt;-
  ggplot(data.frame(ctrl.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,15),
                     expand=c(0,0),
                     breaks=c(0,15)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.ac.p &lt;-
  ggplot(data.frame(cyt.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,15),
                     expand=c(0,0),
                     breaks=c(0,15)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))

ctrl.rna.p &lt;-
  ggplot(data.frame(ctrl.rna)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,3),
                     expand=c(0,0),
                     breaks=c(0,3)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.rna.p &lt;-
  ggplot(data.frame(cyt.rna)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,3),
                     expand=c(0,0),
                     breaks=c(0,3)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))


genes.p &lt;- 
  ggplot() +
  plotGenes(gene,
            gene,
            mcol.color=11,
            mcol.name=2,
            mcol.ensembl=1)
  
genes.p &lt;- 
  genes.p +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  theme(legend.position=&quot;none&quot;) +
  themeYblank() + 
  themeXblank()


plot_grid(ctrl.at.p,
          cyt.at.p,
          ctrl.ac.p,
          cyt.ac.p,
          ctrl.rna.p,
          cyt.rna.p,
          genes.p,
          ncol=1,
          rel_heights=c(rep(0.16, 6), 0.04),
          align=&quot;v&quot;)</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-25-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/39f446f931b1ad9d7ce6bc97a9c85b70886b5343/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-25-1.png" target="_blank">39f446f</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="dexi" class="section level2">
<h2>DEXI</h2>
<pre class="r"><code>## RE ---------------------------------------------------------------
load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_subgroup.rda&quot;)

## Genes ------------------------------------------------------------
load(&quot;../data/CYT/RNA/diffAnalysis/RNA_hi_GRangesBatch.rda&quot;)
res.gr$type[res.gr$baseMean&lt;=1] &lt;- &quot;Not-expressed&quot;
res.gr &lt;- res.gr[res.gr$gene_biotype==&quot;protein_coding&quot;,]

col.df &lt;- data.frame(&quot;type&quot;=c(names(pals$differential), &quot;Not-expressed&quot;),
                     &quot;color&quot;=c(pals$differential, &quot;black&quot;),
                     stringsAsFactors = FALSE)
pals$differential &lt;- c(pals$differential, &quot;Not-expressed&quot;=&quot;black&quot;)
pals[&quot;stable&quot;] &lt;- &quot;grey39&quot;
col.df$color[grep(&quot;grey&quot;, col.df$color)] &lt;- &quot;grey39&quot;
mcols(res.gr) &lt;- unique(dplyr::left_join(data.frame(mcols(res.gr)[,c(1:2,10)]),
                                      col.df))
colnames(mcols(res.gr))[1] &lt;- &quot;ensembl_gene_id&quot;

mcol.color=4
mcol.name=2
mcol.ensembl=1</code></pre>
<pre class="r"><code>i &lt;- grep(&quot;DEXI&quot;, df$bait)

load(paste0(&quot;../output/cyt_04_contacts/UMI4C_norm_results_DEXI.rda&quot;))
load(&quot;../data/CYT/T1D_credibleSet/T1D_credibleSet_fullR2_rs193778.rda&quot;)

uk$lab &lt;- &quot;&quot;
uk$lab[grep(&quot;rs193778&quot;, uk$ID)] &lt;- &quot;rs193778&quot;
uk &lt;- uk[order(uk$inCredible),]
  
xlim &lt;- c(df$start[i], df$end[i])
region &lt;- GRanges(seqnames=paste0(seqnames(res$bait)),
                  ranges=IRanges(start=xlim[1],
                                 end=xlim[2]))

## H3K27ac --------------------------------------------------------------
sm &lt;- 100

ctrl &lt;- rtracklayer::import(&quot;../data/CYT/H3K27ac/Visualization/H3K27ac_hi_ctrl.bw&quot;,
                            which=region)
score(ctrl) &lt;- zoo::rollmean(score(ctrl), sm,
                                fill=c(NA, NA, NA))

plot.ctrl &lt;-
  ggplot(data.frame(ctrl)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;H3K27ac&quot;,
                     limits=c(0,50),
                     expand=c(0,0),
                     breaks=c(0,50)) +
  scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(region), &quot;(Mb)&quot;),
                     breaks=scales::pretty_breaks(),
                     limits=xlim,
                     labels=function(x) round(x/1e6, 2),
                     expand=c(0,0)) +
  themeXblank() +
  theme(plot.margin=margin(0.2,0,0.2,0, &quot;cm&quot;))

cyt &lt;- rtracklayer::import(&quot;../data/CYT/H3K27ac/Visualization/H3K27ac_hi_cyt.bw&quot;,
                            which=region)
score(cyt) &lt;- zoo::rollmean(score(cyt), sm,
                                fill=c(NA, NA, NA))

plot.cyt &lt;-
  ggplot(data.frame(cyt)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;H3K27ac&quot;,
                     limits=c(0,50),
                     expand=c(0,0),
                     breaks=c(0,50)) +
  scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(region), &quot;(Mb)&quot;),
                     breaks=scales::pretty_breaks(),
                     limits=xlim,
                     labels=function(x) round(x/1e6, 2),
                     expand=c(0,0)) +
  themeXblank() +
  theme(plot.margin=margin(0.2,0,0.2,0, &quot;cm&quot;))
  
### Genes ------------------
colnames(mcols(res.gr))[1] &lt;- &quot;ensembl_gene_id&quot;
plot.genes &lt;- plotGenes(genes=res.gr[res.gr$type!=&quot;not-expressed&quot;,],
                        which=region,
                        mcol.color=mcol.color,
                        mcol.name=mcol.name,
                        mcol.ensembl=mcol.ensembl)
  
g.plots &lt;-
  ggplot() +
  plot.genes +
  xlim(xlim) +
  scale_fill_manual(values=pals$differential,
                     name=&quot;Gene expression&quot;,
                     labels=c(gained=&quot;Up-regulated&quot;,
                              lost=&quot;Down-regulated&quot;,
                              stable=&quot;Equal-regulated&quot;,
                              `Not-expressed`=&quot;Not-expressed&quot;)) + 
  scale_color_manual(values=pals$differential,
                     name=&quot;Gene expression&quot;,
                     labels=c(gained=&quot;Up-regulated&quot;,
                              lost=&quot;Down-regulated&quot;,
                              stable=&quot;Equal-regulated&quot;,
                              `Not-expressed`=&quot;Not-expressed&quot;)) + 
  themeYblank()  +
  scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(region), &quot;(Mb)&quot;),
                     breaks=scales::pretty_breaks(),
                     limits=xlim,
                     labels=function(x) round(x/1e6, 2),
                     expand=c(0,0))
  
### UMI-4C ------------------
umi &lt;-
  ggplot(res$norm_trend,
         aes(start, trend)) +
  geom_ribbon(aes(ymin=devM, ymax=devP, group=interaction(group, sample)),
              color=NA, fill=&quot;light grey&quot;) +
  geom_line(aes(color=sample, group=interaction(group, sample)),
            lwd=0.7) +
  annotate(&quot;point&quot;, x=start(res$bait), y=df$ymax[i], pch=25, fill=&quot;black&quot;,
           size=3) +
  annotate(&quot;text&quot;, x=start(res$bait), y=df$ymax[i]-0.2, label=df$bait[i],
           size=3) +
  geom_point(data=uk,
             aes(start, scales::rescale(-log10(pmeta), to=c(0, df$ymax[i])),
                 fill=r2, shape=inCredible, 
                 alpha=scales::rescale(-log10(pmeta), to=c(0, df$ymax[i])),
                 size=r2)) +
  ggrepel::geom_text_repel(data=uk,
                           aes(start, scales::rescale(-log10(pmeta), to=c(0, df$ymax[i])),
                               label=lab)) +
  geom_vline(xintercept=uk$start[uk$lab!=&quot;&quot;], lty=2, color=&quot;grey&quot;) +
  scale_color_manual(values=c(ctrl=&quot;#1f78b4&quot;, treat=&quot;#d95f02&quot;),
                     labels=c(&quot;CYT&quot;, &quot;CTRL&quot;), name=&quot;Treatment&quot;) +
  scale_alpha_continuous(range=c(0.05, 0.7), guide=FALSE) +
  scale_fill_gradient2(low=&quot;white&quot;, high=&quot;goldenrod3&quot;,
                       limits=c(0,1), breaks=c(0,1),
                       name=expression(R^2),
                       guide=guide_colorbar(direction = &quot;horizontal&quot;,
                                            title.position=&quot;top&quot;)) +
  scale_shape_manual(values=c(21,23)) +
  scale_size_continuous(range=c(0.1,5)) +
  scale_y_continuous(name=&quot;# UMI-4C contacts&quot;,
                     limits=c(0, df$ymax[i]),
                     breaks=scales::pretty_breaks(),
                     expand=c(0,0),
                     sec.axis = sec_axis(~.*(max(-log10(uk$pmeta)/df$ymax[i])),
                                         name=expression(-Log[10]*&quot; P&quot;))) +
  scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(region), &quot;(Mb)&quot;),
                     breaks=scales::pretty_breaks(),
                     limits=xlim,
                     labels=function(x) round(x/1e6, 2),
                     expand=c(0,0)) +
  theme(legend.position=&quot;right&quot;) +
  guides(alpha=F, size=F, shape=F)

### RE ------------------
re.sel &lt;- as.data.frame(subsetByOverlaps(resize(re, 3e3, fix=&quot;center&quot;),
                                         region))

ire &lt;-
  ggplot(re.sel) +
  geom_rect(aes(xmin=start, xmax=end, ymin=0, ymax=1, fill=h3k27ac.log2FoldChange)) +
  scale_fill_gradient2(low=&quot;dark grey&quot;,
                       mid=&quot;grey&quot;,
                      high=&quot;dark green&quot;,
                      name=expression(&quot;H3K27ac &quot;*log[2]*&quot; FC&quot;),
                      breaks=scales::pretty_breaks(n=3),
                      midpoint=0,
                      guide = guide_colorbar(direction = &quot;horizontal&quot;,
                                             title.position=&quot;top&quot;)) +
  scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(region), &quot;(Mb)&quot;),
                     breaks=scales::pretty_breaks(),
                     limits=xlim,
                     labels=function(x) round(x/1e6, 2),
                     expand=c(0,0))

  
### Get legends -----------
gene.leg &lt;- get_legend(g.plots)
umi.leg &lt;- get_legend(umi)
ire.leg &lt;- get_legend(ire)

  
legends &lt;- plot_grid(umi.leg, ire.leg, gene.leg, ncol=1)
### Grid ------------------
p &lt;- 
  plot_grid(umi + themeXblank() + theme(legend.position = &quot;none&quot;,
                                        plot.margin=margin(1,0,0,0, &quot;cm&quot;)), 
          ire + themeYblank() + themeXblank() + theme(legend.position = &quot;none&quot;),
          plot.ctrl,
          plot.cyt,
          g.plots + theme(legend.position = &quot;none&quot;),
          ncol=1, 
          rel_heights = c(0.45, 0.05, 0.08, 0.08, 0.25),
          align=&quot;v&quot;)
  
plot &lt;- 
    plot_grid(p, 
              legends,
              ncol=2, rel_widths=c(0.7, 0.3))

plot</code></pre>
<p><img src="figure/cyt_05_t1d.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/39f446f931b1ad9d7ce6bc97a9c85b70886b5343/docs/figure/cyt_05_t1d.Rmd/unnamed-chunk-27-1.png" target="_blank">39f446f</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-09-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="zoom-ins-1" class="section level2">
<h2>Zoom ins</h2>
<pre class="r"><code>win &lt;- 8e2
snp &lt;- GRanges(&quot;chr16:11351211&quot;)
snp &lt;- resize(snp, width=win, fix=&quot;center&quot;)

##-------------------------------
## Load tracks
##-------------------------------
## ATAC-seq
sm.at &lt;- 5

ctrl.at &lt;- rtracklayer::import(&quot;../data/CYT/ATAC/Visualization/ATAC_hi_ctrl.bw&quot;,
                               which=snp)
score(ctrl.at) &lt;- zoo::rollmean(score(ctrl.at), sm.at,
                                fill=c(NA, NA, NA))

cyt.at &lt;- rtracklayer::import(&quot;../data/CYT/ATAC/Visualization/ATAC_hi_cyt.bw&quot;,
                               which=snp)
score(cyt.at) &lt;- zoo::rollmean(score(cyt.at), sm.at,
                               fill=c(NA, NA, NA))

## H3K27ac
sm.ac &lt;- 5

ctrl.ac &lt;- rtracklayer::import(&quot;../data/CYT/H3K27ac/Visualization/H3K27ac_hi_ctrl.bw&quot;,
                               which=snp)
score(ctrl.ac) &lt;- zoo::rollmean(score(ctrl.ac), sm.ac,
                                fill=c(NA, NA, NA))

cyt.ac &lt;- rtracklayer::import(&quot;../data/CYT/H3K27ac/Visualization/H3K27ac_hi_cyt.bw&quot;,
                               which=snp)
score(cyt.ac) &lt;- zoo::rollmean(score(cyt.ac), sm.ac,
                               fill=c(NA, NA, NA))

## Islet TFS
files &lt;- list.files(&quot;~/data/TFsIslet_hg19/visualization&quot;,
                    pattern=&quot;.bw&quot;, full.names=TRUE)[-1]
islet &lt;- lapply(files, rtracklayer::import,
                            which=snp)
for (i in 1:length(islet)) {
  score(islet[[i]]) &lt;- zoo::rollmean(score(islet[[i]]), 10,
                             fill=c(NA, NA, NA))
}

names &lt;- pipelineNGS::getNameFromPath(files, suffix=&quot;.bw&quot;)
len &lt;- sapply(islet, length)

islet &lt;- GRangesList(islet)
islet &lt;- unlist(islet)
islet$TF &lt;- unlist(mapply(rep, names, each=len))

## RE
load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_subgroup.rda&quot;)
re.sel &lt;- subsetByOverlaps(re, snp)

## Conservation
cons &lt;- rtracklayer::import(&quot;~/data/phastCons_46_placentalMammals/placental_mammals.bw&quot;,
                            which=snp)


load(&quot;../data/CYT/T1D_credibleSet/uk_credibleSet_fullr2.rda&quot;)
uk.full &lt;- GRanges(uk.full)
uk &lt;- subsetByOverlaps(uk.full,snp)
uk$lab &lt;- &quot;&quot;
uk$lab[uk$ID==&quot;rs193778&quot;] &lt;- uk$ID[uk$ID==&quot;rs193778&quot;]</code></pre>
<pre class="r"><code>##-------------------------------
## Plot tracks
##-------------------------------
xlims &lt;- c(start(ranges(snp)),
           end(ranges(snp)))

cons.p &lt;- 
  ggplot(data.frame(cons)) +
  geom_area(aes(x=start,
                y=score),
            fill=&quot;maroon4&quot;) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,1),
                     expand=c(0,0),
                     breaks=c(0,1)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))

ctrl.at.p &lt;-
  ggplot(data.frame(ctrl.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,50),
                     expand=c(0,0),
                     breaks=c(0,50)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.at.p &lt;-
  ggplot(data.frame(cyt.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,50),
                     expand=c(0,0),
                     breaks=c(0,50)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
  

ctrl.ac.p &lt;-
  ggplot(data.frame(ctrl.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,50),
                     expand=c(0,0),
                     breaks=c(0,50)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.ac.p &lt;-
  ggplot(data.frame(cyt.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,50),
                     expand=c(0,0),
                     breaks=c(0,50)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))

islet.plot &lt;-
  ggplot(data.frame(islet)) +
  geom_area(aes(x=start,
                y=score, 
                fill=TF),
            alpha=0.6, position=&quot;identity&quot;) +
  scale_y_continuous(name=&quot;&quot;,
                     limits=c(0,5),
                     expand=c(0,0),
                     breaks=c(0,5)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;),
        legend.position=&quot;none&quot;)

re.plot &lt;- 
  ggplot(data.frame(re.sel)) +
  geom_rect(aes(xmin=start, xmax=end, 
                ymin=0, ymax=1, fill=h3k27ac.log2FoldChange),
            color=&quot;black&quot;) +
  scale_fill_gradient(low=pals$differential[&quot;stable&quot;],
                      high=pals$differential[&quot;gained&quot;],
                      limits=c(0,3)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeYblank() +
  themeXblank() +
  theme(legend.position=&quot;none&quot;)


snp.plot &lt;- 
  ggplot(data.frame(uk)) +
  geom_vline(xintercept=start(uk[uk$lab!=&quot;&quot;]), lty=2) +
  geom_point(aes(start, -log10(pmeta),
                 pch=inCredible, 
                 size=r2, fill=r2), 
             color=&quot;black&quot;) +
  scale_shape_manual(values=c(21,23),
                     guide=FALSE) +
  geom_text(aes(start, -log10(pmeta),
                                label=lab),
                            alpha=0.8) +
  scale_fill_gradient2(low=&quot;white&quot;,
                       high=&quot;goldenrod3&quot;,
                       guide=FALSE) +
  scale_size_continuous(range=c(0.05, 4), 
                        guide=FALSE) +
  scale_y_continuous(name=expression(-Log[10]*&quot; P&quot;),
                     breaks=scales::pretty_breaks(n=4),
                     expand=c(0,0),
                     limits=c(0,NA)) +
  scale_x_continuous(limits=xlims,
                     labels=function(x) round(x/1e6, 3),
                     breaks=scales::pretty_breaks(n=2),
                     name=&quot;Coordinates (Mb)&quot;) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))

plot_grid(cons.p,
          islet.plot,
          ctrl.at.p,
          cyt.at.p,
          ctrl.ac.p,
          cyt.ac.p,
          snp.plot,
          re.plot,
          ncol=1,
          rel_heights=c(rep(0.12, 6), 0.24, 0.04),
          align=&quot;v&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=es_ES.UTF-8        LC_COLLATE=C              
 [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] umi4cCatcheR_0.0.0.9000 VSE_0.99                dplyr_1.0.1            
 [4] kableExtra_1.1.0        cowplot_1.0.0           ggplot2_3.3.2          
 [7] GenomicRanges_1.41.5    GenomeInfoDb_1.25.8     IRanges_2.23.10        
[10] S4Vectors_0.27.12       BiocGenerics_0.35.4     workflowr_1.6.2        

loaded via a namespace (and not attached):
  [1] colorspace_1.4-1            ellipsis_0.3.1             
  [3] rio_0.5.16                  rprojroot_1.3-2            
  [5] XVector_0.29.3              fs_1.5.0                   
  [7] rstudioapi_0.11             farver_2.0.3               
  [9] remotes_2.2.0               bit64_4.0.2                
 [11] ggrepel_0.8.2               AnnotationDbi_1.51.3       
 [13] fansi_0.4.1                 xml2_1.3.2                 
 [15] knitr_1.29                  pkgload_1.1.0              
 [17] Rsamtools_2.5.3             dbplyr_1.4.4               
 [19] pipelineNGS_0.0.0.9000      readr_1.3.1                
 [21] compiler_4.0.2              httr_1.4.2                 
 [23] backports_1.1.8             Matrix_1.2-18              
 [25] assertthat_0.2.1            cli_2.0.2                  
 [27] later_1.1.0.1               htmltools_0.5.0            
 [29] prettyunits_1.1.1           tools_4.0.2                
 [31] igraph_1.2.5                gtable_0.3.0               
 [33] glue_1.4.1                  GenomeInfoDbData_1.2.3     
 [35] rappdirs_0.3.1              Rcpp_1.0.5                 
 [37] Biobase_2.49.0              carData_3.0-4              
 [39] cellranger_1.1.0            vctrs_0.3.2                
 [41] Biostrings_2.57.2           rtracklayer_1.49.4         
 [43] xfun_0.16                   stringr_1.4.0              
 [45] ps_1.3.3                    openxlsx_4.1.5             
 [47] testthat_2.3.2              rvest_0.3.6                
 [49] lifecycle_0.2.0             devtools_2.3.1             
 [51] XML_3.99-0.5                zoo_1.8-8                  
 [53] zlibbioc_1.35.0             scales_1.1.1               
 [55] BSgenome_1.57.5             hms_0.5.3                  
 [57] promises_1.1.1              SummarizedExperiment_1.19.6
 [59] yaml_2.2.1                  curl_4.3                   
 [61] memoise_1.1.0               gridExtra_2.3              
 [63] biomaRt_2.45.2              RSQLite_2.2.0              
 [65] stringi_1.4.6               highr_0.8                  
 [67] desc_1.2.0                  pkgbuild_1.1.0             
 [69] zip_2.0.4                   BiocParallel_1.23.2        
 [71] rlang_0.4.7                 pkgconfig_2.0.3            
 [73] bitops_1.0-6                matrixStats_0.56.0         
 [75] lattice_0.20-41             evaluate_0.14              
 [77] purrr_0.3.4                 GenomicAlignments_1.25.3   
 [79] labeling_0.3                bit_4.0.4                  
 [81] processx_3.4.3              tidyselect_1.1.0           
 [83] magrittr_1.5                bookdown_0.20              
 [85] R6_2.4.1                    generics_0.0.2             
 [87] DBI_1.1.0                   DelayedArray_0.15.7        
 [89] pillar_1.4.6                haven_2.3.1                
 [91] whisker_0.4                 foreign_0.8-80             
 [93] withr_2.2.0                 abind_1.4-5                
 [95] RCurl_1.98-1.2              tibble_3.0.3               
 [97] crayon_1.3.4                car_3.0-8                  
 [99] BiocFileCache_1.13.1        rmarkdown_2.3              
[101] progress_1.2.2              usethis_1.6.1              
[103] grid_4.0.2                  readxl_1.3.1               
[105] data.table_1.13.0           blob_1.2.1                 
[107] callr_3.4.3                 git2r_0.27.1               
[109] forcats_0.5.0               digest_0.6.25              
[111] webshot_0.5.2               httpuv_1.5.4               
[113] regioneR_1.21.1             openssl_1.4.2              
[115] munsell_0.5.0               viridisLite_0.3.0          
[117] askpass_1.1                 sessioninfo_1.1.1          </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
