<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mireia Ramos-Rodríguez" />


<title>Proinflammatory cytokine exposure causes profound remodeling of the β-cell regulatory landscape</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">β-cells and T1D</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-α
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_ifna.html">Description</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-γ + IL-1β
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_cyt.html">Description</a>
    </li>
    <li>
      <a href="cyt_01_remodeling.html">1) Induction of chromatin remodeling</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mireia-bioinfo/phdthesis_code">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Proinflammatory cytokine exposure causes profound remodeling of the β-cell regulatory landscape</h1>
<h4 class="author">Mireia Ramos-Rodríguez</h4>

</div>


<div id="quality-control" class="section level2">
<h2>Quality control</h2>
<div id="atac-seq-specific-qc-measures" class="section level3">
<h3>ATAC-seq specific QC measures</h3>
<p>Run Rscript for generating the necessary quality control measures.</p>
<pre><code>Rscript code/CYT_QC_ATAC.R</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/QC/ATAC_stats.rda&quot;)
load(&quot;../data/CYT/ATAC/QC/QC_scores.rda&quot;)

stats &lt;- dplyr::left_join(stats, txt)

lib &lt;-
  ggplot(stats,
       aes(&quot;Lib Size&quot;,
           sampleID)) +
  geom_tile(aes(fill=total_reads),
              color=&quot;black&quot;, size=1) +
  geom_text(aes(label=round(total_reads/1e6, 1))) +
  scale_fill_gradient(low=&quot;white&quot;,
                      high=&quot;purple3&quot;,
                      limits=c(0, max(stats$total_reads)),
                      breaks=c(0, max(stats$total_reads)),
                      labels=c(&quot;0&quot;, &quot;Max&quot;),
                      name=&quot;Library size (x10^6)&quot;) +
  guides(fill=guide_colourbar(ticks=FALSE,
                              frame.colour = &quot;black&quot;,
                              frame.linewidth = 1)) +
  theme(axis.line.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title = element_blank())

align &lt;-
  ggplot(stats,
         aes(&quot;% Align&quot;,
             sampleID)) + 
  geom_tile(aes(fill=alignment_rate),
            color=&quot;black&quot;, size=1) +
  geom_text(aes(label=round(alignment_rate, 1))) +
  scale_fill_gradient(low=&quot;white&quot;,
                      high=&quot;darkorange3&quot;,
                      limits=c(0, max(stats$alignment_rate)),
                      breaks=c(0, max(stats$alignment_rate)),
                      labels=c(0, &quot;Max&quot;),
                      name=&quot;% alignment&quot;) +
  guides(fill=guide_colourbar(ticks=FALSE,
                              frame.colour = &quot;black&quot;,
                              frame.linewidth = 1)) +
  theme(axis.text.y=element_blank(),
        axis.title=element_blank(),
        axis.line=element_blank(),
        axis.ticks=element_blank(),
        plot.margin=margin(0,0,0,0, unit=&#39;pt&#39;))
  
mit &lt;- 
  ggplot(stats,
       aes(&quot;% Mito&quot;,
           sampleID)) +
  geom_tile(aes(fill=chrM_rate),
            color=&quot;black&quot;, size=1) +
  geom_text(aes(label=round(chrM_rate, 1))) +
  scale_fill_gradient(low=&quot;dodgerblue3&quot;,
                      high=&quot;white&quot;,
                      limits=c(0, max(stats$chrM_rate)),
                      breaks=c(0, max(stats$chrM_rate)),
                      labels=c(&quot;0&quot;, &quot;Max&quot;),
                      name=&quot;% Mitochondrial&quot;) +
  guides(fill=guide_colourbar(ticks=FALSE,
                              frame.colour = &quot;black&quot;,
                              frame.linewidth = 1)) +
  theme(axis.text.y=element_blank(),
        axis.title=element_blank(),
        axis.line=element_blank(),
        axis.ticks=element_blank(),
        plot.margin=margin(0,0,0,0, unit=&#39;pt&#39;))


nsc &lt;- 
  ggplot(stats,
       aes(&quot;NSC&quot;,
           sampleID)) +
  geom_tile(aes(fill=NSC),
            color=&quot;black&quot;, size=1) +
  geom_text(aes(label=round(NSC, 1))) +
  scale_fill_gradient(low=&quot;white&quot;,
                      high=&quot;indianred4&quot;,
                      limits=c(1, max(stats$NSC)),
                      breaks=c(1, max(stats$NSC)),
                      labels=c(&quot;1&quot;, &quot;Max&quot;),
                      name=&quot;NSC&quot;) +
  guides(fill=guide_colourbar(ticks=FALSE,
                              frame.colour = &quot;black&quot;,
                              frame.linewidth = 1)) +
  theme(axis.text.y=element_blank(),
        axis.title=element_blank(),
        axis.line=element_blank(),
        axis.ticks=element_blank(),
        plot.margin=margin(0,0,0,0, unit=&#39;pt&#39;))

rsc &lt;- 
  ggplot(stats,
       aes(&quot;RSC&quot;,
           sampleID)) +
  geom_tile(aes(fill=RSC),
            color=&quot;black&quot;, size=1) +
  geom_text(aes(label=round(RSC, 1))) +
  scale_fill_gradient(low=&quot;white&quot;,
                      high=&quot;turquoise4&quot;,
                      limits=c(0, max(stats$RSC)),
                      breaks=c(0, max(stats$RSC)),
                      labels=c(&quot;0&quot;, &quot;Max&quot;),
                      name=&quot;RSC&quot;) +
  guides(fill=guide_colourbar(ticks=FALSE,
                              frame.colour = &quot;black&quot;,
                              frame.linewidth = 1)) +
  theme(axis.text.y=element_blank(),
        axis.title=element_blank(),
        axis.line=element_blank(),
        axis.ticks=element_blank(),
        plot.margin=margin(0,0,0,0, unit=&#39;pt&#39;))


## Get legend -----------------------
lib.leg &lt;- get_legend(lib)
al.leg &lt;- get_legend(align)
mit.leg &lt;- get_legend(mit)
nsc.leg &lt;- get_legend(nsc)
rsc.leg &lt;- get_legend(rsc)


leg &lt;- plot_grid(lib.leg, 
                 al.leg,
                 mit.leg, 
                 nsc.leg, 
                 rsc.leg,
                 nrow=1)

## Create heatmap --------------------------

heat &lt;- plot_grid(lib + theme(legend.position=&quot;none&quot;), 
                  align + theme(legend.position=&quot;none&quot;),
          mit + theme(legend.position=&quot;none&quot;), 
          nsc + theme(legend.position=&quot;none&quot;), 
          rsc + theme(legend.position=&quot;none&quot;),
          nrow=1,
          align=&#39;h&#39;,
          rel_widths=c(0.30, rep(0.1750, 4))
          )


## Final plot ----------------
qc &lt;- 
  plot_grid(heat, leg, nrow=2, rel_heights = c(0.7, 0.3))

qc</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/atac-heatmap-1.png" alt="Sumary of per-replicate sequencing metrics, showing total library sizes, percentage of aligned reads, percentage of mitochondrial aligned reads, normalized strand cross-correlation coefficient (NSC) and relative strand cross-correlation coefficient (RSC)." width="672" />
<p class="caption">
Sumary of per-replicate sequencing metrics, showing total library sizes, percentage of aligned reads, percentage of mitochondrial aligned reads, normalized strand cross-correlation coefficient (NSC) and relative strand cross-correlation coefficient (RSC).
</p>
</div>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/QC/ATAC_tss_enrichment.rda&quot;)

tss$dataset &lt;- factor(tss$dataset, levels=c(&quot;TSS annotation&quot;, &quot;Random control&quot;))
tss$group &lt;- gsub(&quot;[[:digit:]]*_&quot;, &quot;_&quot;, tss$sample)

tss &lt;- tss %&gt;% 
  group_by(dataset, group, Position) %&gt;%
  summarise(mean=mean(mean))

tss.plot &lt;- 
  ggplot(tss,
       aes(Position, mean)) +
  geom_line(aes(group=dataset, color=dataset), lwd=0.7) +
  scale_color_manual(values=c(&quot;seagreen4&quot;, &quot;goldenrod3&quot;), name=&quot;Dataset&quot;) +
  facet_wrap(~group, scales=&quot;free_y&quot;) +
  theme(legend.position=&quot;top&quot;) +
  ylab(&quot;Mean ATAC-seq read counts&quot;) + xlab(&quot;Position relative to TSS (bp)&quot;)

tss.plot</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/atac-tss-enrichment-1.png" alt="Enrichment of ATAC-seq reads around protein-coding TSS compared to a randomized set of regions." width="576" />
<p class="caption">
Enrichment of ATAC-seq reads around protein-coding TSS compared to a randomized set of regions.
</p>
</div>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/QC/ATAC_noise.rda&quot;)

stats$mean.errorbar &lt;- stats$mean
stats &lt;- stats[order(rev(stats$Annotation)),]

stats &lt;- stats[stats$Annotation!=&quot;Unassigned&quot;,] %&gt;%
  group_by(group) %&gt;%
  mutate(cumsum=cumsum(mean))

noise &lt;- 
  ggplot(stats[stats$Annotation!=&quot;Unassigned&quot;,],
       aes(group, mean)) +
  geom_bar(aes(fill=Annotation), stat=&quot;identity&quot;,
           position=&quot;stack&quot;,
           color=&quot;black&quot;, lwd=0.7) +
  geom_errorbar(aes(ymin=cumsum, ymax=cumsum+sd,
                    group=Annotation),
                width=.3, lwd=0.5) +
  scale_fill_manual(values=c(&quot;violetred&quot;, &quot;dark orange&quot;)) +
  scale_y_continuous(name=&quot;Percentage of reads in peaks (%)&quot;) +
  theme(axis.text.x=element_text(angle=30, hjust=1),
        legend.position=&quot;top&quot;,
        axis.title.x=element_blank())

noise</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/atac-signal-to-noise-1.png" alt="Signal-to-noise ratios of ATAC-seq reads located at called peaks vs reads outside peaks." width="384" />
<p class="caption">
Signal-to-noise ratios of ATAC-seq reads located at called peaks vs reads outside peaks.
</p>
</div>
</div>
<div id="correlation-between-replicates" class="section level3">
<h3>Correlation between replicates</h3>
<pre><code>Rscript code/QC_CORR_genome.R data/CYT/ATAC/BAM/ data/CYT/ATAC/QC/
Rscript code/QC_CORR_genome.R data/CYT/H3K27ac/BAM/ data/CYT/H3K27ac/QC/
Rscript code/QC_CORR_genome.R data/CYT/RNA/BAM/ data/CYT/RNA/QC/</code></pre>
<pre class="r"><code> get_upper_tri &lt;- function(cormat){
    cormat[lower.tri(cormat)]&lt;- NA
    return(cormat)
 }
 
  get_lower_tri &lt;- function(cormat){
    cormat[upper.tri(cormat)]&lt;- NA
    return(cormat)
  }</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/QC/COR_10kb_norm.rda&quot;)
mat &lt;- counts

cor.mat.ctrl &lt;- get_lower_tri(cor(mat[,grep(&quot;ctrl&quot;, colnames(mat))], method=&quot;pearson&quot;))
ctrl.m &lt;- reshape2::melt(cor.mat.ctrl, na.rm=TRUE)

c.ctrl.atac &lt;-
  ggplot(data = ctrl.m, aes(Var2, Var1, fill = value))+
 geom_tile(color = &quot;black&quot;, lwd=0.7)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;slateblue4&quot;, mid = &quot;skyblue2&quot;, 
   midpoint = 0.5, limit = c(0,1), space = &quot;Lab&quot;, 
   name=&quot;Pearson\nCorrelation&quot;) +
  geom_text(aes(label=round(value, 2)), size=3) +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),
    axis.title=element_blank(),
    panel.grid.major = element_blank(),
    legend.position=&quot;none&quot;) +
  coord_fixed() + 
  ggtitle(&quot;ATAC-seq genome-wide correlation&quot;)

cor.mat.cyt &lt;- get_upper_tri(cor(mat[,grep(&quot;cyt&quot;, colnames(mat))], method=&quot;pearson&quot;))
cyt.m &lt;- reshape2::melt(cor.mat.cyt, na.rm=TRUE)

c.cyt.atac &lt;-
  ggplot(data = cyt.m, aes(Var2, Var1, fill = value))+
 geom_tile(color = &quot;black&quot;, lwd=0.7)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;slateblue4&quot;, mid = &quot;skyblue2&quot;, 
   midpoint = 0.5, limit = c(0,1), space = &quot;Lab&quot;, 
   name=&quot;Pearson\nCorrelation&quot;) +
  geom_text(aes(label=round(value, 2)), size=3) +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),
    axis.title=element_blank(),
    legend.justification = c(1, 0),
    legend.position = c(0.6, 0.75),
    legend.direction = &quot;horizontal&quot;,
    panel.grid.major = element_blank()) +
  coord_fixed() +
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1,
                title.position = &quot;top&quot;, title.hjust = 0.5))


cor.rep.atac &lt;- plot_grid(c.ctrl.atac, c.cyt.atac)
cor.rep.atac</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/atac-genome-corr-1.png" alt="ATAC-seq correlation using the number of reads in a 10kb binned genome normalized with DESeq2." width="672" />
<p class="caption">
ATAC-seq correlation using the number of reads in a 10kb binned genome normalized with DESeq2.
</p>
</div>
<pre class="r"><code>load(&quot;../data/CYT/H3K27ac/QC/COR_10kb_norm.rda&quot;)
mat &lt;- counts

cor.mat.ctrl &lt;- get_lower_tri(cor(mat[,grep(&quot;ctrl&quot;, colnames(mat))], method=&quot;pearson&quot;))
ctrl.m &lt;- reshape2::melt(cor.mat.ctrl, na.rm=TRUE)

c.ctrl.H3K27ac &lt;-
  ggplot(data = ctrl.m, aes(Var2, Var1, fill = value))+
 geom_tile(color = &quot;black&quot;, lwd=0.7)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;slateblue4&quot;, mid = &quot;skyblue2&quot;, 
   midpoint = 0.5, limit = c(0,1), space = &quot;Lab&quot;, 
   name=&quot;Pearson\nCorrelation&quot;) +
  geom_text(aes(label=round(value, 2)), size=3) +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),
    axis.title=element_blank(),
    panel.grid.major = element_blank(),
    legend.position=&quot;none&quot;) +
  coord_fixed() + 
  ggtitle(&quot;H3k27ac genome-wide correlation&quot;)

cor.mat.cyt &lt;- get_upper_tri(cor(mat[,grep(&quot;cyt&quot;, colnames(mat))], method=&quot;pearson&quot;))
cyt.m &lt;- reshape2::melt(cor.mat.cyt, na.rm=TRUE)

c.cyt.H3K27ac &lt;-
  ggplot(data = cyt.m, aes(Var2, Var1, fill = value))+
 geom_tile(color = &quot;black&quot;, lwd=0.7)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;slateblue4&quot;, mid = &quot;skyblue2&quot;, 
   midpoint = 0.5, limit = c(0,1), space = &quot;Lab&quot;, 
   name=&quot;Pearson\nCorrelation&quot;) +
  geom_text(aes(label=round(value, 2)), size=3) +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),
    axis.title=element_blank(),
    legend.justification = c(1, 0),
    legend.position = c(0.6, 0.75),
    legend.direction = &quot;horizontal&quot;,
    panel.grid.major = element_blank()) +
  coord_fixed() +
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1,
                title.position = &quot;top&quot;, title.hjust = 0.5))


cor.rep.H3K27ac &lt;- plot_grid(c.ctrl.H3K27ac, c.cyt.H3K27ac)
cor.rep.H3K27ac</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/k27-genome-corr-1.png" alt="H3K27ac ChIP-seq correlation using the number of reads in a 10kb binned genome normalized with DESeq2." width="672" />
<p class="caption">
H3K27ac ChIP-seq correlation using the number of reads in a 10kb binned genome normalized with DESeq2.
</p>
</div>
<pre class="r"><code>load(&quot;../data/CYT/RNA/QC/COR_10kb_norm.rda&quot;)

cor.mat.ctrl &lt;- get_lower_tri(cor(mat[,grep(&quot;ctrl&quot;, colnames(mat))], method=&quot;pearson&quot;))
ctrl.m &lt;- reshape2::melt(cor.mat.ctrl, na.rm=TRUE)

c.ctrl.RNA &lt;-
  ggplot(data = ctrl.m, aes(Var2, Var1, fill = value))+
 geom_tile(color = &quot;black&quot;, lwd=0.7)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;slateblue4&quot;, mid = &quot;skyblue2&quot;, 
   midpoint = 0.5, limit = c(0,1), space = &quot;Lab&quot;, 
   name=&quot;Pearson\nCorrelation&quot;) +
  geom_text(aes(label=round(value, 2)), size=3) +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),
    axis.title=element_blank(),
    panel.grid.major = element_blank(),
    legend.position=&quot;none&quot;) +
  coord_fixed() + 
  ggtitle(&quot;RNA-seq genome-wide correlation&quot;)

cor.mat.cyt &lt;- get_upper_tri(cor(mat[,grep(&quot;cyt&quot;, colnames(mat))], method=&quot;pearson&quot;))
cyt.m &lt;- reshape2::melt(cor.mat.cyt, na.rm=TRUE)

c.cyt.RNA &lt;-
  ggplot(data = cyt.m, aes(Var2, Var1, fill = value))+
 geom_tile(color = &quot;black&quot;, lwd=0.7)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;slateblue4&quot;, mid = &quot;skyblue2&quot;, 
   midpoint = 0.5, limit = c(0,1), space = &quot;Lab&quot;, 
   name=&quot;Pearson\nCorrelation&quot;) +
  geom_text(aes(label=round(value, 2)), size=3) +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),
    axis.title=element_blank(),
    legend.justification = c(1, 0),
    legend.position = c(0.6, 0.75),
    legend.direction = &quot;horizontal&quot;,
    panel.grid.major = element_blank()) +
  coord_fixed() +
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1,
                title.position = &quot;top&quot;, title.hjust = 0.5))


cor.rep.RNA &lt;- plot_grid(c.ctrl.RNA, c.cyt.RNA)
cor.rep.RNA</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/rna-genome-corr-1.png" alt="RNA-seq correlation using the number of reads in a 10kb binned genome normalized with DESeq2." width="672" />
<p class="caption">
RNA-seq correlation using the number of reads in a 10kb binned genome normalized with DESeq2.
</p>
</div>
</div>
</div>
<div id="differential-analysis" class="section level2">
<h2>Differential analysis</h2>
<p>The differential analysis in the different assays is performed using <code>DESeq2</code>.</p>
<div id="atac-seq" class="section level3">
<h3>ATAC-seq</h3>
<pre><code>cd data/CYT/ATAC
Rscript ../../code/CYT_diffAnalysis_DESeq2_chrom.R -f 1 -q 0.05 -b TRUE -s hi -e ATAC
Rscript ../../code/CYT_diffAnalysis_DESeq2_chrom.R -f 1 -q 0.05 -b TRUE -s endoc -e ATAC</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/diffAnalysis/ATAC_endoc_fc1_padj0.05_GRangesBatch.rda&quot;)

table(res.gr$type, res.gr$annotation)</code></pre>
<pre><code>        
         Distal Promoter
  gained  12240      267
  lost      679        9
  stable 171672    13354</code></pre>
<pre class="r"><code>volc_ec &lt;- 
  ggplot(data.frame(res.gr),
       aes(log2FoldChange, -log10(padj))) +
  geom_point(aes(color=type), size=0.4) + 
  scale_color_manual(values=pals$differential,
                    name=&quot;RE type&quot;) +
  geom_vline(xintercept=c(1,-1), linetype=2, color=&quot;dark grey&quot;) +
  geom_hline(yintercept=-log10(0.05), linetype=2, color=&quot;dark grey&quot;) +
  xlab(expression(Log[2]*&quot; fold-change&quot;)) + ylab(expression(-Log[10]*&quot; FDR adjusted P&quot;)) +
  ggtitle(expression(&quot;ATAC-seq EndoC-&quot;*beta*H1)) +
  theme(legend.position=&quot;none&quot;)</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/diffAnalysis/ATAC_hi_fc1_padj0.05_GRangesBatch.rda&quot;)

table(res.gr$type, res.gr$annotation)</code></pre>
<pre><code>        
         Distal Promoter
  gained  14505      462
  lost     4544      115
  stable 295383    18090</code></pre>
<pre class="r"><code>volc_hi &lt;- 
  ggplot(data.frame(res.gr),
       aes(log2FoldChange, -log10(padj))) +
  geom_point(aes(color=type), size=0.4) + 
  scale_color_manual(values=pals$differential,
                    name=&quot;RE type&quot;) +
  geom_vline(xintercept=c(1,-1), linetype=2, color=&quot;dark grey&quot;) +
  geom_hline(yintercept=-log10(0.05), linetype=2, color=&quot;dark grey&quot;) +
  xlab(expression(Log[2]*&quot; fold-change&quot;)) + ylab(expression(-Log[10]*&quot; FDR adjusted P&quot;)) +
  ggtitle(expression(&quot;ATAC-seq HI&quot;)) +
  theme(legend.position=&quot;none&quot;)</code></pre>
<pre class="r"><code>plot_grid(volc_ec, 
          volc_hi,
          ncol=2)</code></pre>
<p><img src="figure/cyt_01_remodeling.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="h3k27ac" class="section level3">
<h3>H3K27ac</h3>
<pre><code>cd data/CYT/H3K27ac
Rscript ../../code/CYT_diffAnalysis_DESeq2_chrom.R -f 1 -q 0.05 -b TRUE -s hi -e H3K27ac
Rscript ../../code/CYT_diffAnalysis_DESeq2_chrom.R -f 1 -q 0.05 -b TRUE -s endoc -e H3K27ac</code></pre>
</div>
<div id="rna-seq" class="section level3">
<h3>RNA-seq</h3>
<pre><code>cd data/CYT/RNA
Rscript ../../code/CYT_diffAnalysis_DESeq2_rna.R -f 1 -q 0.05 -b TRUE -s hi
Rscript ../../code/CYT_diffAnalysis_DESeq2_rna.R -f 1 -q 0.05 -b TRUE -s endoc</code></pre>
</div>
</div>
<div id="correlation-between-experiments" class="section level2">
<h2>Correlation between experiments</h2>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_0.8.5     cowplot_1.0.0   ggplot2_3.3.0   workflowr_1.6.1

loaded via a namespace (and not attached):
 [1] Biobase_2.46.0              bit64_0.9-7                
 [3] splines_3.6.3               Formula_1.2-3              
 [5] assertthat_0.2.1            highr_0.8                  
 [7] stats4_3.6.3                latticeExtra_0.6-29        
 [9] blob_1.2.1                  GenomeInfoDbData_1.2.2     
[11] yaml_2.2.1                  RSQLite_2.2.0              
[13] pillar_1.4.3                backports_1.1.5            
[15] lattice_0.20-40             glue_1.3.2                 
[17] digest_0.6.25               GenomicRanges_1.38.0       
[19] RColorBrewer_1.1-2          promises_1.1.0             
[21] XVector_0.26.0              checkmate_2.0.0            
[23] colorspace_1.4-1            htmltools_0.4.0            
[25] httpuv_1.5.2                Matrix_1.2-18              
[27] plyr_1.8.6                  XML_3.99-0.3               
[29] DESeq2_1.26.0               pkgconfig_2.0.3            
[31] genefilter_1.68.0           zlibbioc_1.32.0            
[33] xtable_1.8-4                purrr_0.3.3                
[35] scales_1.1.0                jpeg_0.1-8.1               
[37] later_1.0.0                 BiocParallel_1.20.1        
[39] annotate_1.64.0             git2r_0.26.1               
[41] tibble_2.1.3                htmlTable_1.13.3           
[43] farver_2.0.3                IRanges_2.20.2             
[45] withr_2.1.2                 SummarizedExperiment_1.16.1
[47] nnet_7.3-13                 BiocGenerics_0.32.0        
[49] survival_3.1-11             magrittr_1.5               
[51] crayon_1.3.4                memoise_1.1.0              
[53] evaluate_0.14               fs_1.3.2                   
[55] foreign_0.8-76              tools_3.6.3                
[57] data.table_1.12.8           lifecycle_0.2.0            
[59] matrixStats_0.56.0          stringr_1.4.0              
[61] S4Vectors_0.24.3            locfit_1.5-9.4             
[63] munsell_0.5.0               cluster_2.1.0              
[65] DelayedArray_0.12.2         AnnotationDbi_1.48.0       
[67] compiler_3.6.3              GenomeInfoDb_1.22.0        
[69] rlang_0.4.5                 grid_3.6.3                 
[71] RCurl_1.98-1.1              rstudioapi_0.11            
[73] htmlwidgets_1.5.1           bitops_1.0-6               
[75] base64enc_0.1-3             labeling_0.3               
[77] rmarkdown_2.1               gtable_0.3.0               
[79] DBI_1.1.0                   reshape2_1.4.3             
[81] R6_2.4.1                    gridExtra_2.3              
[83] knitr_1.28                  bit_1.1-15.2               
[85] Hmisc_4.4-0                 rprojroot_1.3-2            
[87] stringi_1.4.6               parallel_3.6.3             
[89] Rcpp_1.0.4                  geneplotter_1.64.0         
[91] vctrs_0.2.4                 rpart_4.1-15               
[93] acepack_1.4.1               png_0.1-7                  
[95] tidyselect_1.0.0            xfun_0.12                  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
