<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mireia Ramos-Rodríguez" />


<title>#1 Chromatin remodeling induced by cytokines</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">β-cells and T1D</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-α
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_ifna.html">Description</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-γ + IL-1β
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_cyt.html">Description</a>
    </li>
    <li>
      <a href="cyt_01_remodeling.html">#1 Chromatin remodeling induced by cytokines</a>
    </li>
    <li>
      <a href="cyt_02_transcription.html">#2 Changes in transcription and translation</a>
    </li>
    <li>
      <a href="cyt_03_subclassification.html">#3 Subclassification of IREs</a>
    </li>
    <li>
      <a href="cyt_04_contacts.html">#4 Changes in 3D chormatin structure</a>
    </li>
    <li>
      <a href="cyt_05_t1d.html">#5 IREs and T1D risk</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IRB
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_irb.html">Description</a>
    </li>
    <li>
      <a href="https://mireia-bioinfo.github.io/plotRegulome">plotRegulome R package</a>
    </li>
    <li>
      <a href="https://github.com/mireia-bioinfo/isletregulome_shiny">IRB Shiny App code</a>
    </li>
    <li>
      <a href="http://www.isletregulome.org">Islet Regulome Browser</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mireia-bioinfo/phdthesis_code">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">#1 Chromatin remodeling induced by cytokines</h1>
<h3 class="subtitle">Proinflammatory cytokine exposure causes profound remodeling of the β-cell regulatory landscape</h3>
<h4 class="author">Mireia Ramos-Rodríguez</h4>

</div>


<div id="analysis-of-atac-seq-data" class="section level2">
<h2>Analysis of ATAC-seq data</h2>
<div id="quality-control" class="section level3">
<h3>Quality control</h3>
<p>Run Rscript for generating the necessary quality control measures.</p>
<pre><code>Rscript code/CYT_QC_ATAC.R</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/QC/ATAC_stats.rda&quot;)
load(&quot;../data/CYT/ATAC/QC/QC_scores.rda&quot;)

stats &lt;- dplyr::left_join(stats, txt)

lib &lt;-
  ggplot(stats,
       aes(&quot;Lib Size&quot;,
           sampleID)) +
  geom_tile(aes(fill=total_reads),
              color=&quot;black&quot;, size=1) +
  geom_text(aes(label=round(total_reads/1e6, 1))) +
  scale_fill_gradient(low=&quot;white&quot;,
                      high=&quot;purple3&quot;,
                      limits=c(0, max(stats$total_reads)),
                      breaks=c(0, max(stats$total_reads)),
                      labels=c(&quot;0&quot;, &quot;Max&quot;),
                      name=&quot;Library size (x10^6)&quot;) +
  guides(fill=guide_colourbar(ticks=FALSE,
                              frame.colour = &quot;black&quot;,
                              frame.linewidth = 1)) +
  theme(axis.line.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title = element_blank())

align &lt;-
  ggplot(stats,
         aes(&quot;% Align&quot;,
             sampleID)) + 
  geom_tile(aes(fill=alignment_rate),
            color=&quot;black&quot;, size=1) +
  geom_text(aes(label=round(alignment_rate, 1))) +
  scale_fill_gradient(low=&quot;white&quot;,
                      high=&quot;darkorange3&quot;,
                      limits=c(0, max(stats$alignment_rate)),
                      breaks=c(0, max(stats$alignment_rate)),
                      labels=c(0, &quot;Max&quot;),
                      name=&quot;% alignment&quot;) +
  guides(fill=guide_colourbar(ticks=FALSE,
                              frame.colour = &quot;black&quot;,
                              frame.linewidth = 1)) +
  theme(axis.text.y=element_blank(),
        axis.title=element_blank(),
        axis.line=element_blank(),
        axis.ticks=element_blank(),
        plot.margin=margin(0,0,0,0, unit=&#39;pt&#39;))
  
mit &lt;- 
  ggplot(stats,
       aes(&quot;% Mito&quot;,
           sampleID)) +
  geom_tile(aes(fill=chrM_rate),
            color=&quot;black&quot;, size=1) +
  geom_text(aes(label=round(chrM_rate, 1))) +
  scale_fill_gradient(low=&quot;dodgerblue3&quot;,
                      high=&quot;white&quot;,
                      limits=c(0, max(stats$chrM_rate)),
                      breaks=c(0, max(stats$chrM_rate)),
                      labels=c(&quot;0&quot;, &quot;Max&quot;),
                      name=&quot;% Mitochondrial&quot;) +
  guides(fill=guide_colourbar(ticks=FALSE,
                              frame.colour = &quot;black&quot;,
                              frame.linewidth = 1)) +
  theme(axis.text.y=element_blank(),
        axis.title=element_blank(),
        axis.line=element_blank(),
        axis.ticks=element_blank(),
        plot.margin=margin(0,0,0,0, unit=&#39;pt&#39;))


nsc &lt;- 
  ggplot(stats,
       aes(&quot;NSC&quot;,
           sampleID)) +
  geom_tile(aes(fill=NSC),
            color=&quot;black&quot;, size=1) +
  geom_text(aes(label=round(NSC, 1))) +
  scale_fill_gradient(low=&quot;white&quot;,
                      high=&quot;indianred4&quot;,
                      limits=c(1, max(stats$NSC)),
                      breaks=c(1, max(stats$NSC)),
                      labels=c(&quot;1&quot;, &quot;Max&quot;),
                      name=&quot;NSC&quot;) +
  guides(fill=guide_colourbar(ticks=FALSE,
                              frame.colour = &quot;black&quot;,
                              frame.linewidth = 1)) +
  theme(axis.text.y=element_blank(),
        axis.title=element_blank(),
        axis.line=element_blank(),
        axis.ticks=element_blank(),
        plot.margin=margin(0,0,0,0, unit=&#39;pt&#39;))

rsc &lt;- 
  ggplot(stats,
       aes(&quot;RSC&quot;,
           sampleID)) +
  geom_tile(aes(fill=RSC),
            color=&quot;black&quot;, size=1) +
  geom_text(aes(label=round(RSC, 1))) +
  scale_fill_gradient(low=&quot;white&quot;,
                      high=&quot;turquoise4&quot;,
                      limits=c(0, max(stats$RSC)),
                      breaks=c(0, max(stats$RSC)),
                      labels=c(&quot;0&quot;, &quot;Max&quot;),
                      name=&quot;RSC&quot;) +
  guides(fill=guide_colourbar(ticks=FALSE,
                              frame.colour = &quot;black&quot;,
                              frame.linewidth = 1)) +
  theme(axis.text.y=element_blank(),
        axis.title=element_blank(),
        axis.line=element_blank(),
        axis.ticks=element_blank(),
        plot.margin=margin(0,0,0,0, unit=&#39;pt&#39;))


## Get legend -----------------------
lib.leg &lt;- get_legend(lib)
al.leg &lt;- get_legend(align)
mit.leg &lt;- get_legend(mit)
nsc.leg &lt;- get_legend(nsc)
rsc.leg &lt;- get_legend(rsc)


leg &lt;- plot_grid(lib.leg, 
                 al.leg,
                 mit.leg, 
                 nsc.leg, 
                 rsc.leg,
                 nrow=1)

## Create heatmap --------------------------

heat &lt;- plot_grid(lib + theme(legend.position=&quot;none&quot;), 
                  align + theme(legend.position=&quot;none&quot;),
          mit + theme(legend.position=&quot;none&quot;), 
          nsc + theme(legend.position=&quot;none&quot;), 
          rsc + theme(legend.position=&quot;none&quot;),
          nrow=1,
          align=&#39;h&#39;,
          rel_widths=c(0.30, rep(0.1750, 4))
          )


## Final plot ----------------
qc &lt;- 
  plot_grid(heat, leg, nrow=2, rel_heights = c(0.7, 0.3))

qc</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/atac-heatmap-1.png" alt="Sumary of per-replicate sequencing metrics, showing total library sizes, percentage of aligned reads, percentage of mitochondrial aligned reads, normalized strand cross-correlation coefficient (NSC) and relative strand cross-correlation coefficient (RSC)." width="672" />
<p class="caption">
Sumary of per-replicate sequencing metrics, showing total library sizes, percentage of aligned reads, percentage of mitochondrial aligned reads, normalized strand cross-correlation coefficient (NSC) and relative strand cross-correlation coefficient (RSC).
</p>
</div>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/QC/ATAC_tss_enrichment.rda&quot;)
# tss &lt;- ungroup(tss)
# save(tss, file=&quot;../data/CYT/ATAC/QC/ATAC_tss_enrichment.rda&quot;)

tss$dataset &lt;- factor(tss$dataset, levels=c(&quot;TSS annotation&quot;, &quot;Random control&quot;))
tss$group &lt;- gsub(&quot;[[:digit:]]*_&quot;, &quot;_&quot;, tss$sample)

tss &lt;- tss %&gt;% 
  group_by(dataset, group, Position) %&gt;%
  summarise(mean=mean(mean))

tss.plot &lt;- 
  ggplot(tss,
       aes(Position, mean)) +
  geom_line(aes(group=dataset, color=dataset), lwd=0.7) +
  scale_color_manual(values=c(&quot;seagreen4&quot;, &quot;goldenrod3&quot;), name=&quot;Dataset&quot;) +
  facet_wrap(~group, scales=&quot;free_y&quot;) +
  theme(legend.position=&quot;top&quot;) +
  ylab(&quot;Mean ATAC-seq read counts&quot;) + xlab(&quot;Position relative to TSS (bp)&quot;)

tss.plot</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/atac-tss-enrichment-1.png" alt="Enrichment of ATAC-seq reads around protein-coding TSS compared to a randomized set of regions." width="576" />
<p class="caption">
Enrichment of ATAC-seq reads around protein-coding TSS compared to a randomized set of regions.
</p>
</div>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/QC/ATAC_noise.rda&quot;)
# stats &lt;- ungroup(stats)
# save(stats, file=&quot;../data/CYT/ATAC/QC/ATAC_noise.rda&quot;)

stats$mean.errorbar &lt;- stats$mean
stats &lt;- stats[order(rev(stats$Annotation)),]

stats &lt;- stats[stats$Annotation!=&quot;Unassigned&quot;,] %&gt;%
  group_by(group) %&gt;%
  mutate(cumsum=cumsum(mean))

noise &lt;- 
  ggplot(stats[stats$Annotation!=&quot;Unassigned&quot;,],
       aes(group, mean)) +
  geom_bar(aes(fill=Annotation), stat=&quot;identity&quot;,
           position=&quot;stack&quot;,
           color=&quot;black&quot;, lwd=0.7) +
  geom_errorbar(aes(ymin=cumsum, ymax=cumsum+sd,
                    group=Annotation),
                width=.3, lwd=0.5) +
  scale_fill_manual(values=c(&quot;violetred&quot;, &quot;dark orange&quot;)) +
  scale_y_continuous(name=&quot;Percentage of reads in peaks (%)&quot;) +
  theme(axis.text.x=element_text(angle=30, hjust=1),
        legend.position=&quot;top&quot;,
        axis.title.x=element_blank())

noise</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/atac-signal-to-noise-1.png" alt="Signal-to-noise ratios of ATAC-seq reads located at called peaks vs reads outside peaks." width="384" />
<p class="caption">
Signal-to-noise ratios of ATAC-seq reads located at called peaks vs reads outside peaks.
</p>
</div>
<pre><code>Rscript code/QC_CORR_genome.R data/CYT/ATAC/BAM/ data/CYT/ATAC/QC/</code></pre>
<pre class="r"><code> get_upper_tri &lt;- function(cormat){
    cormat[lower.tri(cormat)]&lt;- NA
    return(cormat)
 }
 
  get_lower_tri &lt;- function(cormat){
    cormat[upper.tri(cormat)]&lt;- NA
    return(cormat)
  }</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/QC/COR_10kb_norm.rda&quot;)
mat &lt;- counts

cor.mat.ctrl &lt;- get_lower_tri(cor(mat[,grep(&quot;ctrl&quot;, colnames(mat))], method=&quot;pearson&quot;))
ctrl.m &lt;- reshape2::melt(cor.mat.ctrl, na.rm=TRUE)

c.ctrl.atac &lt;-
  ggplot(data = ctrl.m, aes(Var2, Var1, fill = value))+
 geom_tile(color = &quot;black&quot;, lwd=0.7)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;slateblue4&quot;, mid = &quot;skyblue2&quot;, 
   midpoint = 0.5, limit = c(0,1), space = &quot;Lab&quot;, 
   name=&quot;Pearson\nCorrelation&quot;) +
  geom_text(aes(label=round(value, 2)), size=3) +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),
    axis.title=element_blank(),
    panel.grid.major = element_blank(),
    legend.position=&quot;none&quot;) +
  coord_fixed() + 
  ggtitle(&quot;ATAC-seq genome-wide correlation&quot;)

cor.mat.cyt &lt;- get_upper_tri(cor(mat[,grep(&quot;cyt&quot;, colnames(mat))], method=&quot;pearson&quot;))
cyt.m &lt;- reshape2::melt(cor.mat.cyt, na.rm=TRUE)

c.cyt.atac &lt;-
  ggplot(data = cyt.m, aes(Var2, Var1, fill = value))+
 geom_tile(color = &quot;black&quot;, lwd=0.7)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;slateblue4&quot;, mid = &quot;skyblue2&quot;, 
   midpoint = 0.5, limit = c(0,1), space = &quot;Lab&quot;, 
   name=&quot;Pearson\nCorrelation&quot;) +
  geom_text(aes(label=round(value, 2)), size=3) +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),
    axis.title=element_blank(),
    legend.justification = c(1, 0),
    legend.position = c(0.6, 0.75),
    legend.direction = &quot;horizontal&quot;,
    panel.grid.major = element_blank()) +
  coord_fixed() +
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1,
                title.position = &quot;top&quot;, title.hjust = 0.5))


cor.rep.atac &lt;- plot_grid(c.ctrl.atac, c.cyt.atac)
cor.rep.atac</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/atac-genome-corr-1.png" alt="ATAC-seq correlation using the number of reads in a 10kb binned genome normalized with DESeq2." width="672" />
<p class="caption">
ATAC-seq correlation using the number of reads in a 10kb binned genome normalized with DESeq2.
</p>
</div>
</div>
<div id="differential-analysis" class="section level3">
<h3>Differential analysis</h3>
<pre><code>cd data/CYT/ATAC
Rscript ../../code/CYT_diffAnalysis_DESeq2_chrom.R -f 1 -q 0.05 -b TRUE -s hi -e ATAC
Rscript ../../code/CYT_diffAnalysis_DESeq2_chrom.R -f 1 -q 0.05 -b TRUE -s endoc -e ATAC</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/diffAnalysis/ATAC_endoc_fc1_padj0.05_GRangesBatch.rda&quot;)

table(res.gr$type, res.gr$annotation) %&gt;% 
  knitr::kable(format=&quot;html&quot;,
               format.args = list(big.mark = &quot;,&quot;),
               caption = &quot;Number of regions classified according to significance and distance to TSS in ATAC-seq EndoC samples.&quot;) %&gt;% 
  kable_styling(full_width = FALSE) %&gt;% 
  add_header_above(c(&quot;Region type&quot; = 1, &quot;Location respect TSS&quot; = 2))</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Number of regions classified according to significance and distance to TSS in ATAC-seq EndoC samples.
</caption>
<thead>
<tr>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Region type
</div>
</th>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="2">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Location respect TSS
</div>
</th>
</tr>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Distal
</th>
<th style="text-align:right;">
Promoter
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
gained
</td>
<td style="text-align:right;">
12,240
</td>
<td style="text-align:right;">
267
</td>
</tr>
<tr>
<td style="text-align:left;">
lost
</td>
<td style="text-align:right;">
679
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
stable
</td>
<td style="text-align:right;">
171,672
</td>
<td style="text-align:right;">
13,354
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>volc_ec &lt;- 
  ggplot(data.frame(res.gr),
       aes(log2FoldChange, -log10(padj))) +
  geom_point(aes(color=type), size=0.4) + 
  scale_color_manual(values=pals$differential,
                    name=&quot;RE type&quot;) +
  geom_vline(xintercept=c(1,-1), linetype=2, color=&quot;dark grey&quot;) +
  geom_hline(yintercept=-log10(0.05), linetype=2, color=&quot;dark grey&quot;) +
  xlab(expression(Log[2]*&quot; fold-change&quot;)) + ylab(expression(-Log[10]*&quot; FDR adjusted P&quot;)) +
  ggtitle(expression(&quot;ATAC-seq EndoC-&quot;*beta*H1)) +
  theme(legend.position=&quot;none&quot;)</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/diffAnalysis/ATAC_hi_fc1_padj0.05_GRangesBatch.rda&quot;)

table(res.gr$type, res.gr$annotation) %&gt;% 
  knitr::kable(format=&quot;html&quot;,
               format.args = list(big.mark = &quot;,&quot;),
               caption = &quot;Number of regions classified according to significance and distance to TSS in ATAC-seq HI samples.&quot;) %&gt;% 
  kable_styling(full_width = FALSE) %&gt;% 
  add_header_above(c(&quot;Region type&quot; = 1, &quot;Location respect TSS&quot; = 2))</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Number of regions classified according to significance and distance to TSS in ATAC-seq HI samples.
</caption>
<thead>
<tr>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Region type
</div>
</th>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="2">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Location respect TSS
</div>
</th>
</tr>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Distal
</th>
<th style="text-align:right;">
Promoter
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
gained
</td>
<td style="text-align:right;">
14,505
</td>
<td style="text-align:right;">
462
</td>
</tr>
<tr>
<td style="text-align:left;">
lost
</td>
<td style="text-align:right;">
4,544
</td>
<td style="text-align:right;">
115
</td>
</tr>
<tr>
<td style="text-align:left;">
stable
</td>
<td style="text-align:right;">
295,383
</td>
<td style="text-align:right;">
18,090
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>volc_hi &lt;- 
  ggplot(data.frame(res.gr),
       aes(log2FoldChange, -log10(padj))) +
  geom_point(aes(color=type), size=0.4) + 
  scale_color_manual(values=pals$differential,
                    name=&quot;RE type&quot;) +
  geom_vline(xintercept=c(1,-1), linetype=2, color=&quot;dark grey&quot;) +
  geom_hline(yintercept=-log10(0.05), linetype=2, color=&quot;dark grey&quot;) +
  xlab(expression(Log[2]*&quot; fold-change&quot;)) + ylab(expression(-Log[10]*&quot; FDR adjusted P&quot;)) +
  ggtitle(expression(&quot;ATAC-seq HI&quot;)) +
  theme(legend.position=&quot;none&quot;)</code></pre>
<pre class="r"><code>plot_grid(volc_ec, 
          volc_hi,
          ncol=2)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/unnamed-chunk-1-1.png" alt="Volcano plots showing the open chromatin sites in EndoC and human islet (HI) samples. The horizontal line denotes the FDR adjusted P-value threshold set at 0.05 and the vertical lines the log2 fold-change thresholds, at -1 and 1. Gained regions are represented in green and lost regions are shown in red." width="672" />
<p class="caption">
Volcano plots showing the open chromatin sites in EndoC and human islet (HI) samples. The horizontal line denotes the FDR adjusted P-value threshold set at 0.05 and the vertical lines the log2 fold-change thresholds, at -1 and 1. Gained regions are represented in green and lost regions are shown in red.
</p>
</div>
<pre class="r"><code>## Load regions
load(&quot;../data/CYT/ATAC/diffAnalysis/ATAC_endoc_fc1_padj0.05_GRangesBatch.rda&quot;)
ec &lt;- res.gr

load(&quot;../data/CYT/ATAC/diffAnalysis/ATAC_hi_fc1_padj0.05_GRangesBatch.rda&quot;)
hi &lt;- res.gr

## Find overlaps
ols &lt;- findOverlaps(hi, ec)

## Get fold-change in HI vs type in EndoC
df &lt;- cbind(data.frame(hi)[queryHits(ols),c(6,8)],
            data.frame(ec)[subjectHits(ols),c(13)])
colnames(df)[3] &lt;- &quot;type_ec&quot;

## Plot results
ggplot(df,
       aes(type_ec, log2FoldChange)) + 
  geom_hline(yintercept=c(1,-1), lty=2, color=&quot;grey&quot;) +
  geom_hline(yintercept=0, lty=1, color=&quot;grey&quot;) +
  geom_boxplot(aes(color=type_ec), notch=F, outlier.shape=NA,
               lwd=1) +
  scale_color_manual(values=pals$differential,
                    name=&quot;RE type&quot;) +
  scale_y_continuous(name=expression(&quot;HI &quot;*log[2]*&quot; FC&quot;)) +
  theme(legend.position=&quot;none&quot;,
        strip.background = element_rect(fill=&quot;white&quot;, linetype=1, size=.5, color=&quot;black&quot;)) + 
  scale_x_discrete(name=expression(&quot;EndoC-&quot;*beta*&quot;H1 region type&quot;),
                   labels=function(x) Hmisc::capitalize(x)) +
  coord_cartesian(ylim=c(-2,3))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/atac-comp-hi-ec-1.png" alt="Boxplot of HI log2FC at ATAC-seq regions classified as gained, lost or stable in EndoC cells. Horizontal dashed lines show the upper and lower log2 FC thresholds." width="384" />
<p class="caption">
Boxplot of HI log2FC at ATAC-seq regions classified as gained, lost or stable in EndoC cells. Horizontal dashed lines show the upper and lower log2 FC thresholds.
</p>
</div>
</div>
</div>
<div id="analysis-of-h3k27ac-chip-seq-data" class="section level2">
<h2>Analysis of H3K27ac ChIP-seq data</h2>
<div id="quality-control-1" class="section level3">
<h3>Quality control</h3>
<pre><code>Rscript code/QC_CORR_genome.R data/CYT/H3K27ac/BAM/ data/CYT/H3K27ac/QC/</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/H3K27ac/QC/COR_10kb_norm.rda&quot;)
mat &lt;- counts

cor.mat.ctrl &lt;- get_lower_tri(cor(mat[,grep(&quot;ctrl&quot;, colnames(mat))], method=&quot;pearson&quot;))
ctrl.m &lt;- reshape2::melt(cor.mat.ctrl, na.rm=TRUE)

c.ctrl.H3K27ac &lt;-
  ggplot(data = ctrl.m, aes(Var2, Var1, fill = value))+
 geom_tile(color = &quot;black&quot;, lwd=0.7)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;slateblue4&quot;, mid = &quot;skyblue2&quot;, 
   midpoint = 0.5, limit = c(0,1), space = &quot;Lab&quot;, 
   name=&quot;Pearson\nCorrelation&quot;) +
  geom_text(aes(label=round(value, 2)), size=3) +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),
    axis.title=element_blank(),
    panel.grid.major = element_blank(),
    legend.position=&quot;none&quot;) +
  coord_fixed() + 
  ggtitle(&quot;H3k27ac genome-wide correlation&quot;)

cor.mat.cyt &lt;- get_upper_tri(cor(mat[,grep(&quot;cyt&quot;, colnames(mat))], method=&quot;pearson&quot;))
cyt.m &lt;- reshape2::melt(cor.mat.cyt, na.rm=TRUE)

c.cyt.H3K27ac &lt;-
  ggplot(data = cyt.m, aes(Var2, Var1, fill = value))+
 geom_tile(color = &quot;black&quot;, lwd=0.7)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;slateblue4&quot;, mid = &quot;skyblue2&quot;, 
   midpoint = 0.5, limit = c(0,1), space = &quot;Lab&quot;, 
   name=&quot;Pearson\nCorrelation&quot;) +
  geom_text(aes(label=round(value, 2)), size=3) +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),
    axis.title=element_blank(),
    legend.justification = c(1, 0),
    legend.position = c(0.6, 0.75),
    legend.direction = &quot;horizontal&quot;,
    panel.grid.major = element_blank()) +
  coord_fixed() +
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1,
                title.position = &quot;top&quot;, title.hjust = 0.5))


cor.rep.H3K27ac &lt;- plot_grid(c.ctrl.H3K27ac, c.cyt.H3K27ac)
cor.rep.H3K27ac</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/k27-genome-corr-1.png" alt="H3K27ac ChIP-seq correlation using the number of reads in a 10kb binned genome normalized with DESeq2." width="672" />
<p class="caption">
H3K27ac ChIP-seq correlation using the number of reads in a 10kb binned genome normalized with DESeq2.
</p>
</div>
</div>
<div id="differential-analysis-1" class="section level3">
<h3>Differential analysis</h3>
<pre><code>cd data/CYT/H3K27ac
Rscript ../../code/CYT_diffAnalysis_DESeq2_chrom.R -f 1 -q 0.05 -b TRUE -s hi -e H3K27ac
Rscript ../../code/CYT_diffAnalysis_DESeq2_chrom.R -f 1 -q 0.05 -b TRUE -s endoc -e H3K27ac</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/H3K27ac/diffAnalysis/H3K27ac_endoc_fc1_padj0.05_GRangesBatch.rda&quot;)

table(res.gr$type, res.gr$annotation) %&gt;% 
  knitr::kable(format=&quot;html&quot;,
               format.args = list(big.mark = &quot;,&quot;),
               caption = &quot;Number of regions classified according to significance and distance to TSS in H3K27ac EndoC samples.&quot;) %&gt;% 
  kable_styling(full_width = FALSE) %&gt;% 
  add_header_above(c(&quot;Region type&quot; = 1, &quot;Location respect TSS&quot; = 2))</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Number of regions classified according to significance and distance to TSS in H3K27ac EndoC samples.
</caption>
<thead>
<tr>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Region type
</div>
</th>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="2">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Location respect TSS
</div>
</th>
</tr>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Distal
</th>
<th style="text-align:right;">
Promoter
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
gained
</td>
<td style="text-align:right;">
3,180
</td>
<td style="text-align:right;">
201
</td>
</tr>
<tr>
<td style="text-align:left;">
lost
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
stable
</td>
<td style="text-align:right;">
134,512
</td>
<td style="text-align:right;">
11,930
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>volc_ec &lt;- 
  ggplot(data.frame(res.gr),
       aes(log2FoldChange, -log10(padj))) +
  geom_point(aes(color=type), size=0.4) + 
  scale_color_manual(values=pals$differential,
                    name=&quot;RE type&quot;) +
  geom_vline(xintercept=c(1,-1), linetype=2, color=&quot;dark grey&quot;) +
  geom_hline(yintercept=-log10(0.05), linetype=2, color=&quot;dark grey&quot;) +
  xlab(expression(Log[2]*&quot; fold-change&quot;)) + ylab(expression(-Log[10]*&quot; FDR adjusted P&quot;)) +
  ggtitle(expression(&quot;H3K27ac EndoC-&quot;*beta*H1)) +
  theme(legend.position=&quot;none&quot;)</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/H3K27ac/diffAnalysis/H3K27ac_hi_fc1_padj0.05_GRangesBatch.rda&quot;)

table(res.gr$type, res.gr$annotation) %&gt;% 
  knitr::kable(format=&quot;html&quot;,
               format.args = list(big.mark = &quot;,&quot;),
               caption = &quot;Number of regions classified according to significance and distance to TSS in H3K27ac HI samples.&quot;) %&gt;% 
  kable_styling(full_width = FALSE) %&gt;% 
  add_header_above(c(&quot;Region type&quot; = 1, &quot;Location respect TSS&quot; = 2))</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Number of regions classified according to significance and distance to TSS in H3K27ac HI samples.
</caption>
<thead>
<tr>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Region type
</div>
</th>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="2">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Location respect TSS
</div>
</th>
</tr>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Distal
</th>
<th style="text-align:right;">
Promoter
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
gained
</td>
<td style="text-align:right;">
4,022
</td>
<td style="text-align:right;">
219
</td>
</tr>
<tr>
<td style="text-align:left;">
lost
</td>
<td style="text-align:right;">
5,895
</td>
<td style="text-align:right;">
250
</td>
</tr>
<tr>
<td style="text-align:left;">
stable
</td>
<td style="text-align:right;">
110,543
</td>
<td style="text-align:right;">
11,006
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>volc_hi &lt;- 
  ggplot(data.frame(res.gr),
       aes(log2FoldChange, -log10(padj))) +
  geom_point(aes(color=type), size=0.4) + 
  scale_color_manual(values=pals$differential,
                    name=&quot;RE type&quot;) +
  geom_vline(xintercept=c(1,-1), linetype=2, color=&quot;dark grey&quot;) +
  geom_hline(yintercept=-log10(0.05), linetype=2, color=&quot;dark grey&quot;) +
  xlab(expression(Log[2]*&quot; fold-change&quot;)) + ylab(expression(-Log[10]*&quot; FDR adjusted P&quot;)) +
  ggtitle(expression(&quot;H3K27ac HI&quot;)) +
  theme(legend.position=&quot;none&quot;)</code></pre>
<pre class="r"><code>plot_grid(volc_ec, 
          volc_hi,
          ncol=2)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/unnamed-chunk-2-1.png" alt="Volcano plots showing the enriched H3K27ac sites in EndoC and human islet (HI) samples. The horizontal line denotes the FDR adjusted P-value threshold set at 0.05 and the vertical lines the log2 fold-change thresholds, at -1 and 1. Gained regions are represented in green and lost regions are shown in red." width="672" />
<p class="caption">
Volcano plots showing the enriched H3K27ac sites in EndoC and human islet (HI) samples. The horizontal line denotes the FDR adjusted P-value threshold set at 0.05 and the vertical lines the log2 fold-change thresholds, at -1 and 1. Gained regions are represented in green and lost regions are shown in red.
</p>
</div>
<pre class="r"><code>## Load regions
load(&quot;../data/CYT/H3K27ac/diffAnalysis/H3K27ac_endoc_fc1_padj0.05_GRangesBatch.rda&quot;)
ec &lt;- res.gr

load(&quot;../data/CYT/H3K27ac/diffAnalysis/H3K27ac_hi_fc1_padj0.05_GRangesBatch.rda&quot;)
hi &lt;- res.gr

## Find overlaps
ols &lt;- findOverlaps(hi, ec)

## Get fold-change in HI vs type in EndoC
df &lt;- cbind(data.frame(hi)[queryHits(ols),c(6,8)],
            data.frame(ec)[subjectHits(ols),c(13)])
colnames(df)[3] &lt;- &quot;type_ec&quot;

## Plot results
ggplot(df,
       aes(type_ec, log2FoldChange)) + 
  geom_hline(yintercept=c(1,-1), lty=2, color=&quot;grey&quot;) +
  geom_hline(yintercept=0, lty=1, color=&quot;grey&quot;) +
  geom_boxplot(aes(color=type_ec), notch=F, outlier.shape=NA,
               lwd=1) +
  scale_color_manual(values=pals$differential,
                    name=&quot;RE type&quot;) +
  scale_y_continuous(name=expression(&quot;HI &quot;*log[2]*&quot; FC&quot;)) +
  scale_x_discrete(name=expression(&quot;EndoC-&quot;*beta*&quot;H1 region type&quot;),
                   labels=function(x) Hmisc::capitalize(x)) +
  theme(legend.position=&quot;none&quot;,
        strip.background = element_rect(fill=&quot;white&quot;, linetype=1, size=.5, color=&quot;black&quot;)) + 
  coord_cartesian(ylim=c(-2,3))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/cyt_01_remodeling.Rmd/k37-comp-hi-ec-1.png" alt="Boxplot of HI log2FC at H3K27ac regions classified as gained, lost or stable in EndoC cells. Horizontal dashed lines show the upper and lower log2 FC thresholds." width="384" />
<p class="caption">
Boxplot of HI log2FC at H3K27ac regions classified as gained, lost or stable in EndoC cells. Horizontal dashed lines show the upper and lower log2 FC thresholds.
</p>
</div>
</div>
</div>
<div id="defining-induced-regulatory-elements" class="section level2">
<h2>Defining Induced Regulatory Elements</h2>
<div id="in-endoc-betah1" class="section level3">
<h3>In EndoC-<span class="math inline">\(\beta\)</span>H1</h3>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/diffAnalysis/ATAC_endoc_fc1_padj0.05_GRangesBatch.rda&quot;)
at &lt;- res.gr

load(&quot;../data/CYT/H3K27ac/diffAnalysis/H3K27ac_endoc_fc1_padj0.05_GRangesBatch.rda&quot;)
k27 &lt;- res.gr

ols &lt;- findOverlaps(at, k27, maxgap=200)

re &lt;- at[queryHits(ols),]
colnames(mcols(re)) &lt;- paste0(&quot;atac.&quot;, colnames(mcols(re)))
mcols(k27) &lt;- mcols(k27)[,c(1,3,7,8)]
colnames(mcols(k27)) &lt;- paste0(&quot;h3k27ac.&quot;, colnames(mcols(k27)))
mcols(re) &lt;- cbind(mcols(re),
                   mcols(k27)[subjectHits(ols),])

## Create new types of IREs
re$type &lt;- NA
re$type[re$h3k27ac.type==&quot;stable&quot; &amp; re$atac.type==&quot;stable&quot;] &lt;- &quot;SRE&quot;
re$type[re$h3k27ac.type==&quot;gained&quot; &amp; (re$atac.type %in% c(&quot;gained&quot;, &quot;stable&quot;))] &lt;- &quot;IRE&quot;

dir.create(&quot;../data/CYT/REs&quot;, F)
save(re, file=&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges.rda&quot;)

## Create less stringent IREs
re$type &lt;- NA
re$h3k27ac.type[re$h3k27ac.log2FoldChange &gt; 0.8 &amp; re$h3k27ac.padj &lt;= 0.05] &lt;- &quot;gained&quot;
re$type[re$h3k27ac.type==&quot;stable&quot; &amp; re$atac.type==&quot;stable&quot;] &lt;- &quot;SRE&quot;
re$type[re$h3k27ac.type==&quot;gained&quot; &amp; (re$atac.type %in% c(&quot;gained&quot;, &quot;stable&quot;))] &lt;- &quot;IRE&quot;
save(re, file=&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_k27.8.rda&quot;)</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges.rda&quot;)

cor &lt;- cor.test(re$atac.log2FoldChange, re$h3k27ac.log2FoldChange)

re.df &lt;- data.frame(re)[,-c(1:5)]

re.df$type &lt;- factor(re.df$type, 
                     levels=c(&quot;SRE&quot;, &quot;IRE&quot;))

re.df &lt;- re.df[order(re.df$type),]

ggplot(re.df,
       aes(atac.log2FoldChange, h3k27ac.log2FoldChange)) +
  geom_point(aes(color=h3k27ac.type,
                 fill=atac.type), pch=21) +
  scale_fill_manual(values=pals$differential) +
  scale_color_manual(values=pals$differential) +
  geom_vline(xintercept=c(-1,1), lty=2, color=&quot;grey&quot;) +
  geom_hline(yintercept=c(-1,1), lty=2, color=&quot;grey&quot;) +
  annotate(&quot;text&quot;, x=4, y=-2, label=paste0(&quot;r2=&quot;, round(cor$estimate, 2),
                                      &quot;\nP&lt;2x10-16&quot;)) +
  theme(legend.position=&quot;none&quot;) +
  scale_y_continuous(limits=c(-3,6),
                     name=expression(&quot;H3K27ac &quot;*log[2]*&quot; FC&quot;)) +
  scale_x_continuous(limits=c(-3,6),
                     name=expression(&quot;ATAC-seq &quot;*log[2]*&quot; FC&quot;))</code></pre>
<p><img src="figure/cyt_01_remodeling.Rmd/plot-cor-res-ec-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
<div id="in-human-islets" class="section level3">
<h3>In human islets</h3>
<pre class="r"><code>load(&quot;../data/CYT/ATAC/diffAnalysis/ATAC_hi_fc1_padj0.05_GRangesBatch.rda&quot;)
at &lt;- res.gr

load(&quot;../data/CYT/H3K27ac/diffAnalysis/H3K27ac_hi_fc1_padj0.05_GRangesBatch.rda&quot;)
k27 &lt;- res.gr

ols &lt;- findOverlaps(at, k27, maxgap=200)

re &lt;- at[queryHits(ols),]
colnames(mcols(re)) &lt;- paste0(&quot;atac.&quot;, colnames(mcols(re)))
mcols(k27) &lt;- mcols(k27)[,c(1,3,7,8)]
colnames(mcols(k27)) &lt;- paste0(&quot;h3k27ac.&quot;, colnames(mcols(k27)))
mcols(re) &lt;- cbind(mcols(re),
                   mcols(k27)[subjectHits(ols),])

## Create new types of IREs
re$type &lt;- NA
re$type[re$h3k27ac.type==&quot;stable&quot; &amp; re$atac.type==&quot;stable&quot;] &lt;- &quot;SRE&quot;
re$type[re$h3k27ac.type==&quot;gained&quot; &amp; (re$atac.type %in% c(&quot;gained&quot;, &quot;stable&quot;))] &lt;- &quot;IRE&quot;

dir.create(&quot;../data/CYT/REs&quot;, F)
save(re, file=&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges.rda&quot;)

## Create less stringent IREs
re$type &lt;- NA
re$h3k27ac.type[re$h3k27ac.log2FoldChange &gt; 0.8 &amp; re$h3k27ac.padj &lt;= 0.05] &lt;- &quot;gained&quot;
re$type[re$h3k27ac.type==&quot;stable&quot; &amp; re$atac.type==&quot;stable&quot;] &lt;- &quot;SRE&quot;
re$type[re$h3k27ac.type==&quot;gained&quot; &amp; (re$atac.type %in% c(&quot;gained&quot;, &quot;stable&quot;))] &lt;- &quot;IRE&quot;
save(re, file=&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_k27.8.rda&quot;)</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges.rda&quot;)

cor &lt;- cor.test(re$atac.log2FoldChange, re$h3k27ac.log2FoldChange)

re.df &lt;- data.frame(re)[,-c(1:5)]

re.df$type &lt;- factor(re.df$type, 
                     levels=c(&quot;SRE&quot;, &quot;IRE&quot;))

re.df &lt;- re.df[order(re.df$type),]

ggplot(re.df,
       aes(atac.log2FoldChange, h3k27ac.log2FoldChange)) +
  geom_point(aes(color=h3k27ac.type,
                 fill=atac.type), pch=21) +
  scale_fill_manual(values=pals$differential) +
  scale_color_manual(values=pals$differential) +
  geom_vline(xintercept=c(-1,1), lty=2, color=&quot;grey&quot;) +
  geom_hline(yintercept=c(-1,1), lty=2, color=&quot;grey&quot;) +
  annotate(&quot;text&quot;, x=4, y=-2, label=paste0(&quot;r2=&quot;, round(cor$estimate, 2),
                                      &quot;\nP&lt;2x10-16&quot;)) +
  theme(legend.position=&quot;none&quot;) +
  scale_y_continuous(limits=c(-3,6),
                     name=expression(&quot;H3K27ac &quot;*log[2]*&quot; FC&quot;)) +
  scale_x_continuous(limits=c(-3,6),
                     name=expression(&quot;ATAC-seq &quot;*log[2]*&quot; FC&quot;))</code></pre>
<p><img src="figure/cyt_01_remodeling.Rmd/plot-cor-res-hi-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
<div id="clustering" class="section level3">
<h3>Clustering</h3>
<p>Interestingly, the set of IREs identified in EndoC-<span class="math inline">\(\beta\)</span>H1 cells allows us to cluster both EndoC-<span class="math inline">\(\beta\)</span>H1 and Human Islet ATAC-seq and H3K27ac regions according to the suffered treatment and not sample type.</p>
<pre class="r"><code>################################################
## Clustering samples according to EndoC IREs ##
################################################
library(DESeq2)
library(BiocParallel)
register(MulticoreParam(4))

## ATAC-seq ------------------------------------
# Create SAF
load(&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges.rda&quot;)
saf &lt;- data.frame(re)[grep(&quot;IRE&quot;, re$type),c(1:3,6)]
colnames(saf) &lt;- c(&quot;Chr&quot;, &quot;Start&quot;, &quot;End&quot;, &quot;GeneID&quot;)
saf$Strand &lt;- &quot;+&quot;

# Get counts at IREs
files &lt;- list.files(&quot;../data/CYT/ATAC/BAM&quot;,
                    pattern=&quot;.offset.bam$&quot;,
                    full.names=TRUE)

atac &lt;- Rsubread::featureCounts(files,
                                annot.ext=saf,
                                nthreads=10)

names &lt;- pipelineNGS::getNameFromPath(files, suffix=&quot;.offset.bam&quot;)
colnames(atac$counts) &lt;- names

# Normalize with DESeq2
coldata &lt;- data.frame(sample=names,
                      treatment=unlist(lapply(strsplit(names, &quot;_&quot;),
                                              function(x) x[2])),
                      batch=unlist(lapply(strsplit(names, &quot;_&quot;),
                                          function(x) x[1])))
coldata$tissue &lt;- gsub(&quot;[[:digit:]]*&quot;, &quot;&quot;, coldata$batch)

dds &lt;- DESeqDataSetFromMatrix(countData=atac$counts,
                              colData=coldata,
                              design=~batch+treatment)

dds &lt;- DESeq(dds,
             parallel=TRUE)

rld &lt;- rlog(dds)
vsd &lt;- vst(dds)
save(rld,
     file=file.path(out_dir, &quot;IREs_ATAC_clustering_rld.rda&quot;))

## H3K27ac -------------------------------------------
# Create SAF
load(&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges.rda&quot;)
id &lt;- unique(re$h3k27ac.GeneID[grep(&quot;IRE&quot;, re$type)])

load(&quot;../data/CYT/H3K27ac/diffAnalysis/H3K27ac_endoc_fc1_padj0.05_GRangesBatch.rda&quot;)
res.gr &lt;- res.gr[res.gr$GeneID %in% id,]
saf &lt;- data.frame(res.gr)[,c(1:3,6)]
colnames(saf) &lt;- c(&quot;Chr&quot;, &quot;Start&quot;, &quot;End&quot;, &quot;GeneID&quot;)
saf$Strand &lt;- &quot;+&quot;

# Get counts at IREs
files &lt;- list.files(&quot;../../data/CYT/H3K27ac/BAM&quot;,
                    pattern=&quot;.bam$&quot;,
                    full.names=TRUE)
files &lt;- files[!grepl(&quot;raw&quot;, files)]

atac &lt;- Rsubread::featureCounts(files,
                                annot.ext=saf,
                                nthreads=10)

names &lt;- pipelineNGS::getNameFromPath(files, suffix=&quot;.bam&quot;)
colnames(atac$counts) &lt;- names

# Normalize with DESeq2
coldata &lt;- data.frame(sample=names,
                      treatment=unlist(lapply(strsplit(names, &quot;_&quot;),
                                              function(x) x[2])),
                      batch=unlist(lapply(strsplit(names, &quot;_&quot;),
                                          function(x) x[1])))
coldata$tissue &lt;- gsub(&quot;[[:digit:]]*&quot;, &quot;&quot;, coldata$batch)

dds &lt;- DESeqDataSetFromMatrix(countData=atac$counts,
                              colData=coldata,
                              design=~batch+treatment)

dds &lt;- DESeq(dds,
             parallel=TRUE)

rld &lt;- rlog(dds)

save(rld,
     file=file.path(out_dir, &quot;IREs_H3K27ac_clustering_rld.rda&quot;))</code></pre>
<pre class="r"><code>load(file.path(out_dir, &quot;IREs_ATAC_clustering_rld.rda&quot;))

mat &lt;- assay(rld)
mat &lt;- limma::removeBatchEffect(mat, rld$batch)

colors &lt;- colorRampPalette( rev(RColorBrewer::brewer.pal(9, &quot;BuPu&quot;)) )(255)
rows &lt;- data.frame(&quot;Treatment&quot;=rld$treatment,
                     &quot;Tissue&quot;=rld$tissue)
  
ha = HeatmapAnnotation(df = rows,
                       col = list(Treatment = c(cyt=&quot;dark orange&quot;, ctrl=&quot;steelblue3&quot;),
                                   Tissue = c(endoc=&quot;violetred3&quot;, hi=&quot;palegreen3&quot;)))
  
sampleDists &lt;- dist(t(mat), method=&quot;euclidean&quot;)
sampleDistMatrix &lt;- as.matrix(sampleDists)
hclust &lt;- hclust(sampleDists)
heat &lt;- Heatmap(sampleDistMatrix+1,
                  col=colors,
                  heatmap_legend_param = list(title = &quot;Distance&quot;, 
                                              color_bar = &quot;continuous&quot;,
                                              legend_direction=&quot;horizontal&quot;),
                  top_annotation=ha,
                  rect_gp = gpar(col= &quot;black&quot;),
                  cluster_rows=hclust,
                  cluster_columns=hclust)
  
draw(heat, heatmap_legend_side = &quot;bottom&quot;, annotation_legend_side = &quot;right&quot;)</code></pre>
<p><img src="figure/cyt_01_remodeling.Rmd/atac-heatmap-ires-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>load(file.path(out_dir, &quot;IREs_H3K27ac_clustering_rld.rda&quot;))

rld$batch[1] &lt;- &quot;endoc3&quot;
mat &lt;- assay(rld)
mat &lt;- limma::removeBatchEffect(mat, rld$batch)</code></pre>
<pre><code>Coefficients not estimable: batch8 </code></pre>
<pre class="r"><code>colors &lt;- colorRampPalette( rev(RColorBrewer::brewer.pal(9, &quot;BuPu&quot;)) )(255)
rows &lt;- data.frame(&quot;Treatment&quot;=rld$treatment,
                     &quot;Tissue&quot;=rld$tissue)
  
ha = HeatmapAnnotation(df = rows,
                       col = list(Treatment = c(cyt=&quot;dark orange&quot;, ctrl=&quot;steelblue3&quot;),
                                   Tissue = c(endoc=&quot;violetred3&quot;, hi=&quot;palegreen3&quot;)))
  
sampleDists &lt;- dist(t(mat), method=&quot;euclidean&quot;)
sampleDistMatrix &lt;- as.matrix(sampleDists)
hclust &lt;- hclust(sampleDists)
heat &lt;- Heatmap(sampleDistMatrix+1,
                  col=colors,
                  heatmap_legend_param = list(title = &quot;Distance&quot;, 
                                              color_bar = &quot;continuous&quot;,
                                              legend_direction=&quot;horizontal&quot;),
                  top_annotation=ha,
                  rect_gp = gpar(col= &quot;black&quot;),
                  cluster_rows=hclust,
                  cluster_columns=hclust)
  
draw(heat, heatmap_legend_side = &quot;bottom&quot;, annotation_legend_side = &quot;right&quot;)</code></pre>
<p><img src="figure/cyt_01_remodeling.Rmd/k27-heatmap-ires-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="characterization-of-ires-in-endoc-betah1" class="section level2">
<h2>Characterization of IREs in EndoC-<span class="math inline">\(\beta\)</span>H1</h2>
<pre class="r"><code>load(&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges.rda&quot;)

re.df &lt;- data.frame(re)[,-c(1:5)]

t &lt;- table(re.df$type, re.df$atac.annotation)
t</code></pre>
<pre><code>     
      Distal Promoter
  IRE   3594      204
  SRE  70391     9982</code></pre>
<pre class="r"><code>## Make groups
re.tss &lt;- unique(re.df[!is.na(re.df$type),])

re.tss$anno.group &lt;- NA
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)&gt;200e3] &lt;- &quot;&gt;200kb&quot;
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)&lt;=200e3 &amp;
                    abs(re.tss$atac.distanceToTSS)&gt;20e3] &lt;- &quot;20-200kb&quot;
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)&lt;=20e3 &amp;
                    abs(re.tss$atac.distanceToTSS)&gt;2e3] &lt;- &quot;2-20kb&quot;
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)&lt;=2e3] &lt;- &quot;0-2kb&quot;

len.ire &lt;- sum(grepl(&quot;IRE&quot;, re.tss$type))
len.sre &lt;- sum(grepl(&quot;SRE&quot;, re.tss$type))

sum.tss &lt;- re.tss %&gt;%
  group_by(type, anno.group) %&gt;%
  summarise(num=length(unique(atac.GeneID)))

sum.tss$perc &lt;- NA
sum.tss$perc[grep(&quot;IRE&quot;, sum.tss$type)] &lt;- sum.tss$num[grep(&quot;IRE&quot;, sum.tss$type)]/len.ire*100
sum.tss$perc[grep(&quot;SRE&quot;, sum.tss$type)] &lt;- sum.tss$num[grep(&quot;SRE&quot;, sum.tss$type)]/len.sre*100

sum.tss$anno.group &lt;- factor(sum.tss$anno.group,
                             levels=c(&quot;0-2kb&quot;, &quot;2-20kb&quot;, &quot;20-200kb&quot;, &quot;&gt;200kb&quot;))

tss.plot &lt;- 
  ggplot(sum.tss,
       aes(anno.group, perc)) +
  geom_bar(aes(fill=type), color=&quot;black&quot;, lwd=0.7, stat=&quot;identity&quot;, position=&quot;dodge&quot;) +
  geom_vline(xintercept=1.5, lty=2, color=&quot;dark red&quot;) +
  scale_fill_manual(values=pals$re,
                    name=&quot;RE type&quot;, labels=function(x) paste0(x, &quot;s&quot;)) +
  theme(legend.position=&quot;top&quot;) +
  xlab(&quot;Distance to TSS&quot;) + 
  scale_y_continuous(name=&quot;Percentage of RE&quot;,
                     labels=function(x) paste0(x, &quot;%&quot;),
                     breaks=scales::pretty_breaks())

tss.plot</code></pre>
<p><img src="figure/cyt_01_remodeling.Rmd/tss-plot-1.png" width="384" style="display: block; margin: auto;" /></p>
<pre class="r"><code>scope=1e3
bin=50

ire &lt;- re[grep(&quot;IRE&quot;, re$type),]
rnd &lt;- regioneR::randomizeRegions(ire)

ire.cons &lt;- pipelineNGS::calculateMeanCons(ire,
                                           scope=scope,
                                           bin=bin)
ire.cons$re_type &lt;- &quot;IREs&quot;
rnd.cons &lt;- pipelineNGS::calculateMeanCons(rnd,
                                           scope=scope,
                                           bin=bin)
rnd.cons$re_type &lt;- &quot;Random IREs&quot;

ire.cons &lt;- rbind(ire.cons, rnd.cons)
save(ire.cons, file=file.path(out_path, &quot;IRE_endoc_fc1_conservation.rda&quot;))</code></pre>
<pre class="r"><code>load(file.path(out_dir, &quot;IRE_endoc_fc1_conservation.rda&quot;))

cons.plot &lt;- 
  ggplot(ire.cons,
       aes(position, meanCons)) +
  geom_line(aes(lty=re_type), lwd=0.7, color=pals$re[&quot;IRE&quot;]) +
  scale_linetype_discrete(name=&quot;Region type&quot;) +
  geom_vline(xintercept=0, lty=2, color=&quot;grey&quot;) +
  ylab(&quot;Mean PhastCons46way score&quot;) +
  xlab(&quot;Position from peak center (bp)&quot;) +
  theme(legend.position = &quot;top&quot;) +
  guides(linetype=guide_legend(nrow=2))

cons.plot</code></pre>
<p><img src="figure/cyt_01_remodeling.Rmd/plot-cons-1.png" width="336" style="display: block; margin: auto;" /></p>
<pre class="r"><code>library(maRge)

out_homer &lt;- file.path(out_dir, &quot;HOMER_IREs_endoc_fc1_padj0.05_mask/&quot;)

deNovoMotifHOMER(bed=paste0(&quot;../data/CYT/bedfiles/IREs_endoc_fc1_padj0.05.bed&quot;),
                 path_output=out_homer,
                 other_param=&quot;-mask&quot;,
                 path_homer=&quot;~/tools/homer/&quot;)</code></pre>
<pre class="r"><code>htmltools::includeHTML(file.path(out_dir, &quot;HOMER_IREs_endoc_fc1_padj0.05_mask/homerResults.html&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] DESeq2_1.29.4               SummarizedExperiment_1.19.4
 [3] DelayedArray_0.15.1         matrixStats_0.56.0         
 [5] Biobase_2.49.0              ComplexHeatmap_2.5.3       
 [7] dplyr_1.0.0                 kableExtra_1.1.0           
 [9] cowplot_1.0.0               ggplot2_3.3.1              
[11] GenomicRanges_1.41.1        GenomeInfoDb_1.25.0        
[13] IRanges_2.23.5              S4Vectors_0.27.8           
[15] BiocGenerics_0.35.2         workflowr_1.6.2            

loaded via a namespace (and not attached):
 [1] bitops_1.0-6           fs_1.4.1               bit64_0.9-7           
 [4] webshot_0.5.2          RColorBrewer_1.1-2     httr_1.4.1            
 [7] rprojroot_1.3-2        tools_4.0.0            backports_1.1.7       
[10] R6_2.4.1               rpart_4.1-15           Hmisc_4.4-0           
[13] DBI_1.1.0              colorspace_1.4-1       nnet_7.3-14           
[16] GetoptLong_1.0.2       withr_2.2.0            gridExtra_2.3         
[19] tidyselect_1.1.0       bit_1.1-15.2           compiler_4.0.0        
[22] git2r_0.27.1           rvest_0.3.6            htmlTable_2.0.1       
[25] xml2_1.3.2             labeling_0.3           checkmate_2.0.0       
[28] scales_1.1.1           readr_1.3.1            genefilter_1.71.0     
[31] stringr_1.4.0          digest_0.6.25          foreign_0.8-80        
[34] rmarkdown_2.3          XVector_0.29.1         jpeg_0.1-8.1          
[37] base64enc_0.1-3        pkgconfig_2.0.3        htmltools_0.5.0       
[40] limma_3.45.9           highr_0.8              htmlwidgets_1.5.1     
[43] rlang_0.4.6            GlobalOptions_0.1.2    rstudioapi_0.11       
[46] RSQLite_2.2.0          shape_1.4.4            generics_0.0.2        
[49] farver_2.0.3           BiocParallel_1.23.0    acepack_1.4.1         
[52] RCurl_1.98-1.2         magrittr_1.5           Formula_1.2-3         
[55] GenomeInfoDbData_1.2.3 Matrix_1.2-18          Rcpp_1.0.4.6          
[58] munsell_0.5.0          lifecycle_0.2.0        stringi_1.4.6         
[61] yaml_2.2.1             zlibbioc_1.35.0        plyr_1.8.6            
[64] blob_1.2.1             promises_1.1.0         crayon_1.3.4          
[67] lattice_0.20-41        splines_4.0.0          annotate_1.67.0       
[70] circlize_0.4.10        hms_0.5.3              locfit_1.5-9.4        
[73] knitr_1.28             pillar_1.4.4           rjson_0.2.20          
[76] reshape2_1.4.4         geneplotter_1.67.0     XML_3.99-0.3          
[79] glue_1.4.1             evaluate_0.14          latticeExtra_0.6-29   
[82] data.table_1.12.8      png_0.1-7              vctrs_0.3.0           
[85] httpuv_1.5.3.1         gtable_0.3.0           purrr_0.3.4           
[88] clue_0.3-57            xfun_0.14              xtable_1.8-4          
[91] later_1.0.0            survival_3.1-12        viridisLite_0.3.0     
[94] tibble_3.0.1           AnnotationDbi_1.51.0   memoise_1.1.0         
[97] cluster_2.1.0          ellipsis_0.3.1        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
