<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mireia Ramos-Rodríguez" />


<title>#3 Subclassification of IREs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">β-cells and T1D</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-α
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_ifna.html">Description</a>
    </li>
    <li>
      <a href="ifn_01_accessibility.html">#1 Chromatin remodeling induced by IFN-α</a>
    </li>
    <li>
      <a href="ifn_02_transcription.html">#2 Chromatin remodeling correlates with transcription changes</a>
    </li>
    <li>
      <a href="ifn_03_footprint.html">#3 IRF1, STAT1 and STAT2 are key regulators of IFNα signaling</a>
    </li>
    <li>
      <a href="ifn_04_modules.html">#4 mRNA and protein modules regulated by IFN-α</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-γ + IL-1β
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_cyt.html">Description</a>
    </li>
    <li>
      <a href="cyt_01_remodeling.html">#1 Chromatin remodeling induced by cytokines</a>
    </li>
    <li>
      <a href="cyt_02_transcription.html">#2 Changes in transcription and translation</a>
    </li>
    <li>
      <a href="cyt_03_subclassification.html">#3 Subclassification of IREs</a>
    </li>
    <li>
      <a href="cyt_04_contacts.html">#4 Changes in 3D chormatin structure</a>
    </li>
    <li>
      <a href="cyt_05_t1d.html">#5 IREs and T1D risk</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IRB
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_irb.html">Description</a>
    </li>
    <li>
      <a href="https://mireia-bioinfo.github.io/plotRegulome">plotRegulome R package</a>
    </li>
    <li>
      <a href="https://github.com/mireia-bioinfo/isletregulome_shiny">IRB Shiny App code</a>
    </li>
    <li>
      <a href="http://www.isletregulome.org">Islet Regulome Browser</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mireia-bioinfo/phdthesis_code">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">#3 Subclassification of IREs</h1>
<h3 class="subtitle">Primed and neo induced regulatory elements mediate cytokine response</h3>
<h4 class="author">Mireia Ramos-Rodríguez</h4>

</div>


<div id="subclassification-of-ires" class="section level1">
<h1>Subclassification of IREs</h1>
<p>To determine which IREs were not accessible in control conditions, we call peaks from merged control BAM file with a relaxed threshold: <em>P</em>&lt;0.05.</p>
<pre><code>macs2 callpeak -f BAM -t data/CYT/ATAC/BAM/merged/endoc_ctrl.offset.bam -g hs --outdir data/CYT/ATAC/Peaks/relaxed/ -n endoc_ctrl_relaxed --tempdir data/CYT/ATAC/Peaks/tmp/ -p 0.05 --nomodel --shift -100 --extsize 200</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges.rda&quot;)
re &lt;- re[!is.na(re$type),]

## Save first subgroup
re$subgroup1 &lt;- NA
re$subgroup1[grep(&quot;SRE&quot;, re$type)] &lt;- &quot;SRE&quot;
re$subgroup1[re$atac.type==&quot;gained&quot; &amp; re$h3k27ac.type==&quot;gained&quot;] &lt;- &quot;Opening IRE&quot;
re$subgroup1[re$atac.type==&quot;stable&quot; &amp; re$h3k27ac.type==&quot;gained&quot;] &lt;- &quot;Primed IRE&quot;

## Save 2nd subgroup
re$subgroup2 &lt;- re$subgroup1

relaxed &lt;- regioneR::toGRanges(paste0(&quot;../../data/ATAC/Peaks/relaxed/endoc_ctrl_relaxed_peaks.narrowPeak&quot;))
ols &lt;- subsetByOverlaps(re[re$subgroup1==&quot;Opening IRE&quot;,],
                        relaxed)$atac.GeneID

re$subgroup2[re$atac.GeneID %in% ols &amp; 
               re$subgroup1==&quot;Opening IRE&quot;] &lt;- &quot;Other IRE&quot;
re$subgroup2[!(re$atac.GeneID %in% ols) &amp;
               re$subgroup1==&quot;Opening IRE&quot;] &lt;- &quot;Neo IRE&quot;

save(re, file=&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda&quot;)

## With less stringent threshold --- 
load(&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_k27.8.rda&quot;)
re &lt;- re[!is.na(re$type),]

## Save first subgroup
re$subgroup1 &lt;- NA
re$subgroup1[grep(&quot;SRE&quot;, re$type)] &lt;- &quot;SRE&quot;
re$subgroup1[re$atac.type==&quot;gained&quot; &amp; re$h3k27ac.type==&quot;gained&quot;] &lt;- &quot;Opening IRE&quot;
re$subgroup1[re$atac.type==&quot;stable&quot; &amp; re$h3k27ac.type==&quot;gained&quot;] &lt;- &quot;Primed IRE&quot;

## Save 2nd subgroup
re$subgroup2 &lt;- re$subgroup1

relaxed &lt;- regioneR::toGRanges(paste0(&quot;../data/CYT/ATAC/Peaks/relaxed/endoc_ctrl_relaxed_peaks.narrowPeak&quot;))
ols &lt;- subsetByOverlaps(re[re$subgroup1==&quot;Opening IRE&quot;,],
                        relaxed)$atac.GeneID

re$subgroup2[re$atac.GeneID %in% ols &amp; 
               re$subgroup1==&quot;Opening IRE&quot;] &lt;- &quot;Other IRE&quot;
re$subgroup2[!(re$atac.GeneID %in% ols) &amp;
               re$subgroup1==&quot;Opening IRE&quot;] &lt;- &quot;Neo IRE&quot;

save(re, file=&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup_k27.8.rda&quot;)</code></pre>
<pre class="r"><code>load(&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda&quot;)
re &lt;- data.frame(mcols(re)[,c(1,13,17,18)])

load(&quot;../data/CYT/ATAC/diffAnalysis/ATAC_endoc_normCountsBatch.rda&quot;)
counts &lt;- data.frame(counts)
counts$atac.GeneID &lt;- rownames(counts)
counts &lt;- reshape2::melt(counts,
                         id.var=11,
                         value.var=1:10,
                         value.name=&quot;counts&quot;,
                         variable.name=&quot;sample&quot;)
counts$treatment &lt;- unlist(lapply(strsplit(as.character(counts$sample), &quot;_&quot;),
                                  function(x) x[2]))

counts &lt;- counts %&gt;%
  group_by(atac.GeneID, treatment) %&gt;%
  summarise(counts=mean(counts))

colnames(counts)[3] &lt;- &quot;ATAC-seq&quot;

re.counts &lt;- dplyr::left_join(re, counts)

load(&quot;../data/CYT/H3K27ac/diffAnalysis/H3K27ac_endoc_normCountsBatch.rda&quot;)
counts &lt;- data.frame(counts)
counts$h3k27ac.GeneID &lt;- rownames(counts)
counts &lt;- reshape2::melt(counts,
                         id.var=9,
                         value.var=1:8,
                         value.name=&quot;counts&quot;,
                         variable.name=&quot;sample&quot;)
counts$treatment &lt;- unlist(lapply(strsplit(as.character(counts$sample), &quot;_&quot;),
                                  function(x) x[2]))

counts &lt;- counts %&gt;%
  group_by(h3k27ac.GeneID, treatment) %&gt;%
  summarise(counts=mean(counts))

colnames(counts)[3] &lt;- &quot;H3K27ac&quot;

re.counts &lt;- dplyr::left_join(re.counts, counts)

re.counts &lt;- reshape2::melt(re.counts,
                            id.vars=1:5,
                            value.vars=6:7,
                            value.name=&quot;counts&quot;,
                            variable.name=&quot;experiment&quot;)

distr.reads &lt;- 
  ggplot(re.counts[!grepl(&quot;Other&quot;, re.counts$subgroup2),],
       aes(subgroup2, log2(counts+1))) +
  geom_boxplot(aes(color=treatment), notch=TRUE, lwd=0.7) +
  scale_color_manual(values=pals$treatment,
                     labels=function(x) toupper(x),
                     name=&quot;Treatment&quot;) +
  facet_wrap(~experiment) +
  ylab(expression(Log[2]*&quot; normalized counts&quot;)) +
  scale_x_discrete(labels=function(x) paste0(x, &quot;s&quot;)) +
  theme(axis.text.x=element_text(angle=30, hjust=1),
        axis.title.x=element_blank(),
        strip.background = element_rect(fill=&quot;white&quot;, color=&quot;black&quot;,
                                        size=0.5, linetype=1))

distr.reads</code></pre>
<p><img src="figure/cyt_03_subclassification.Rmd/distr-reads-sub-ires-1.png" width="480" style="display: block; margin: auto;" /></p>
<div id="characterization" class="section level2">
<h2>Characterization</h2>
<pre class="r"><code>win.width=30e3

load(&quot;../data/CYT/RNA/diffAnalysis/RNA_endoc_GRangesBatch.rda&quot;)

rna &lt;- res.gr[res.gr$gene_biotype==&quot;protein_coding&quot;,]
win &lt;- promoters(rna)
win &lt;- resize(rna, width=win.width, fix=&quot;center&quot;)

load(&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda&quot;)
ols &lt;- findOverlaps(re, win)

anno &lt;- cbind(mcols(re)[queryHits(ols), c(1,17,18)],
              mcols(win)[subjectHits(ols), c(1,2,5)])

load(&quot;../data/CYT/RNA/diffAnalysis/RNA_endoc_normCountsBatch.rda&quot;)
counts &lt;- data.frame(counts)
counts$GeneID &lt;- rownames(counts)
counts$GeneID &lt;- gsub(&quot;\\.[[:digit:]]*&quot;, &quot;&quot;, counts$GeneID)

counts &lt;- reshape2::melt(counts,
                         id.vars=11,
                         value.vars=1:10,
                         value.name=&quot;counts&quot;,
                         variable.name=&quot;sample&quot;)
counts$treatment &lt;- unlist(lapply(strsplit(as.character(counts$sample), &quot;_&quot;),
                                  function(x) x[2]))

counts.mean &lt;- counts %&gt;%
  group_by(GeneID, treatment) %&gt;%
  summarise(counts=mean(counts))

anno.counts &lt;- left_join(data.frame(anno), counts.mean)
anno.counts &lt;- unique(anno.counts[,-1])


test &lt;- anno.counts[anno.counts$subgroup2!=&quot;SRE&quot;,] %&gt;%
  group_by(subgroup1) %&gt;%
  summarise(pval=wilcox.test(counts[treatment==&quot;ctrl&quot;],
                             counts[treatment==&quot;cyt&quot;],
                             paired=F)$p.value)

test$lab &lt;- &quot;&quot;
test$lab[test$pval&lt;0.05] &lt;- &quot;*&quot;
test$lab[test$pval&lt;0.01] &lt;- &quot;**&quot;
test$lab[test$pval&lt;0.001] &lt;- &quot;***&quot;

ggplot(anno.counts[anno.counts$subgroup2!=&quot;SRE&quot;,],
       aes(subgroup1, log2(counts + 1))) +
  geom_boxplot(aes(color=treatment), notch=TRUE, 
               outlier.shape=NA, lwd=1) +
  geom_text(data=test,
            aes(x=subgroup1, y=20, label=lab),
            size=5) +
  scale_color_manual(values=pals$treatment,
                     name=&quot;Treatment&quot;,
                     labels=function(x) toupper(x)) +
  ylab(expression(&quot;RNA-seq &quot;*log[2]*&quot; counts&quot;)) +
  scale_x_discrete(labels=function(x) paste0(x, &quot;s&quot;)) +
  theme(legend.position=&quot;top&quot;,
        axis.title.x=element_blank())</code></pre>
<p><img src="figure/cyt_03_subclassification.Rmd/assoc-sub-rna-1.png" width="384" style="display: block; margin: auto;" /></p>
<pre class="r"><code>load(paste0(&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda&quot;))
scope=1e3
bin=50

## Opening IREs
op &lt;- re[grep(&quot;Opening&quot;, re$subgroup1),]
rnd &lt;- regioneR::randomizeRegions(op)

op.cons &lt;- pipelineNGS::calculateMeanCons(op,
                                           scope=scope,
                                           bin=bin)
op.cons$re_type &lt;- &quot;Opening IRE&quot;
rnd.cons &lt;- pipelineNGS::calculateMeanCons(rnd,
                                           scope=scope,
                                           bin=bin)
rnd.cons$re_type &lt;- &quot;Random Opening IRE&quot;

ire.cons &lt;- rbind(op.cons, rnd.cons)

## Primed IREs
pr &lt;- re[grep(&quot;Primed&quot;, re$subgroup1),]
rnd &lt;- regioneR::randomizeRegions(pr)

pr.cons &lt;- pipelineNGS::calculateMeanCons(pr,
                                           scope=scope,
                                           bin=bin)
pr.cons$re_type &lt;- &quot;Primed IRE&quot;
rnd.cons &lt;- pipelineNGS::calculateMeanCons(rnd,
                                           scope=scope,
                                           bin=bin)
rnd.cons$re_type &lt;- &quot;Random Primed IRE&quot;

## Final dataset
ire.cons &lt;- rbind(ire.cons, pr.cons, rnd.cons)
save(ire.cons, file=file.path(out_dir, &quot;CONS_IREs_sub.rda&quot;))</code></pre>
<pre class="r"><code>load(file.path(out_dir, &quot;CONS_IREs_sub.rda&quot;))

ire.cons$color &lt;- gsub(&quot;Random &quot;, &quot;&quot;, ire.cons$re_type)
ire.cons$lty &lt;- gsub(&quot;Opening &quot;, &quot;&quot;, gsub(&quot;Primed &quot;, &quot;&quot;, ire.cons$re_type))

cons.plot &lt;- 
  ggplot(ire.cons,
       aes(position, meanCons)) +
  geom_line(aes(lty=lty, color=color), 
            lwd=1) +
  scale_linetype_discrete(name=&quot;Region type&quot;) +
  scale_color_manual(values=pals$re,
                     name=&quot;Region type&quot;,
                     labels=function(x) paste0(x, &quot;s&quot;)) +
  geom_vline(xintercept=0, lty=2, color=&quot;grey&quot;) +
  ylab(&quot;Mean PhastCons46way Score&quot;) +
  xlab(&quot;Position from peak center (bp)&quot;) +
  theme(legend.position = &quot;top&quot;) +
  guides(linetype=guide_legend(nrow=2),
         color=guide_legend(nrow=2))

cons.plot</code></pre>
<p><img src="figure/cyt_03_subclassification.Rmd/plot-cons-sub-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>load(&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda&quot;)
re.df &lt;- data.frame(re)[,-c(1:5)]

## Make groups
re.tss &lt;- unique(re.df[!is.na(re.df$type),])

re.tss$anno.group &lt;- NA
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)&gt;200e3] &lt;- &quot;&gt;200kb&quot;
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)&lt;=200e3 &amp;
                    abs(re.tss$atac.distanceToTSS)&gt;20e3] &lt;- &quot;20-200kb&quot;
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)&lt;=20e3 &amp;
                    abs(re.tss$atac.distanceToTSS)&gt;2e3] &lt;- &quot;2-20kb&quot;
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)&lt;=2e3] &lt;- &quot;0-2kb&quot;

len.neo &lt;- sum(grepl(&quot;Neo&quot;, re.tss$subgroup2))
len.primed &lt;- sum(grepl(&quot;Primed&quot;, re.tss$subgroup2))
len.sre &lt;- sum(grepl(&quot;SRE&quot;, re.tss$subgroup2))

sum.tss &lt;- re.tss %&gt;%
  group_by(subgroup2, anno.group) %&gt;%
  summarise(num=length(unique(atac.GeneID)))

sum.tss$perc &lt;- NA
sum.tss$perc[grep(&quot;Neo&quot;, sum.tss$subgroup2)] &lt;- sum.tss$num[grep(&quot;Neo&quot;, sum.tss$subgroup2)]/len.neo*100
sum.tss$perc[grep(&quot;Primed&quot;, sum.tss$subgroup2)] &lt;- sum.tss$num[grep(&quot;Primed&quot;, sum.tss$subgroup2)]/len.primed*100
sum.tss$perc[grep(&quot;SRE&quot;, sum.tss$subgroup2)] &lt;- sum.tss$num[grep(&quot;SRE&quot;, sum.tss$subgroup2)]/len.sre*100

sum.tss &lt;- sum.tss[!is.na(sum.tss$perc),]
sum.tss$anno.group &lt;- factor(sum.tss$anno.group,
                             levels=c(&quot;0-2kb&quot;, &quot;2-20kb&quot;, &quot;20-200kb&quot;, &quot;&gt;200kb&quot;))

tss.plot &lt;- 
  ggplot(sum.tss,
       aes(anno.group, perc)) +
  geom_bar(aes(fill=subgroup2), color=&quot;black&quot;, lwd=0.7, stat=&quot;identity&quot;, position=&quot;dodge&quot;) +
  geom_vline(xintercept=1.5, lty=2, color=&quot;dark red&quot;) +
  scale_fill_manual(values=pals$re,
                    name=&quot;RE type&quot;, labels=function(x) paste0(x, &quot;s&quot;)) +
  annotate(&quot;text&quot;, x=c(1,3), y=c(60, 60), label=c(&quot;Promoters&quot;, &quot;Enhancers&quot;)) +
  theme(legend.position=&quot;top&quot;) +
  xlab(&quot;Distance to TSS&quot;) + 
  scale_y_continuous(name=&quot;Percentage of RE&quot;,
                     labels=function(x) paste0(x, &quot;%&quot;),
                     breaks=scales::pretty_breaks()) +
  guides(fill=guide_legend(ncol=1))

tss.plot</code></pre>
<p><img src="figure/cyt_03_subclassification.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>load(&quot;../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda&quot;)

## Do it on enhancers IREs
sel &lt;- re[grep(&quot;IRE&quot;, re$type),]


files &lt;- list.files(&quot;~/data/ChromHMM_ENCODE-Broad/&quot;,
                    pattern=&quot;*.bed&quot;,
                    full.names=TRUE)
names &lt;- pipelineNGS::getNameFromPath(files,
                                      prefix=&quot;wgEncodeBroadHmm&quot;, 
                                      suffix=&quot;HMM.bed&quot;)

chrom &lt;- lapply(files, regioneR::toGRanges)
names(chrom) &lt;- names

overlaps &lt;- data.frame()
for (i in 1:length(chrom)) {
  hmm &lt;- chrom[[i]]
  ols &lt;- findOverlaps(sel,
                      hmm)
  df &lt;- cbind(sel$type[queryHits(ols)],
              sel$atac.GeneID[queryHits(ols)],
              hmm$name[subjectHits(ols)])
  df &lt;- as.data.frame(df)
  colnames(df) &lt;- c(&quot;type&quot;, &quot;peakID&quot;, &quot;chromHMM&quot;)
  df$ENCODE &lt;- names(chrom)[i]  
  overlaps &lt;- rbind(overlaps, df)
}

ids.enh &lt;- unique(overlaps$peakID[grep(&quot;Enhancer&quot;, overlaps$chromHMM)])
length(ids.enh)
ids.prom &lt;- unique(overlaps$peakID[grep(&quot;Promoter&quot;, overlaps$chromHMM)])
length(ids.prom)
ids.het &lt;- unique(overlaps$peakID[grep(&quot;Hetero&quot;, overlaps$chromHMM)])
length(ids.het)

ids.het.enh &lt;- ids.het[ids.het %in% ids.enh]
length(ids.het.enh)

## Test with a randomized set
rndm &lt;- regioneR::randomizeRegions(sel)
rndm$geneID &lt;- paste0(&quot;random_&quot;, 1:length(rndm))

overlaps.rndm &lt;- data.frame()
for (i in 1:length(chrom)) {
  hmm &lt;- chrom[[i]]
  ols &lt;- findOverlaps(rndm,
                      hmm)
  df &lt;- cbind(rndm$geneID[queryHits(ols)],
              hmm$name[subjectHits(ols)])
  df &lt;- as.data.frame(df)
  colnames(df) &lt;- c(&quot;peakID&quot;, &quot;chromHMM&quot;)
  df$ENCODE &lt;- names(chrom)[i]  
  overlaps.rndm &lt;- rbind(overlaps.rndm, df)
}

ids.enh.rndm &lt;- unique(overlaps.rndm$peakID[grep(&quot;Enhancer&quot;, overlaps.rndm$chromHMM)])
length(ids.enh.rndm)
ids.prom.rndm &lt;- unique(overlaps.rndm$peakID[grep(&quot;Promoter&quot;, overlaps.rndm$chromHMM)])
length(ids.prom.rndm)
ids.het.rndm &lt;- unique(overlaps.rndm$peakID[grep(&quot;Hetero&quot;, overlaps.rndm$chromHMM)])
length(ids.het.rndm)

ids.het.enh.rndm &lt;- ids.het.rndm[ids.het.rndm %in% ids.enh.rndm]
length(ids.het.enh.rndm)

df &lt;- data.frame(&quot;class&quot;=rep(c(&quot;Enhancer&quot;, &quot;Promoter&quot;), 2),
                 &quot;regions&quot;=c(rep(&quot;IREs&quot;, 2), rep(&quot;Random&quot;, 2)),
                 &quot;overlaps&quot;=c(length(ids.enh), length(ids.prom), 
                              length(ids.enh.rndm), length(ids.prom.rndm)))

mat &lt;- matrix(df$overlaps, ncol=2)

save(df, file=file.path(out_dir, &quot;ENH_chromHMM_IREs.rda&quot;))</code></pre>
<pre class="r"><code>load(file.path(out_dir, &quot;ENH_chromHMM_IREs.rda&quot;))

bar.chrom &lt;- 
  ggplot(df[df$class!=&quot;Promoter&quot;,],
       aes(class, overlaps)) +
  geom_bar(aes(fill=regions), stat=&quot;identity&quot;,
           position=&quot;dodge&quot;, color=&quot;black&quot;,
           width=0.7) +
  xlab(&quot;ChromHMM class&quot;) +
  scale_y_continuous(labels=scales::comma,
                     breaks=scales::pretty_breaks(),
                     name=&quot;# of overlapping IREs&quot;) +
  scale_fill_manual(values=c(&quot;#e37418&quot;, &quot;#0c3a5c&quot;),
                    name=&quot;Source&quot;) +
  annotate(&quot;text&quot;, 1, 2800, label=&quot;*&quot;, size=8) +
  theme(legend.position=c(0.55, 0.85))

bar.chrom</code></pre>
<p><img src="figure/cyt_03_subclassification.Rmd/unnamed-chunk-3-1.png" width="336" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="transcription-factor-analysis" class="section level1">
<h1>Transcription factor analysis</h1>
<div id="de-novo-motif-analysis" class="section level2">
<h2><em>de novo</em> motif analysis</h2>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] purrr_0.3.4          dplyr_1.0.0          kableExtra_1.1.0    
 [4] cowplot_1.0.0        ggplot2_3.3.2        GenomicRanges_1.41.5
 [7] GenomeInfoDb_1.25.8  IRanges_2.23.10      S4Vectors_0.27.12   
[10] BiocGenerics_0.35.4  workflowr_1.6.2     

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.0       xfun_0.16              reshape2_1.4.4        
 [4] colorspace_1.4-1       vctrs_0.3.2            generics_0.0.2        
 [7] viridisLite_0.3.0      htmltools_0.5.0        yaml_2.2.1            
[10] rlang_0.4.7            later_1.1.0.1          pillar_1.4.6          
[13] glue_1.4.1             withr_2.2.0            plyr_1.8.6            
[16] GenomeInfoDbData_1.2.3 lifecycle_0.2.0        stringr_1.4.0         
[19] zlibbioc_1.35.0        munsell_0.5.0          gtable_0.3.0          
[22] rvest_0.3.6            evaluate_0.14          labeling_0.3          
[25] knitr_1.29             httpuv_1.5.4           Rcpp_1.0.5            
[28] readr_1.3.1            promises_1.1.1         backports_1.1.8       
[31] scales_1.1.1           webshot_0.5.2          XVector_0.29.3        
[34] farver_2.0.3           fs_1.4.2               hms_0.5.3             
[37] digest_0.6.25          stringi_1.4.6          rprojroot_1.3-2       
[40] grid_4.0.0             tools_4.0.0            bitops_1.0-6          
[43] magrittr_1.5           RCurl_1.98-1.2         tibble_3.0.3          
[46] crayon_1.3.4           pkgconfig_2.0.3        ellipsis_0.3.1        
[49] xml2_1.3.2             httr_1.4.2             rmarkdown_2.3         
[52] rstudioapi_0.11        R6_2.4.1               git2r_0.27.1          
[55] compiler_4.0.0        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
