<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mireia Ramos-Rodríguez" />


<title>#4 Changes in 3D chormatin structure</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">β-cells and T1D</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-α
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_ifna.html">Description</a>
    </li>
    <li>
      <a href="ifn_01_accessibility.html">#1 Chromatin remodeling induced by IFN-α</a>
    </li>
    <li>
      <a href="ifn_02_transcription.html">#2 Chromatin remodeling correlates with transcription changes</a>
    </li>
    <li>
      <a href="ifn_03_footprint.html">#3 IRF1, STAT1 and STAT2 are key regulators of IFN-α signaling</a>
    </li>
    <li>
      <a href="ifn_04_modules.html">#4 mRNA and protein modules regulated by IFN-α</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-γ + IL-1β
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_cyt.html">Description</a>
    </li>
    <li>
      <a href="cyt_01_remodeling.html">#1 Chromatin remodeling induced by cytokines</a>
    </li>
    <li>
      <a href="cyt_02_transcription.html">#2 Changes in transcription and translation</a>
    </li>
    <li>
      <a href="cyt_03_subclassification.html">#3 Subclassification of IREs</a>
    </li>
    <li>
      <a href="cyt_04_contacts.html">#4 Changes in 3D chormatin structure</a>
    </li>
    <li>
      <a href="cyt_05_t1d.html">#5 IREs and T1D risk</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IRB
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_irb.html">Description</a>
    </li>
    <li>
      <a href="https://mireia-bioinfo.github.io/plotRegulome">plotRegulome R package</a>
    </li>
    <li>
      <a href="https://github.com/mireia-bioinfo/isletregulome_shiny">IRB Shiny App code</a>
    </li>
    <li>
      <a href="http://www.isletregulome.org">Islet Regulome Browser</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mireia-bioinfo/phdthesis_code">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">#4 Changes in 3D chormatin structure</h1>
<h4 class="author">Mireia Ramos-Rodríguez</h4>

</div>


<hr />
<div id="details" class="section level1">
<h1>Details</h1>
<ul>
<li><strong>Original publication</strong>:</li>
</ul>
<blockquote>
<p><strong>Ramos-Rodríguez, M.</strong>, Raurell-Vila, H., Colli, M.L. et al. The impact of proinflammatory cytokines on the β-cell regulatory landscape provides insights into the genetics of type 1 diabetes. <em>Nat Genet.</em> <strong>51</strong>, 1588–1595 (2019) <a href="https://doi.org/10.1038/s41588-019-0524-6" class="uri">https://doi.org/10.1038/s41588-019-0524-6</a></p>
</blockquote>
<ul>
<li><p><strong>Contents</strong>: Analyses and figures contained in this document correspond to the following figures/sections of the original publication:</p>
<ul>
<li><strong>Results</strong>: “Changes in islet 3D chromatin structure”.</li>
<li><strong>Figure 3</strong>: “Cytokine exposure induces changes in human islet 3D chromatin structure”. Panels <strong>a</strong> to <strong>c</strong>.</li>
<li><strong>Extended Data Figure 5</strong>: “3D chromatin changes induced by exposure of human islets to pro-inflammatory cytokines”. Panels <strong>b</strong> to <strong>d</strong>.</li>
</ul></li>
</ul>
<hr />
</div>
<div id="process-and-calculate-differential-contacts-at-regulatory-elements" class="section level1">
<h1>Process and calculate differential contacts at regulatory elements</h1>
<pre><code>Rscript code/CYT_UMI4C_processing.R</code></pre>
<pre class="r"><code>devtools::load_all(&quot;~/tools/umi4cCatcheR/&quot;)
library(umi4cPackage)
conf &lt;- &quot;../data/CYT/UMI4C/conf/&quot;
umi4cPackage::p4cLoadConfFiles(conf)

df &lt;- read.delim(&quot;../data/CYT/UMI4C/UMI4C_promoters_views.tsv&quot;, stringsAsFactors=T, header=T)

tracks &lt;- gtrack.ls()

for (i in 1:nrow(df)) {
  sel &lt;- tracks[grepl(df$bait[i], tracks) &amp;
                   grepl(&quot;_m_&quot;, tracks)]
  
  res &lt;- process4CProfiles(treat_name=sel[2],
                         ctrl_name=sel[1],
                         scope=1e6,
                         min_win_mols=50,
                         name_bait=df$bait[i])
  
  diff &lt;- diffContacts(res,
                       times_mean=20,
                       exclude_viewpoint=3e3,
                       adj.threshold=0.05,
                       adj.method=&quot;none&quot;,
                       min_mols_test=0,
                       min_odds_ratio=1)

  save(res, diff,
       file=file.path(out_dir&quot;UMI4C_norm_results_&quot;, df$bait[i], &quot;.rda&quot;))
}</code></pre>
</div>
<div id="plot-umi-4c-at-up-regulated-gene-promoters" class="section level1">
<h1>Plot UMI-4C at up-regulated gene promoters</h1>
<pre class="r"><code>## RE ---------------------------------------------------------------
load(&quot;../data/CYT/REs/REs_hi_fc1_padj0.05_granges_subgroup.rda&quot;)

## Genes ------------------------------------------------------------
load(&quot;../data/CYT//RNA/diffAnalysis/RNA_hi_GRangesBatch.rda&quot;)
res.gr$type[res.gr$baseMean&lt;=1] &lt;- &quot;not-expressed&quot;
res.gr &lt;- res.gr[res.gr$gene_biotype==&quot;protein_coding&quot;,]

col.df &lt;- data.frame(&quot;type&quot;=c(names(pals$differential), &quot;not-expressed&quot;),
                     &quot;color&quot;=c(pals$differential, &quot;black&quot;),
                     stringsAsFactors = FALSE)
col.df$color[grep(&quot;grey&quot;, col.df$color)] &lt;- &quot;grey39&quot;
mcols(res.gr) &lt;- dplyr::left_join(data.frame(mcols(res.gr)[,c(1:2,10)]),
                                      col.df)
colnames(mcols(res.gr))[1] &lt;- &quot;ensembl_gene_id&quot;

mcol.color=4
mcol.name=2
mcol.ensembl=1</code></pre>
<pre class="r"><code>df &lt;- read.delim(&quot;../data/CYT/UMI4C/UMI4C_promoters_views.tsv&quot;, stringsAsFactors=T, header=T)

for (i in 1:length(df$bait)) {
  load(file.path(out_dir, paste0(&quot;UMI4C_norm_results_&quot;, df$bait[i], &quot;.rda&quot;)))
  
  xlim &lt;- c(df$start[i], df$end[i])
  region &lt;- GRanges(seqnames=paste0(seqnames(res$bait)),
                    ranges=IRanges(start=xlim[1],
                                   end=xlim[2]))
  
  ### Genes ------------------
  plot.genes &lt;- plotGenes(genes=res.gr[res.gr$type!=&quot;not-expressed&quot;,],
                          which=region,
                          mcol.color=mcol.color,
                          mcol.name=mcol.name,
                          mcol.ensembl=mcol.ensembl)
  
  g.plots &lt;-
    ggplot() +
    plot.genes +
    xlim(xlim) +
    scale_fill_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    scale_color_manual(values=pals$differential,
                       name=&quot;Gene expression&quot;,
                       labels=c(gained=&quot;Up-regulated&quot;,
                                lost=&quot;Down-regulated&quot;,
                                stable=&quot;Equal-regulated&quot;)) + 
    themeXblank() +
    themeYblank()
  
  
  ### UMI-4C ------------------
  umi &lt;-
    ggplot(res$norm_trend,
           aes(start, trend)) +
    geom_ribbon(aes(ymin=devM, ymax=devP, group=interaction(group, sample)),
                color=NA, fill=&quot;light grey&quot;) +
    geom_line(aes(color=sample, group=interaction(group, sample)),
              lwd=0.7) +
    annotate(&quot;point&quot;, x=start(res$bait), y=df$ymax[i], pch=25, fill=&quot;black&quot;,
             size=3) +
    annotate(&quot;text&quot;, x=start(res$bait), y=df$ymax[i]-0.2, label=df$bait[i],
             size=3) +
    scale_color_manual(values=c(ctrl=&quot;#1f78b4&quot;, treat=&quot;#d95f02&quot;),
                       labels=c(&quot;CYT&quot;, &quot;CTRL&quot;), name=&quot;&quot;) +
    scale_y_continuous(name=&quot;# UMI-4C contacts&quot;,
                       limits=c(0, df$ymax[i]),
                       breaks=scales::pretty_breaks(),
                       expand=c(0,0)) +
    xlim(xlim) +
    theme(legend.position=&quot;right&quot;)
  
  ### RE ------------------
  re.sel &lt;- as.data.frame(subsetByOverlaps(resize(re, 3e3, fix=&quot;center&quot;),
                                           region))
  
  ire &lt;-
    ggplot(re.sel) +
    geom_rect(aes(xmin=start, xmax=end, ymin=0, ymax=1, fill=h3k27ac.log2FoldChange)) +
    scale_fill_gradient2(low=&quot;dark grey&quot;,
                         mid=&quot;grey&quot;,
                        high=&quot;dark green&quot;,
                        name=expression(&quot;H3K27ac &quot;*log[2]*&quot; FC&quot;),
                        breaks=scales::pretty_breaks(n=3),
                        midpoint=0,
                        guide = guide_colorbar(direction = &quot;horizontal&quot;,
                                               title.position=&quot;top&quot;)) +
    scale_x_continuous(name=paste(&quot;Coordinates&quot;, seqnames(region), &quot;(Mb)&quot;),
                       breaks=scales::pretty_breaks(),
                       limits=xlim,
                       labels=function(x) round(x/1e6, 2),
                       expand=c(0,0))
  
  
  ### Get legends -----------
  gene.leg &lt;- get_legend(g.plots)
  umi.leg &lt;- get_legend(umi)
  diff.leg &lt;- get_legend(diff$plot)
  ire.leg &lt;- get_legend(ire)
  
  
  legends &lt;- plot_grid(gene.leg, umi.leg, diff.leg, ire.leg, ncol=1)
  ### Grid ------------------
  p &lt;- 
    plot_grid(g.plots + theme(plot.margin = margin(1,0,0,0, &quot;cm&quot;),
                              legend.position = &quot;none&quot;),
            umi + themeXblank() + theme(legend.position = &quot;none&quot;), 
            diff$plot + xlim(xlim) + themeXblank() + themeYblank() + theme(legend.position = &quot;none&quot;),
            ire + themeYblank() + theme(legend.position = &quot;none&quot;),
            ncol=1, 
            rel_heights = c(0.25, 0.45, 0.1, 0.2),
            align=&quot;v&quot;)
  
  plot &lt;- 
    plot_grid(p, 
              legends,
              ncol=2, rel_widths=c(0.7, 0.3))
  
  print(plot)
}</code></pre>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-1">
Past versions of import-pdfs-img-1.png
</button>
</p>
<div id="fig-import-pdfs-img-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-1.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-2">
Past versions of import-pdfs-img-2.png
</button>
</p>
<div id="fig-import-pdfs-img-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-2.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-3">
Past versions of import-pdfs-img-3.png
</button>
</p>
<div id="fig-import-pdfs-img-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-3.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-4">
Past versions of import-pdfs-img-4.png
</button>
</p>
<div id="fig-import-pdfs-img-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-4.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-5">
Past versions of import-pdfs-img-5.png
</button>
</p>
<div id="fig-import-pdfs-img-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-5.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-6">
Past versions of import-pdfs-img-6.png
</button>
</p>
<div id="fig-import-pdfs-img-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-6.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-7">
Past versions of import-pdfs-img-7.png
</button>
</p>
<div id="fig-import-pdfs-img-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-7.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-8">
Past versions of import-pdfs-img-8.png
</button>
</p>
<div id="fig-import-pdfs-img-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-8.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-9">
Past versions of import-pdfs-img-9.png
</button>
</p>
<div id="fig-import-pdfs-img-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-9.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-10">
Past versions of import-pdfs-img-10.png
</button>
</p>
<div id="fig-import-pdfs-img-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-10.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-11">
Past versions of import-pdfs-img-11.png
</button>
</p>
<div id="fig-import-pdfs-img-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-11.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-12">
Past versions of import-pdfs-img-12.png
</button>
</p>
<div id="fig-import-pdfs-img-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-12.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-13">
Past versions of import-pdfs-img-13.png
</button>
</p>
<div id="fig-import-pdfs-img-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-13.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-14">
Past versions of import-pdfs-img-14.png
</button>
</p>
<div id="fig-import-pdfs-img-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-14.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-15">
Past versions of import-pdfs-img-15.png
</button>
</p>
<div id="fig-import-pdfs-img-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-15.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cyt_04_contacts.Rmd/import-pdfs-img-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-import-pdfs-img-16">
Past versions of import-pdfs-img-16.png
</button>
</p>
<div id="fig-import-pdfs-img-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/import-pdfs-img-16.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="plot-zoom-ins" class="section level2">
<h2>Plot zoom-ins</h2>
<div id="tnfsf10" class="section level3">
<h3>TNFSF10</h3>
<pre class="r"><code>load(file.path(out_dir, &quot;UMI4C_norm_results_TNFSF10.rda&quot;))

wins &lt;- as.character(diff$results$id[diff$results$sign==&quot;yes&quot;])

coord &lt;- diff$results[diff$results$id %in% wins, c(8:9,1)]
reg &lt;- GRanges(paste0(as.character(seqnames(res$bait)),
                        &quot;:&quot;, min(coord$start), &quot;-&quot;, max(coord$end)))

tracks &lt;- c(list.files(&quot;../data/CYT/ATAC/Visualization&quot;,
                       pattern=&quot;.bw&quot;,
                       full.names=T),
            list.files(&quot;../data/CYT/H3K27ac/Visualization&quot;,
                       pattern=&quot;.bw&quot;,
                       full.names=T))

tracks &lt;- tracks[!grepl(&quot;[[:digit:]]_&quot;, tracks) &amp;
                   grepl(&quot;hi_&quot;, tracks)]</code></pre>
<pre class="r"><code>## ATAC-seq -------------------
sm.at &lt;- 20

ctrl.at &lt;- rtracklayer::import(tracks[grepl(&quot;ATAC&quot;, tracks) &amp; grepl(&quot;ctrl&quot;, tracks)],
                               which=reg)
score(ctrl.at) &lt;- zoo::rollmean(score(ctrl.at), sm.at,
                                fill=c(NA, NA, NA))

cyt.at &lt;- rtracklayer::import(tracks[grepl(&quot;ATAC&quot;, tracks) &amp; grepl(&quot;cyt&quot;, tracks)],
                               which=reg)
score(cyt.at) &lt;- zoo::rollmean(score(cyt.at), sm.at,
                               fill=c(NA, NA, NA))

## H3K27ac -------------------
sm.ac &lt;- 20
ctrl.ac &lt;- rtracklayer::import(tracks[grepl(&quot;H3K27ac&quot;, tracks) &amp; grepl(&quot;ctrl&quot;, tracks)],
                               which=reg)
score(ctrl.ac) &lt;- zoo::rollmean(score(ctrl.ac), sm.ac,
                                fill=c(NA, NA, NA))

cyt.ac &lt;- rtracklayer::import(tracks[grepl(&quot;H3K27ac&quot;, tracks) &amp; grepl(&quot;cyt&quot;, tracks)],
                               which=reg)
score(cyt.ac) &lt;- zoo::rollmean(score(cyt.ac), sm.ac,
                               fill=c(NA, NA, NA))</code></pre>
<pre class="r"><code>##--------------------
## Plot
##--------------------
xlims &lt;- c(start(ranges(reg)),
           end(ranges(reg)))

ctrl.at.p &lt;-
  ggplot(data.frame(ctrl.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,30),
                     expand=c(0,0),
                     breaks=c(0,30)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.at.p &lt;-
  ggplot(data.frame(cyt.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,30),
                     expand=c(0,0),
                     breaks=c(0,30)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
  

ctrl.ac.p &lt;-
  ggplot(data.frame(ctrl.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,60),
                     expand=c(0,0),
                     breaks=c(0,60)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.ac.p &lt;-
  ggplot(data.frame(cyt.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,60),
                     expand=c(0,0),
                     breaks=c(0,60)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))

plot_grid(ctrl.at.p,
          cyt.at.p,
          ctrl.ac.p,
          cyt.ac.p,
          align=&quot;v&quot;,
          ncol=1)</code></pre>
<p><img src="figure/cyt_04_contacts.Rmd/tnfsf-plot-1.png" width="192" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-tnfsf-plot-1">
Past versions of tnfsf-plot-1.png
</button>
</p>
<div id="fig-tnfsf-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/tnfsf-plot-1.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gbp1" class="section level3">
<h3>GBP1</h3>
<div id="region-1" class="section level4">
<h4>Region 1</h4>
<pre class="r"><code>load(file.path(out_dir, &quot;UMI4C_norm_results_GBP1.rda&quot;))

wins &lt;- as.character(diff$results$id[diff$results$sign==&quot;yes&quot;])[2]
wins &lt;- c(wins, &quot;window_117&quot;)
coord &lt;- diff$results[diff$results$id %in% wins, c(8:9,1)]
reg &lt;- GRanges(paste0(as.character(seqnames(res$bait)),
                        &quot;:&quot;, min(coord$start), &quot;-&quot;, max(coord$end)))

tracks &lt;- c(list.files(&quot;../data/CYT/ATAC/Visualization&quot;,
                       pattern=&quot;.bw&quot;,
                       full.names=T),
            list.files(&quot;../data/CYT/H3K27ac/Visualization&quot;,
                       pattern=&quot;.bw&quot;,
                       full.names=T))

tracks &lt;- tracks[!grepl(&quot;[[:digit:]]_&quot;, tracks) &amp;
                   grepl(&quot;hi_&quot;, tracks)]</code></pre>
<pre class="r"><code>## ATAC-seq -------------------
sm.at &lt;- 20

ctrl.at &lt;- rtracklayer::import(tracks[grepl(&quot;ATAC&quot;, tracks) &amp; grepl(&quot;ctrl&quot;, tracks)],
                               which=reg)
score(ctrl.at) &lt;- zoo::rollmean(score(ctrl.at), sm.at,
                                fill=c(NA, NA, NA))

cyt.at &lt;- rtracklayer::import(tracks[grepl(&quot;ATAC&quot;, tracks) &amp; grepl(&quot;cyt&quot;, tracks)],
                               which=reg)
score(cyt.at) &lt;- zoo::rollmean(score(cyt.at), sm.at,
                               fill=c(NA, NA, NA))

## H3K27ac -------------------
sm.ac &lt;- 20
ctrl.ac &lt;- rtracklayer::import(tracks[grepl(&quot;H3K27ac&quot;, tracks) &amp; grepl(&quot;ctrl&quot;, tracks)],
                               which=reg)
score(ctrl.ac) &lt;- zoo::rollmean(score(ctrl.ac), sm.ac,
                                fill=c(NA, NA, NA))

cyt.ac &lt;- rtracklayer::import(tracks[grepl(&quot;H3K27ac&quot;, tracks) &amp; grepl(&quot;cyt&quot;, tracks)],
                               which=reg)
score(cyt.ac) &lt;- zoo::rollmean(score(cyt.ac), sm.ac,
                               fill=c(NA, NA, NA))</code></pre>
<pre class="r"><code>##--------------------
## Plot
##--------------------
xlims &lt;- c(start(ranges(reg)),
           end(ranges(reg)))

ctrl.at.p &lt;-
  ggplot(data.frame(ctrl.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,15),
                     expand=c(0,0),
                     breaks=c(0,15)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.at.p &lt;-
  ggplot(data.frame(cyt.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,15),
                     expand=c(0,0),
                     breaks=c(0,15)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
  

ctrl.ac.p &lt;-
  ggplot(data.frame(ctrl.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,10),
                     expand=c(0,0),
                     breaks=c(0,10)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.ac.p &lt;-
  ggplot(data.frame(cyt.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,10),
                     expand=c(0,0),
                     breaks=c(0,10)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))

plot_grid(ctrl.at.p,
          cyt.at.p,
          ctrl.ac.p,
          cyt.ac.p,
          align=&quot;v&quot;,
          ncol=1)</code></pre>
<p><img src="figure/cyt_04_contacts.Rmd/gbp-1-plot-1.png" width="192" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gbp-1-plot-1">
Past versions of gbp-1-plot-1.png
</button>
</p>
<div id="fig-gbp-1-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/gbp-1-plot-1.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="region-2" class="section level4">
<h4>Region 2</h4>
<pre class="r"><code>load(file.path(out_dir, &quot;UMI4C_norm_results_GBP1.rda&quot;))

wins &lt;- as.character(diff$results$id[diff$results$sign==&quot;yes&quot;])[3]

coord &lt;- diff$results[diff$results$id %in% wins, c(8:9,1)]
reg &lt;- GRanges(paste0(as.character(seqnames(res$bait)),
                        &quot;:&quot;, min(coord$start), &quot;-&quot;, max(coord$end)))

tracks &lt;- c(list.files(&quot;../data/CYT/ATAC/Visualization&quot;,
                       pattern=&quot;.bw&quot;,
                       full.names=T),
            list.files(&quot;../data/CYT/H3K27ac/Visualization&quot;,
                       pattern=&quot;.bw&quot;,
                       full.names=T))

tracks &lt;- tracks[!grepl(&quot;[[:digit:]]_&quot;, tracks) &amp;
                   grepl(&quot;hi_&quot;, tracks)]</code></pre>
<pre class="r"><code>## ATAC-seq -------------------
sm.at &lt;- 20

ctrl.at &lt;- rtracklayer::import(tracks[grepl(&quot;ATAC&quot;, tracks) &amp; grepl(&quot;ctrl&quot;, tracks)],
                               which=reg)
score(ctrl.at) &lt;- zoo::rollmean(score(ctrl.at), sm.at,
                                fill=c(NA, NA, NA))

cyt.at &lt;- rtracklayer::import(tracks[grepl(&quot;ATAC&quot;, tracks) &amp; grepl(&quot;cyt&quot;, tracks)],
                               which=reg)
score(cyt.at) &lt;- zoo::rollmean(score(cyt.at), sm.at,
                               fill=c(NA, NA, NA))

## H3K27ac -------------------
sm.ac &lt;- 20
ctrl.ac &lt;- rtracklayer::import(tracks[grepl(&quot;H3K27ac&quot;, tracks) &amp; grepl(&quot;ctrl&quot;, tracks)],
                               which=reg)
score(ctrl.ac) &lt;- zoo::rollmean(score(ctrl.ac), sm.ac,
                                fill=c(NA, NA, NA))

cyt.ac &lt;- rtracklayer::import(tracks[grepl(&quot;H3K27ac&quot;, tracks) &amp; grepl(&quot;cyt&quot;, tracks)],
                               which=reg)
score(cyt.ac) &lt;- zoo::rollmean(score(cyt.ac), sm.ac,
                               fill=c(NA, NA, NA))</code></pre>
<pre class="r"><code>##--------------------
## Plot
##--------------------
xlims &lt;- c(start(ranges(reg)),
           end(ranges(reg)))

ctrl.at.p &lt;-
  ggplot(data.frame(ctrl.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,40),
                     expand=c(0,0),
                     breaks=c(0,40)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.at.p &lt;-
  ggplot(data.frame(cyt.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,40),
                     expand=c(0,0),
                     breaks=c(0,40)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
  

ctrl.ac.p &lt;-
  ggplot(data.frame(ctrl.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,60),
                     expand=c(0,0),
                     breaks=c(0,60)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.ac.p &lt;-
  ggplot(data.frame(cyt.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,60),
                     expand=c(0,0),
                     breaks=c(0,60)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))

plot_grid(ctrl.at.p,
          cyt.at.p,
          ctrl.ac.p,
          cyt.ac.p,
          align=&quot;v&quot;,
          ncol=1)</code></pre>
<p><img src="figure/cyt_04_contacts.Rmd/gbp-2-plot-1.png" width="192" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gbp-2-plot-1">
Past versions of gbp-2-plot-1.png
</button>
</p>
<div id="fig-gbp-2-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/gbp-2-plot-1.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="region-3" class="section level4">
<h4>Region 3</h4>
<pre class="r"><code>load(file.path(out_dir, &quot;UMI4C_norm_results_GBP1.rda&quot;))

wins &lt;- as.character(diff$results$id[diff$results$sign==&quot;yes&quot;])[4]

coord &lt;- diff$results[diff$results$id %in% wins, c(8:9,1)]
reg &lt;- GRanges(paste0(as.character(seqnames(res$bait)),
                        &quot;:&quot;, min(coord$start), &quot;-&quot;, max(coord$end)))

tracks &lt;- c(list.files(&quot;../data/CYT/ATAC/Visualization&quot;,
                       pattern=&quot;.bw&quot;,
                       full.names=T),
            list.files(&quot;../data/CYT/H3K27ac/Visualization&quot;,
                       pattern=&quot;.bw&quot;,
                       full.names=T))

tracks &lt;- tracks[!grepl(&quot;[[:digit:]]_&quot;, tracks) &amp;
                   grepl(&quot;hi_&quot;, tracks)]</code></pre>
<pre class="r"><code>## ATAC-seq -------------------
sm.at &lt;- 20

ctrl.at &lt;- rtracklayer::import(tracks[grepl(&quot;ATAC&quot;, tracks) &amp; grepl(&quot;ctrl&quot;, tracks)],
                               which=reg)
score(ctrl.at) &lt;- zoo::rollmean(score(ctrl.at), sm.at,
                                fill=c(NA, NA, NA))

cyt.at &lt;- rtracklayer::import(tracks[grepl(&quot;ATAC&quot;, tracks) &amp; grepl(&quot;cyt&quot;, tracks)],
                               which=reg)
score(cyt.at) &lt;- zoo::rollmean(score(cyt.at), sm.at,
                               fill=c(NA, NA, NA))

## H3K27ac -------------------
sm.ac &lt;- 20
ctrl.ac &lt;- rtracklayer::import(tracks[grepl(&quot;H3K27ac&quot;, tracks) &amp; grepl(&quot;ctrl&quot;, tracks)],
                               which=reg)
score(ctrl.ac) &lt;- zoo::rollmean(score(ctrl.ac), sm.ac,
                                fill=c(NA, NA, NA))

cyt.ac &lt;- rtracklayer::import(tracks[grepl(&quot;H3K27ac&quot;, tracks) &amp; grepl(&quot;cyt&quot;, tracks)],
                               which=reg)
score(cyt.ac) &lt;- zoo::rollmean(score(cyt.ac), sm.ac,
                               fill=c(NA, NA, NA))</code></pre>
<pre class="r"><code>##--------------------
## Plot
##--------------------
xlims &lt;- c(start(ranges(reg)),
           end(ranges(reg)))

ctrl.at.p &lt;-
  ggplot(data.frame(ctrl.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,60),
                     expand=c(0,0),
                     breaks=c(0,60)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.at.p &lt;-
  ggplot(data.frame(cyt.at)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,60),
                     expand=c(0,0),
                     breaks=c(0,60)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
  

ctrl.ac.p &lt;-
  ggplot(data.frame(ctrl.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;ctrl&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,50),
                     expand=c(0,0),
                     breaks=c(0,50)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))
    

cyt.ac.p &lt;-
  ggplot(data.frame(cyt.ac)) +
  geom_area(aes(x=start,
                y=score),
            fill=pals$treatment[&quot;cyt&quot;]) +
  scale_y_continuous(name=&quot;&quot;, 
                     limits=c(0,50),
                     expand=c(0,0),
                     breaks=c(0,50)) +
  xlim(xlims) +
  themeXblank() +
  theme(plot.margin=unit(c(0.5,0,0,0), &quot;cm&quot;))

plot_grid(ctrl.at.p,
          cyt.at.p,
          ctrl.ac.p,
          cyt.ac.p,
          align=&quot;v&quot;,
          ncol=1)</code></pre>
<p><img src="figure/cyt_04_contacts.Rmd/gbp-3-plot-1.png" width="192" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gbp-3-plot-1">
Past versions of gbp-3-plot-1.png
</button>
</p>
<div id="fig-gbp-3-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/gbp-3-plot-1.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
</div>
<div id="distribution-of-umi-4c-contact-changes-at-promoters-of-up-regulated-genes" class="section level1">
<h1>Distribution of UMI-4C contact changes at promoters of up-regulated genes</h1>
<pre class="r"><code>files &lt;- list.files(out_dir,
                    pattern=&quot;test&quot;,
                    full.names=T)
test.all &lt;- data.frame()
for (i in files) {
  load(i)
  
  test &lt;- test[,c(4,21,25)]
  name &lt;- pipelineNGS::getNameFromPath(i, prefix=&quot;UMI4C_test_promoters_&quot;,
                                       suffix=&quot;.rda&quot;)
  test$bait &lt;- name
  test.all &lt;- rbind(test.all, test)
}

test.all &lt;- test.all[!grepl(&quot;Other&quot;, test.all$subgroup2),]</code></pre>
<pre class="r"><code>ggplot(test.all,
       aes(subgroup2, log2_foldChange)) +
  geom_hline(yintercept=0, lty=2, color=&quot;grey&quot;) + 
  geom_boxplot(aes(color=subgroup2), notch=T, lwd=1) +
  scale_color_manual(values=pals$re) +
  scale_x_discrete(name=&quot;RE type&quot;, 
                   labels=function(x) paste0(x, &quot;s&quot;)) +
  ylab(expression(Log[2]~FC~UMI4C~contacts)) +
  theme(legend.position=&quot;none&quot;,
        axis.text.x=element_text(angle=30, hjust=1))</code></pre>
<p><img src="figure/cyt_04_contacts.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mireia-bioinfo/phdthesis_code/blob/90643f925355ff475a87c85d2c0b75a17bcc8d99/docs/figure/cyt_04_contacts.Rmd/unnamed-chunk-2-1.png" target="_blank">90643f9</a>
</td>
<td>
mireia-bioinfo
</td>
<td>
2020-08-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=es_ES.UTF-8        LC_COLLATE=C              
 [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] umi4cCatcheR_0.0.0.9000 RColorBrewer_1.1-2      GenomicRanges_1.41.5   
 [4] GenomeInfoDb_1.25.8     IRanges_2.23.10         S4Vectors_0.27.12      
 [7] BiocGenerics_0.35.4     cowplot_1.0.0           ggplot2_3.3.2          
[10] dplyr_1.0.1             workflowr_1.6.2        

loaded via a namespace (and not attached):
 [1] bitops_1.0-6                matrixStats_0.56.0         
 [3] fs_1.5.0                    usethis_1.6.1              
 [5] devtools_2.3.1              rprojroot_1.3-2            
 [7] tools_4.0.2                 backports_1.1.8            
 [9] R6_2.4.1                    colorspace_1.4-1           
[11] withr_2.2.0                 tidyselect_1.1.0           
[13] gridExtra_2.3               prettyunits_1.1.1          
[15] processx_3.4.3              compiler_4.0.2             
[17] git2r_0.27.1                cli_2.0.2                  
[19] Biobase_2.49.0              desc_1.2.0                 
[21] DelayedArray_0.15.7         rtracklayer_1.49.4         
[23] labeling_0.3                bookdown_0.20              
[25] scales_1.1.1                callr_3.4.3                
[27] askpass_1.1                 stringr_1.4.0              
[29] digest_0.6.25               Rsamtools_2.5.3            
[31] rmarkdown_2.3               XVector_0.29.3             
[33] pkgconfig_2.0.3             htmltools_0.5.0            
[35] sessioninfo_1.1.1           rlang_0.4.7                
[37] rstudioapi_0.11             generics_0.0.2             
[39] farver_2.0.3                zoo_1.8-8                  
[41] BiocParallel_1.23.2         RCurl_1.98-1.2             
[43] magrittr_1.5                GenomeInfoDbData_1.2.3     
[45] Matrix_1.2-18               Rcpp_1.0.5                 
[47] munsell_0.5.0               fansi_0.4.1                
[49] lifecycle_0.2.0             stringi_1.4.6              
[51] whisker_0.4                 yaml_2.2.1                 
[53] SummarizedExperiment_1.19.6 zlibbioc_1.35.0            
[55] pkgbuild_1.1.0              grid_4.0.2                 
[57] promises_1.1.1              crayon_1.3.4               
[59] lattice_0.20-41             Biostrings_2.57.2          
[61] magick_2.4.0                knitr_1.29                 
[63] ps_1.3.3                    pillar_1.4.6               
[65] pkgload_1.1.0               XML_3.99-0.5               
[67] glue_1.4.1                  evaluate_0.14              
[69] pdftools_2.3.1              qpdf_1.1                   
[71] remotes_2.2.0               vctrs_0.3.2                
[73] httpuv_1.5.4                testthat_2.3.2             
[75] gtable_0.3.0                purrr_0.3.4                
[77] assertthat_0.2.1            xfun_0.16                  
[79] later_1.1.0.1               tibble_3.0.3               
[81] GenomicAlignments_1.25.3    memoise_1.1.0              
[83] ellipsis_0.3.1             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
