<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mireia Ramos-Rodríguez" />


<title>#3 IRF1, STAT1 and STAT2 are key regulators of IFN-α signaling</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">β-cells and T1D</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-α
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_ifna.html">Description</a>
    </li>
    <li>
      <a href="ifn_01_accessibility.html">#1 Chromatin remodeling induced by IFN-α</a>
    </li>
    <li>
      <a href="ifn_02_transcription.html">#2 Chromatin remodeling correlates with transcription changes</a>
    </li>
    <li>
      <a href="ifn_03_footprint.html">#3 IRF1, STAT1 and STAT2 are key regulators of IFN-α signaling</a>
    </li>
    <li>
      <a href="ifn_04_modules.html">#4 mRNA and protein modules regulated by IFN-α</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IFN-γ + IL-1β
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_cyt.html">Description</a>
    </li>
    <li>
      <a href="cyt_01_remodeling.html">#1 Chromatin remodeling induced by cytokines</a>
    </li>
    <li>
      <a href="cyt_02_transcription.html">#2 Changes in transcription and translation</a>
    </li>
    <li>
      <a href="cyt_03_subclassification.html">#3 Subclassification of IREs</a>
    </li>
    <li>
      <a href="cyt_04_contacts.html">#4 Changes in 3D chormatin structure</a>
    </li>
    <li>
      <a href="cyt_05_t1d.html">#5 IREs and T1D risk</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    IRB
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index_irb.html">Description</a>
    </li>
    <li>
      <a href="https://mireia-bioinfo.github.io/plotRegulome">plotRegulome R package</a>
    </li>
    <li>
      <a href="https://github.com/mireia-bioinfo/isletregulome_shiny">IRB Shiny App code</a>
    </li>
    <li>
      <a href="http://www.isletregulome.org">Islet Regulome Browser</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mireia-bioinfo/phdthesis_code">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">#3 IRF1, STAT1 and STAT2 are key regulators of IFN-α signaling</h1>
<h4 class="author">Mireia Ramos-Rodríguez</h4>

</div>


<hr />
<div id="details" class="section level2">
<h2>Details</h2>
<ul>
<li><p><strong>Original publication</strong>:</p>
<blockquote>
<p>Colli, M.L., <strong>Ramos-Rodríguez, M.</strong>, Nakayasu, E.S. et al. An integrated multi-omics approach identifies the landscape of interferon-α-mediated responses of human pancreatic beta cells. <em>Nat Commun</em> <strong>11</strong>, 2584 (2020). <a href="https://doi.org/10.1038/s41467-020-16327-0" class="uri">https://doi.org/10.1038/s41467-020-16327-0</a></p>
</blockquote></li>
<li><p><strong>Contents</strong>: Analyses and figures contained in this document correspond to the following figures/sections of the original publication:</p>
<ul>
<li><strong>Results</strong>: “IRF1, STAT1 and STAT2 are key regulators of IFN-<span class="math inline">\(\alpha\)</span> signaling”.</li>
<li><strong>Figure 2</strong>: “IRF1, STAT1 and STAT2 regulate IFN<span class="math inline">\(\alpha\)</span>-induced transcription and the expression of checkpoint proteins”. Panel <strong>b</strong>.</li>
</ul></li>
</ul>
<hr />
</div>
<div id="methods" class="section level2">
<h2>Methods</h2>
<ol style="list-style-type: decimal">
<li><strong>Assign each open chromatin region</strong> (stable/gained/lost) to the nearest gene that is annotated in any of the clusters. Such open chromatin region will also be annotated to the <strong>cluster where the gene belongs</strong>. Filter out the open chromatin regions that are more than 1Mb away from the TSS of any gene.
<ul>
<li>In the case of open chromatin present at 2h we are annotating them to clusters defined at two hours. The same with 24h (see table below).</li>
</ul></li>
<li>Perform <strong>footprint analysis</strong> of the TF regulating the cluster in the open chromatin regions that were annotated to the cluster.</li>
</ol>
<table>
<thead>
<tr class="header">
<th>2 hours</th>
<th>24 hours</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Cluster 1</td>
<td>Cluster 1A</td>
</tr>
<tr class="even">
<td></td>
<td>Cluster 1B</td>
</tr>
<tr class="odd">
<td></td>
<td>Cluster 1C</td>
</tr>
<tr class="even">
<td>Cluster 2</td>
<td>Cluster 2A</td>
</tr>
<tr class="odd">
<td></td>
<td>Cluster 2B</td>
</tr>
<tr class="even">
<td>Cluster 3</td>
<td>Cluster 3</td>
</tr>
</tbody>
</table>
<p><img src="figure/ifn_03_footprint.Rmd/footprint_methods.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code>clusters &lt;- c(&quot;cluster1&quot;, &quot;cluster2&quot;, &quot;cluster3&quot;, &quot;cluster1A&quot;, &quot;cluster1B&quot;, &quot;cluster1C&quot;, &quot;cluster2B&quot;)
tf &lt;- read.csv(&quot;../data/IFNa/clusters_RNA/tf_clusters_ifna.csv&quot;,
               stringsAsFactors=FALSE)
colnames(tf) &lt;- clusters

genes &lt;- read.csv(&quot;../data/IFNa/clusters_RNA/gene_clusters_ifna.csv&quot;,
                  stringsAsFactors=FALSE)
colnames(genes) &lt;- clusters

## Create df for TF
cluster &lt;- c()
for (i in clusters) cluster &lt;- c(cluster, rep(i, nrow(tf[tf[,i]!=&quot;&quot;,])))
tfs &lt;- c()
for (i in 1:length(clusters)) tfs &lt;- c(tfs, tf[tf[,i]!=&quot;&quot;,i])

clust.tf &lt;- data.frame(&quot;cluster&quot;=cluster,
                       &quot;TFs&quot;=tfs)
save(clust.tf, file=&quot;../data/IFNa/clusters_RNA/tf_clusters_ifna.rda&quot;)

## Create df for genes
cluster &lt;- c()
for (i in clusters) cluster &lt;- c(cluster, rep(i, nrow(genes[genes[,i]!=&quot;&quot;,])))
tfs &lt;- c()
for (i in 1:length(clusters)) tfs &lt;- c(tfs, genes[genes[,i]!=&quot;&quot;,i])

clust.genes &lt;- data.frame(&quot;cluster&quot;=cluster,
                       &quot;external_gene_name&quot;=tfs)
save(clust.genes, file=&quot;../data/IFNa/clusters_RNA/gene_clusters_ifna.rda&quot;)


## Obtain coordinates for genes
load(&quot;../data/IFNa/RNA/diffAnalysis/res_2h.rda&quot;)
coord &lt;- res.df_2h[,c(21:24)]
coord$external_gene_name &lt;- toupper(coord$external_gene_name)


table(clust.genes$external_gene_name %in% coord$external_gene_name)

na &lt;- clust.genes[!(clust.genes$external_gene_name %in% unique(coord$external_gene_name)),]
alt &lt;- getNameFromAlias(na$external_gene_name)

## Replace for symbols
clust.genes$external_gene_name &lt;- as.character(clust.genes$external_gene_name)
clust.genes &lt;- clust.genes[order(clust.genes$external_gene_name),]
alt &lt;- alt[order(alt$alias_symbol),]

for (i in 1:nrow(alt)) {
  clust.genes$external_gene_name[clust.genes$external_gene_name==alt$alias_symbol[i]] &lt;- alt$symbol[i]
}

table(clust.genes$external_gene_name %in% coord$external_gene_name)

clust.genes.coord &lt;- dplyr::inner_join(clust.genes, coord)
clust.genes.coord$cluster &lt;- as.character(clust.genes.coord$cluster)
clust.genes.gr &lt;- regioneR::toGRanges(clust.genes.coord[,c(3:5, 1:2)])
save(clust.genes.gr, file=&quot;../data/IFNa/clusters_RNA/gene_clusters_ifna_granges.rda&quot;)
table(clust.genes.gr$cluster)

## Annotate open chromatin regions to closest gene in corresponding cluster
rm(list=ls())
load(&quot;../data/IFNa/clusters_RNA/gene_clusters_ifna_granges.rda&quot;)

load(&quot;../data/IFNa/ATAC/diffAnalysis/res.2h.rda&quot;)
res.2h &lt;- regioneR::toGRanges(res.2h.df[,c(17:19, 1:16)])
load(&quot;../data/IFNa/ATAC/diffAnalysis/res.24h.rda&quot;)
res.24h &lt;- regioneR::toGRanges(res.24h.df[,c(17:19, 1:16)])
rm(res.24h.df, res.2h.df)

c2 &lt;- c(&quot;cluster1&quot;, &quot;cluster2&quot;, &quot;cluster3&quot;) ## clusters at 2h
c24 &lt;- c(&quot;cluster3&quot;, &quot;cluster1A&quot;, &quot;cluster1B&quot;, &quot;cluster1C&quot;, &quot;cluster2B&quot;) ## clusters at 24h
cl.atac.2h &lt;- ChIPseeker::annotatePeak(res.2h, TxDb=clust.genes.gr[clust.genes.gr$cluster %in% c2],
                                       verbose=FALSE)
cl.atac.24h &lt;- ChIPseeker::annotatePeak(res.24h, TxDb=clust.genes.gr[clust.genes.gr$cluster %in% c24],
                                        verbose=FALSE)

## Filter out regions if distanceToTSS &gt; 1Mb
dist &lt;- 1000000
cl.atac.2h &lt;- data.frame(cl.atac.2h)
cl.atac.24h &lt;- data.frame(cl.atac.24h)

cl.atac.2h &lt;- unique(cl.atac.2h[abs(cl.atac.2h$distanceToTSS) &lt;= dist,])
cl.atac.24h &lt;- unique(cl.atac.24h[abs(cl.atac.24h$distanceToTSS) &lt;= dist,])

save(cl.atac.2h, file=file.path(out_dir, &quot;atac_clusters_ifna_2h.rda&quot;))
save(cl.atac.24h, file=file.path(out_dir, &quot;atac_clusters_ifna_24h.rda&quot;))</code></pre>
<p>The number of regions annotated to each RNA cluster is:</p>
<pre class="r"><code>load(file.path(out_dir, &quot;atac_clusters_ifna_2h.rda&quot;))
load(file.path(out_dir, &quot;atac_clusters_ifna_24h.rda&quot;))

cl.atac.2h.gr &lt;- regioneR::toGRanges(cl.atac.2h[,-c(4:5)])
cl.atac.24h.gr &lt;- regioneR::toGRanges(cl.atac.24h[,-c(4:5)])

knitr::kable(table(cl.atac.2h$cluster, cl.atac.2h$type),
             caption=&quot;OCRs at 2h annotated to each RNA-seq cluster.&quot;)</code></pre>
<table>
<caption>
<span id="tab:unnamed-chunk-2">Table 1: </span>OCRs at 2h annotated to each RNA-seq cluster.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
gained
</th>
<th style="text-align:right;">
stable
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
cluster1
</td>
<td style="text-align:right;">
419
</td>
<td style="text-align:right;">
24609
</td>
</tr>
<tr>
<td style="text-align:left;">
cluster2
</td>
<td style="text-align:right;">
947
</td>
<td style="text-align:right;">
55003
</td>
</tr>
<tr>
<td style="text-align:left;">
cluster3
</td>
<td style="text-align:right;">
146
</td>
<td style="text-align:right;">
14001
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>knitr::kable(table(cl.atac.24h$cluster, cl.atac.24h$type),
             caption=&quot;OCRs at 24h annotated to each RNA-seq cluster.&quot;)</code></pre>
<table>
<caption>
<span id="tab:unnamed-chunk-2">Table 1: </span>OCRs at 24h annotated to each RNA-seq cluster.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
gained
</th>
<th style="text-align:right;">
lost
</th>
<th style="text-align:right;">
stable
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
cluster1A
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
4991
</td>
</tr>
<tr>
<td style="text-align:left;">
cluster1B
</td>
<td style="text-align:right;">
65
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
12765
</td>
</tr>
<tr>
<td style="text-align:left;">
cluster1C
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
11738
</td>
</tr>
<tr>
<td style="text-align:left;">
cluster2B
</td>
<td style="text-align:right;">
97
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
28204
</td>
</tr>
<tr>
<td style="text-align:left;">
cluster3
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
15537
</td>
</tr>
</tbody>
</table>
</div>
<div id="footprint-analysis" class="section level2">
<h2>Footprint analysis</h2>
<p>The code used for calculating the footprints of all TFs outputed by the DREM model was ran using the following Rscript:</p>
<pre><code>Rscript code/IFNa_TF_footprint.R</code></pre>
<p><strong>Warning</strong>: Currently, <code>TFBSTools</code> is not working as expected…</p>
<pre class="r"><code>library(TFBSTools)
library(ggseqlogo)

## Load data ---------------------------------
load(&quot;../data/IFNa/clusters_RNA/tf_clusters_ifna.rda&quot;)
clust.tf$cluster &lt;- as.character(clust.tf$cluster)
clust.tf$TFs &lt;- as.character(clust.tf$TFs)

opts &lt;- list()
opts[[&quot;species&quot;]] &lt;- 9606
opts[[&quot;matrixtype&quot;]] &lt;- &quot;PWM&quot;
opts[[&quot;all_versions&quot;]] &lt;- TRUE
matJaspar &lt;- getMatrixSet(JASPAR2016::JASPAR2016, opts)

## Load matrix FOXA2
mot.f &lt;- read.delim(&quot;../data/IFNa/TF_motifs/fox2.motif&quot;, header=FALSE)
mot.f &lt;- t(mot.f)
rownames(mot.f) &lt;- c(&quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;)
mot &lt;- mot.f*1000

pfm.foxa2 &lt;- PFMatrix(ID=&quot;Human-Islets-FOXA2&quot;,
                name=&quot;FOXA2&quot;,
                profileMatrix=mot)

## Obtain list of TF and TF IDs
tf.id &lt;- data.frame(&quot;tf&quot;=unique(clust.tf$TFs),
                    &quot;id&quot;=NA,
                    stringsAsFactors=FALSE)
tf.id &lt;- rbind(tf.id, c(&quot;FOXA2&quot;, &quot;HumanIslet-FOXA2&quot;))

sel &lt;- c(&quot;STAT1&quot;, &quot;STAT2&quot;, &quot;IRF1&quot;)
tf.id &lt;- tf.id[tf.id$tf %in% sel,]

for (i in unique(clust.tf$TFs)) {
  mot &lt;- matJaspar[grep(i, name(matJaspar))]
    
    if (length(mot) &gt; 0) {
      name &lt;- ID(mot)[[1]]
      tf.id$id[tf.id$tf==i] &lt;- name
    } else {
      next
    }
}

tf.id &lt;- tf.id[!is.na(tf.id$id),]

##------------------------------------------------
## 2 hours
##------------------------------------------------
## Plot footprints (manually)
# dir.create(&quot;Figs_2h/&quot;, F)

## Set labels
upstream=100
downstream=100

for (i in c2) {
  for (j in unique(clust.tf$TFs[clust.tf$cluster==i])) {
    if (!(j %in% tf.id$tf)) next
    else {
      v &lt;- matJaspar[grep(tf.id$id[tf.id$tf==j], ID(matJaspar))]
      
      if (length(v)==0) {
        mot &lt;- pfm.foxa2
      } else {
        mot &lt;- Matrix(matJaspar[grep(tf.id$id[tf.id$tf==j], ID(matJaspar))][[1]])
      }
      
      load(file.path(out_dir, paste0(&quot;footprint_&quot;, i, &quot;_&quot;, j, &quot;_2h.rda&quot;)))
      ctrl$Treatment &lt;- &quot;CTRL&quot;
      ifn$Treatment &lt;- &quot;IFNa&quot;
      df &lt;- rbind(ctrl, ifn)
      
      ## Set breaks and labels
      nmotif = ncol(mot)
      breaks &lt;- c(min(ctrl$Position), (upstream+1), (upstream+nmotif), max(ctrl$Position))
      breaks &lt;- c(breaks, round(breaks[2]/2, 0), round((breaks[4]-breaks[3])/2 + breaks[3], 0))
      breaks &lt;- sort(breaks)
      labels &lt;- c(paste0(&quot;-&quot;, upstream), 
                  paste0(&quot;-&quot;, upstream/2), 
                  &quot;0&quot;, &quot;0&quot;, 
                  paste(downstream/2), 
                  paste(downstream))
      
      xlab = &quot;Dist. to motif (bp)&quot;
      ylab = &quot;Cut-site probability&quot;
      
      profile &lt;- ggplot(df, aes(Position, Probability)) +
                  geom_line(aes(color=Treatment, lty=Strand), lwd=1, alpha=0.8) +
                  scale_linetype_manual(values=c(&quot;Forward&quot;=1, &quot;Reverse&quot;=6)) +
                  scale_color_manual(values=c(&quot;CTRL&quot;=&quot;steelblue3&quot;, &quot;IFNa&quot;=&quot;tomato2&quot;)) +
                  geom_vline(xintercept=breaks[3], lty=2, color=&quot;dark grey&quot;) +
                  geom_vline(xintercept=breaks[4], lty=2, color=&quot;dark grey&quot;) +
                  theme(legend.position=&quot;bottom&quot;) +
                  scale_x_continuous(breaks=breaks,
                             labels=labels) +
                  xlab(xlab) + ylab(ylab)
      
        logo &lt;- ggseqlogo(mot, seq_type=&quot;dna&quot;) +
        ggtitle(paste0(j, &quot; (&quot;, ID(matJaspar[grep(tf.id$id[tf.id$tf==j], ID(matJaspar))][[1]]), &quot;)&quot;)) +
        theme(plot.title = element_text(size=35))
      
      final &lt;- plot_grid(logo, profile, ncol=1, rel_heights=c(1,3))
    }
  }
}

final

##-----------------------------------------------------
## 24 hours
##-----------------------------------------------------
# dir.create(&quot;Figs_24h/&quot;, F)

for (i in c24) {
  for (j in unique(clust.tf$TFs[clust.tf$cluster==i])) {
    if (!(j %in% tf.id$tf)) next
    else {
      v &lt;- matJaspar[grep(tf.id$id[tf.id$tf==j], ID(matJaspar))]
      
      if (length(v)==0) {
        mot &lt;- pfm.foxa2
      } else {
        mot &lt;- Matrix(matJaspar[grep(tf.id$id[tf.id$tf==j], ID(matJaspar))][[1]])
      }
      
      load(file.path(out_dir, paste0(&quot;footprint_&quot;, i, &quot;_&quot;, j, &quot;_24h.rda&quot;)))
      ctrl$Treatment &lt;- &quot;CTRL&quot;
      ifn$Treatment &lt;- &quot;IFNa&quot;
      df &lt;- rbind(ctrl, ifn)
      
      ## Set breaks and labels
      nmotif = ncol(mot)
      breaks &lt;- c(min(ctrl$Position), (upstream+1), (upstream+nmotif), max(ctrl$Position))
      breaks &lt;- c(breaks, round(breaks[2]/2, 0), round((breaks[4]-breaks[3])/2 + breaks[3], 0))
      breaks &lt;- sort(breaks)
      labels &lt;- c(paste0(&quot;-&quot;, upstream), 
                  paste0(&quot;-&quot;, upstream/2), 
                  &quot;0&quot;, &quot;0&quot;, 
                  paste(downstream/2), 
                  paste(downstream))
      
      xlab = &quot;Dist. to motif (bp)&quot;
      ylab = &quot;Cut-site probability&quot;
      
      profile &lt;- ggplot(df, aes(Position, Probability)) +
                  geom_line(aes(color=Treatment, lty=Strand), lwd=1, alpha=0.8) +
                  scale_linetype_manual(values=c(&quot;Forward&quot;=1, &quot;Reverse&quot;=6)) +
                  scale_color_manual(values=c(&quot;CTRL&quot;=&quot;steelblue3&quot;, &quot;IFNa&quot;=&quot;tomato2&quot;)) +
                  geom_vline(xintercept=breaks[3], lty=2, color=&quot;dark grey&quot;) +
                  geom_vline(xintercept=breaks[4], lty=2, color=&quot;dark grey&quot;) +
                  theme(legend.position=&quot;bottom&quot;) +
                  scale_x_continuous(breaks=breaks,
                             labels=labels) +
                  xlab(xlab) + ylab(ylab)
      
        logo &lt;- ggseqlogo(mot, seq_type=&quot;dna&quot;) +
        ggtitle(paste0(j, &quot; (&quot;, ID(matJaspar[grep(tf.id$id[tf.id$tf==j], ID(matJaspar))][[1]]), &quot;)&quot;)) +
        theme(plot.title = element_text(size=35))
      
      final &lt;- plot_grid(logo, profile, ncol=1, rel_heights=c(1,3))
    }
  }
}

final</code></pre>
<p><img src="figure/ifn_03_footprint.Rmd/plot-results-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/ifn_03_footprint.Rmd/plot-results-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] DESeq2_1.29.7               SummarizedExperiment_1.19.6
 [3] DelayedArray_0.15.7         matrixStats_0.56.0         
 [5] Matrix_1.2-18               Biobase_2.49.0             
 [7] ComplexHeatmap_2.5.3        dplyr_1.0.0                
 [9] kableExtra_1.1.0            cowplot_1.0.0              
[11] ggplot2_3.3.2               GenomicRanges_1.41.5       
[13] GenomeInfoDb_1.25.8         IRanges_2.23.10            
[15] S4Vectors_0.27.12           BiocGenerics_0.35.4        
[17] workflowr_1.6.2            

loaded via a namespace (and not attached):
 [1] bitops_1.0-6             fs_1.4.2                 bit64_0.9-7.1           
 [4] webshot_0.5.2            RColorBrewer_1.1-2       httr_1.4.2              
 [7] rprojroot_1.3-2          tools_4.0.0              backports_1.1.8         
[10] R6_2.4.1                 DBI_1.1.0                colorspace_1.4-1        
[13] GetoptLong_1.0.2         withr_2.2.0              tidyselect_1.1.0        
[16] bit_1.1-15.2             compiler_4.0.0           git2r_0.27.1            
[19] rvest_0.3.6              xml2_1.3.2               labeling_0.3            
[22] rtracklayer_1.49.4       bookdown_0.20            scales_1.1.1            
[25] readr_1.3.1              genefilter_1.71.0        Rsamtools_2.5.3         
[28] stringr_1.4.0            digest_0.6.25            rmarkdown_2.3           
[31] XVector_0.29.3           pkgconfig_2.0.3          htmltools_0.5.0         
[34] highr_0.8                BSgenome_1.57.5          regioneR_1.21.1         
[37] rlang_0.4.7              GlobalOptions_0.1.2      rstudioapi_0.11         
[40] RSQLite_2.2.0            farver_2.0.3             shape_1.4.4             
[43] generics_0.0.2           BiocParallel_1.23.2      RCurl_1.98-1.2          
[46] magrittr_1.5             GenomeInfoDbData_1.2.3   Rcpp_1.0.5              
[49] munsell_0.5.0            lifecycle_0.2.0          stringi_1.4.6           
[52] yaml_2.2.1               zlibbioc_1.35.0          blob_1.2.1              
[55] promises_1.1.1           crayon_1.3.4             lattice_0.20-41         
[58] Biostrings_2.57.2        splines_4.0.0            annotate_1.67.0         
[61] circlize_0.4.10          hms_0.5.3                magick_2.4.0            
[64] locfit_1.5-9.4           knitr_1.29               pillar_1.4.6            
[67] rjson_0.2.20             geneplotter_1.67.0       XML_3.99-0.5            
[70] glue_1.4.1               evaluate_0.14            png_0.1-7               
[73] vctrs_0.3.2              httpuv_1.5.4             gtable_0.3.0            
[76] purrr_0.3.4              clue_0.3-57              xfun_0.16               
[79] xtable_1.8-4             later_1.1.0.1            survival_3.2-3          
[82] viridisLite_0.3.0        tibble_3.0.3             GenomicAlignments_1.25.3
[85] AnnotationDbi_1.51.3     memoise_1.1.0            cluster_2.1.0           
[88] ellipsis_0.3.1          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
