---
title: '#3 Subclassification of IREs'
subtitle: "Primed and neo induced regulatory elements mediate cytokine response"
author: "Mireia Ramos-Rodríguez"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(warning = FALSE,
                      message = FALSE,
                      eval = TRUE,
                      fig.height = 4)

library(GenomicRanges)
library(ggplot2)
library(cowplot)
library(kableExtra)
library(dplyr)
library(igraph)
library(ggraph)

theme_set(cowplot::theme_cowplot())
load("../code/CYT_color_palettes.rda")

## Create directory to save results
out_dir <- "../output/cyt_03_subclassification/"
dir.create(out_dir, F)
```

***

# Details

- **Original publication**:

> **Ramos-Rodríguez, M.**, Raurell-Vila, H., Colli, M.L. et al. The impact of proinflammatory cytokines on the β-cell regulatory landscape provides insights into the genetics of type 1 diabetes. *Nat Genet.* **51**, 1588–1595 (2019) https://doi.org/10.1038/s41588-019-0524-6

- **Contents**: Analyses and figures contained in this document correspond to the following figures/sections of the original publication:

  + **Results**: "Primed and neo-regulatory elements mediate cytokine response".
  + **Figure 2**: "The β-cell response to proinflammatory cytokines unveils neo and primed IREs". Panels **c** to **f**.
  + **Extended Data Figure 3**: "Characterization of $\beta$-cell IREs". Panels **a** to **k**.

***

# Subclassification of IREs

To determine which IREs were not accessible in control conditions, we call peaks from merged control BAM file with a relaxed threshold: *P*<0.05.

```
macs2 callpeak -f BAM -t data/CYT/ATAC/BAM/merged/endoc_ctrl.offset.bam -g hs --outdir data/CYT/ATAC/Peaks/relaxed/ -n endoc_ctrl_relaxed --tempdir data/CYT/ATAC/Peaks/tmp/ -p 0.05 --nomodel --shift -100 --extsize 200
```

```{r generate-subgroups, eval=FALSE}
load("../data/CYT/REs/REs_endoc_fc1_padj0.05_granges.rda")
re <- re[!is.na(re$type),]

## Save first subgroup
re$subgroup1 <- NA
re$subgroup1[grep("SRE", re$type)] <- "SRE"
re$subgroup1[re$atac.type=="gained" & re$h3k27ac.type=="gained"] <- "Opening IRE"
re$subgroup1[re$atac.type=="stable" & re$h3k27ac.type=="gained"] <- "Primed IRE"

## Save 2nd subgroup
re$subgroup2 <- re$subgroup1

relaxed <- regioneR::toGRanges(paste0("../../data/ATAC/Peaks/relaxed/endoc_ctrl_relaxed_peaks.narrowPeak"))
ols <- subsetByOverlaps(re[re$subgroup1=="Opening IRE",],
                        relaxed)$atac.GeneID

re$subgroup2[re$atac.GeneID %in% ols & 
               re$subgroup1=="Opening IRE"] <- "Other IRE"
re$subgroup2[!(re$atac.GeneID %in% ols) &
               re$subgroup1=="Opening IRE"] <- "Neo IRE"

save(re, file="../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda")

## With less stringent threshold --- 
load("../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_k27.8.rda")
re <- re[!is.na(re$type),]

## Save first subgroup
re$subgroup1 <- NA
re$subgroup1[grep("SRE", re$type)] <- "SRE"
re$subgroup1[re$atac.type=="gained" & re$h3k27ac.type=="gained"] <- "Opening IRE"
re$subgroup1[re$atac.type=="stable" & re$h3k27ac.type=="gained"] <- "Primed IRE"

## Save 2nd subgroup
re$subgroup2 <- re$subgroup1

relaxed <- regioneR::toGRanges(paste0("../data/CYT/ATAC/Peaks/relaxed/endoc_ctrl_relaxed_peaks.narrowPeak"))
ols <- subsetByOverlaps(re[re$subgroup1=="Opening IRE",],
                        relaxed)$atac.GeneID

re$subgroup2[re$atac.GeneID %in% ols & 
               re$subgroup1=="Opening IRE"] <- "Other IRE"
re$subgroup2[!(re$atac.GeneID %in% ols) &
               re$subgroup1=="Opening IRE"] <- "Neo IRE"

save(re, file="../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup_k27.8.rda")
```

```{r distr-reads-sub-ires, fig.height=3, fig.width=5}
load("../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda")
re <- data.frame(mcols(re)[,c(1,13,17,18)])

load("../data/CYT/ATAC/diffAnalysis/ATAC_endoc_normCountsBatch.rda")
counts <- data.frame(counts)
counts$atac.GeneID <- rownames(counts)
counts <- reshape2::melt(counts,
                         id.var=11,
                         value.var=1:10,
                         value.name="counts",
                         variable.name="sample")
counts$treatment <- unlist(lapply(strsplit(as.character(counts$sample), "_"),
                                  function(x) x[2]))

counts <- counts %>%
  group_by(atac.GeneID, treatment) %>%
  summarise(counts=mean(counts))

colnames(counts)[3] <- "ATAC-seq"

re.counts <- dplyr::left_join(re, counts)

load("../data/CYT/H3K27ac/diffAnalysis/H3K27ac_endoc_normCountsBatch.rda")
counts <- data.frame(counts)
counts$h3k27ac.GeneID <- rownames(counts)
counts <- reshape2::melt(counts,
                         id.var=9,
                         value.var=1:8,
                         value.name="counts",
                         variable.name="sample")
counts$treatment <- unlist(lapply(strsplit(as.character(counts$sample), "_"),
                                  function(x) x[2]))

counts <- counts %>%
  group_by(h3k27ac.GeneID, treatment) %>%
  summarise(counts=mean(counts))

colnames(counts)[3] <- "H3K27ac"

re.counts <- dplyr::left_join(re.counts, counts)

re.counts <- reshape2::melt(re.counts,
                            id.vars=1:5,
                            value.vars=6:7,
                            value.name="counts",
                            variable.name="experiment")

distr.reads <- 
  ggplot(re.counts[!grepl("Other", re.counts$subgroup2),],
       aes(subgroup2, log2(counts+1))) +
  geom_boxplot(aes(color=treatment), notch=TRUE, lwd=0.7) +
  scale_color_manual(values=pals$treatment,
                     labels=function(x) toupper(x),
                     name="Treatment") +
  facet_wrap(~experiment) +
  ylab(expression(Log[2]*" normalized counts")) +
  scale_x_discrete(labels=function(x) paste0(x, "s")) +
  theme(axis.text.x=element_text(angle=30, hjust=1),
        axis.title.x=element_blank(),
        strip.background = element_rect(fill="white", color="black",
                                        size=0.5, linetype=1))

distr.reads
```

## Characterization

```{r assoc-sub-rna, fig.width=4, fig.height=4}
win.width=30e3

load("../data/CYT/RNA/diffAnalysis/RNA_endoc_GRangesBatch.rda")

rna <- res.gr[res.gr$gene_biotype=="protein_coding",]
win <- promoters(rna)
win <- resize(rna, width=win.width, fix="center")

load("../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda")
ols <- findOverlaps(re, win)

anno <- cbind(mcols(re)[queryHits(ols), c(1,17,18)],
              mcols(win)[subjectHits(ols), c(1,2,5)])

load("../data/CYT/RNA/diffAnalysis/RNA_endoc_normCountsBatch.rda")
counts <- data.frame(counts)
counts$GeneID <- rownames(counts)
counts$GeneID <- gsub("\\.[[:digit:]]*", "", counts$GeneID)

counts <- reshape2::melt(counts,
                         id.vars=11,
                         value.vars=1:10,
                         value.name="counts",
                         variable.name="sample")
counts$treatment <- unlist(lapply(strsplit(as.character(counts$sample), "_"),
                                  function(x) x[2]))

counts.mean <- counts %>%
  group_by(GeneID, treatment) %>%
  summarise(counts=mean(counts))

anno.counts <- left_join(data.frame(anno), counts.mean)
anno.counts <- unique(anno.counts[,-1])


test <- anno.counts[anno.counts$subgroup2!="SRE",] %>%
  group_by(subgroup1) %>%
  summarise(pval=wilcox.test(counts[treatment=="ctrl"],
                             counts[treatment=="cyt"],
                             paired=F)$p.value)

test$lab <- ""
test$lab[test$pval<0.05] <- "*"
test$lab[test$pval<0.01] <- "**"
test$lab[test$pval<0.001] <- "***"

ggplot(anno.counts[anno.counts$subgroup2!="SRE",],
       aes(subgroup1, log2(counts + 1))) +
  geom_boxplot(aes(color=treatment), notch=TRUE, 
               outlier.shape=NA, lwd=1) +
  geom_text(data=test,
            aes(x=subgroup1, y=20, label=lab),
            size=5) +
  scale_color_manual(values=pals$treatment,
                     name="Treatment",
                     labels=function(x) toupper(x)) +
  ylab(expression("RNA-seq "*log[2]*" counts")) +
  scale_x_discrete(labels=function(x) paste0(x, "s")) +
  theme(legend.position="top",
        axis.title.x=element_blank())
```


```{r sub-cons, eval=FALSE}
load(paste0("../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda"))
scope=1e3
bin=50

## Opening IREs
op <- re[grep("Opening", re$subgroup1),]
rnd <- regioneR::randomizeRegions(op)

op.cons <- pipelineNGS::calculateMeanCons(op,
                                           scope=scope,
                                           bin=bin)
op.cons$re_type <- "Opening IRE"
rnd.cons <- pipelineNGS::calculateMeanCons(rnd,
                                           scope=scope,
                                           bin=bin)
rnd.cons$re_type <- "Random Opening IRE"

ire.cons <- rbind(op.cons, rnd.cons)

## Primed IREs
pr <- re[grep("Primed", re$subgroup1),]
rnd <- regioneR::randomizeRegions(pr)

pr.cons <- pipelineNGS::calculateMeanCons(pr,
                                           scope=scope,
                                           bin=bin)
pr.cons$re_type <- "Primed IRE"
rnd.cons <- pipelineNGS::calculateMeanCons(rnd,
                                           scope=scope,
                                           bin=bin)
rnd.cons$re_type <- "Random Primed IRE"

## Final dataset
ire.cons <- rbind(ire.cons, pr.cons, rnd.cons)
save(ire.cons, file=file.path(out_dir, "CONS_IREs_sub.rda"))
```

```{r plot-cons-sub, fig.height=4, fig.width=6}
load(file.path(out_dir, "CONS_IREs_sub.rda"))

ire.cons$color <- gsub("Random ", "", ire.cons$re_type)
ire.cons$lty <- gsub("Opening ", "", gsub("Primed ", "", ire.cons$re_type))

cons.plot <- 
  ggplot(ire.cons,
       aes(position, meanCons)) +
  geom_line(aes(lty=lty, color=color), 
            lwd=1) +
  scale_linetype_discrete(name="Region type") +
  scale_color_manual(values=pals$re,
                     name="Region type",
                     labels=function(x) paste0(x, "s")) +
  geom_vline(xintercept=0, lty=2, color="grey") +
  ylab("Mean PhastCons46way Score") +
  xlab("Position from peak center (bp)") +
  theme(legend.position = "top") +
  guides(linetype=guide_legend(nrow=2),
         color=guide_legend(nrow=2))

cons.plot
```


 
```{r}
load("../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda")
re.df <- data.frame(re)[,-c(1:5)]

## Make groups
re.tss <- unique(re.df[!is.na(re.df$type),])

re.tss$anno.group <- NA
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)>200e3] <- ">200kb"
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)<=200e3 &
                    abs(re.tss$atac.distanceToTSS)>20e3] <- "20-200kb"
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)<=20e3 &
                    abs(re.tss$atac.distanceToTSS)>2e3] <- "2-20kb"
re.tss$anno.group[abs(re.tss$atac.distanceToTSS)<=2e3] <- "0-2kb"

len.neo <- sum(grepl("Neo", re.tss$subgroup2))
len.primed <- sum(grepl("Primed", re.tss$subgroup2))
len.sre <- sum(grepl("SRE", re.tss$subgroup2))

sum.tss <- re.tss %>%
  group_by(subgroup2, anno.group) %>%
  summarise(num=length(unique(atac.GeneID)))

sum.tss$perc <- NA
sum.tss$perc[grep("Neo", sum.tss$subgroup2)] <- sum.tss$num[grep("Neo", sum.tss$subgroup2)]/len.neo*100
sum.tss$perc[grep("Primed", sum.tss$subgroup2)] <- sum.tss$num[grep("Primed", sum.tss$subgroup2)]/len.primed*100
sum.tss$perc[grep("SRE", sum.tss$subgroup2)] <- sum.tss$num[grep("SRE", sum.tss$subgroup2)]/len.sre*100

sum.tss <- sum.tss[!is.na(sum.tss$perc),]
sum.tss$anno.group <- factor(sum.tss$anno.group,
                             levels=c("0-2kb", "2-20kb", "20-200kb", ">200kb"))

tss.plot <- 
  ggplot(sum.tss,
       aes(anno.group, perc)) +
  geom_bar(aes(fill=subgroup2), color="black", lwd=0.7, stat="identity", position="dodge") +
  geom_vline(xintercept=1.5, lty=2, color="dark red") +
  scale_fill_manual(values=pals$re,
                    name="RE type", labels=function(x) paste0(x, "s")) +
  annotate("text", x=c(1,3), y=c(60, 60), label=c("Promoters", "Enhancers")) +
  theme(legend.position="top") +
  xlab("Distance to TSS") + 
  scale_y_continuous(name="Percentage of RE",
                     labels=function(x) paste0(x, "%"),
                     breaks=scales::pretty_breaks()) +
  guides(fill=guide_legend(ncol=1))

tss.plot
```

```{r, eval=FALSE}
load("../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda")

## Do it on enhancers IREs
sel <- re[grep("IRE", re$type),]


files <- list.files("~/data/ChromHMM_ENCODE-Broad/",
                    pattern="*.bed",
                    full.names=TRUE)
names <- pipelineNGS::getNameFromPath(files,
                                      prefix="wgEncodeBroadHmm", 
                                      suffix="HMM.bed")

chrom <- lapply(files, regioneR::toGRanges)
names(chrom) <- names

overlaps <- data.frame()
for (i in 1:length(chrom)) {
  hmm <- chrom[[i]]
  ols <- findOverlaps(sel,
                      hmm)
  df <- cbind(sel$type[queryHits(ols)],
              sel$atac.GeneID[queryHits(ols)],
              hmm$name[subjectHits(ols)])
  df <- as.data.frame(df)
  colnames(df) <- c("type", "peakID", "chromHMM")
  df$ENCODE <- names(chrom)[i]  
  overlaps <- rbind(overlaps, df)
}

ids.enh <- unique(overlaps$peakID[grep("Enhancer", overlaps$chromHMM)])
length(ids.enh)
ids.prom <- unique(overlaps$peakID[grep("Promoter", overlaps$chromHMM)])
length(ids.prom)
ids.het <- unique(overlaps$peakID[grep("Hetero", overlaps$chromHMM)])
length(ids.het)

ids.het.enh <- ids.het[ids.het %in% ids.enh]
length(ids.het.enh)

## Test with a randomized set
rndm <- regioneR::randomizeRegions(sel)
rndm$geneID <- paste0("random_", 1:length(rndm))

overlaps.rndm <- data.frame()
for (i in 1:length(chrom)) {
  hmm <- chrom[[i]]
  ols <- findOverlaps(rndm,
                      hmm)
  df <- cbind(rndm$geneID[queryHits(ols)],
              hmm$name[subjectHits(ols)])
  df <- as.data.frame(df)
  colnames(df) <- c("peakID", "chromHMM")
  df$ENCODE <- names(chrom)[i]  
  overlaps.rndm <- rbind(overlaps.rndm, df)
}

ids.enh.rndm <- unique(overlaps.rndm$peakID[grep("Enhancer", overlaps.rndm$chromHMM)])
length(ids.enh.rndm)
ids.prom.rndm <- unique(overlaps.rndm$peakID[grep("Promoter", overlaps.rndm$chromHMM)])
length(ids.prom.rndm)
ids.het.rndm <- unique(overlaps.rndm$peakID[grep("Hetero", overlaps.rndm$chromHMM)])
length(ids.het.rndm)

ids.het.enh.rndm <- ids.het.rndm[ids.het.rndm %in% ids.enh.rndm]
length(ids.het.enh.rndm)

df <- data.frame("class"=rep(c("Enhancer", "Promoter"), 2),
                 "regions"=c(rep("IREs", 2), rep("Random", 2)),
                 "overlaps"=c(length(ids.enh), length(ids.prom), 
                              length(ids.enh.rndm), length(ids.prom.rndm)))

mat <- matrix(df$overlaps, ncol=2)

save(df, file=file.path(out_dir, "ENH_chromHMM_IREs.rda"))
```

```{r, fig.width=3.5, fig.height=4}
load(file.path(out_dir, "ENH_chromHMM_IREs.rda"))

bar.chrom <- 
  ggplot(df[df$class!="Promoter",],
       aes(class, overlaps)) +
  geom_bar(aes(fill=regions), stat="identity",
           position="dodge", color="black",
           width=0.7) +
  xlab("ChromHMM class") +
  scale_y_continuous(labels=scales::comma,
                     breaks=scales::pretty_breaks(),
                     name="# of overlapping IREs") +
  scale_fill_manual(values=c("#e37418", "#0c3a5c"),
                    name="Source") +
  annotate("text", 1, 2800, label="*", size=8) +
  theme(legend.position=c(0.55, 0.85))

bar.chrom
```

# Transcription factor analysis

## *de novo* motif analysis

```{r run-denovo, eval=FALSE}
library(maRge)

deNovoMotifHOMER(bed="../data/CYT/bedfiles/IREs_endoc_fc1_padj0.05_opening_distal.bed",
                 path_output=file.path(out_dir, "HOMER_IREs_endoc_opening_distal_mask"),
                 other_param="-mask",
                 path_homer="~/tools/homer/")

deNovoMotifHOMER(bed="../data/CYT/bedfiles/IREs_endoc_fc1_padj0.05_primed_distal.bed",
                 path_output=file.path(out_dir, "HOMER_IREs_endoc_primed_distal_mask"),
                 other_param="-mask",
                 path_homer="~/tools/homer/")

deNovoMotifHOMER(bed="../data/CYT/bedfiles/IREs_endoc_fc1_padj0.05_neo_distal.bed",
                 path_output=file.path(out_dir, "HOMER_IREs_endoc_neo_distal_mask"),
                 other_param="-mask",
                 path_homer="~/tools/homer/")
```


```{r res-denovo}
htmltools::includeHTML(file.path(out_dir, "HOMER_IREs_endoc_opening_distal_mask/homerResults.html"))
htmltools::includeHTML(file.path(out_dir, "HOMER_IREs_endoc_primed_distal_mask/homerResults.html"))
```

## Footprint of ISRE motif

```
Rscript code/CYT_runFootprintAnalysis.R
```

```{r, fig.height=3, fig.width=4.5}
load(file.path(out_dir, "FOOT_ISRE_neo_distal_cov.rda"))
cov$condition <- gsub("endoc_", "", gsub(".5Tag", "", cov$condition))

df <- XML::readHTMLTable(file.path(out_dir, "HOMER_IREs_endoc_neo_distal_mask/homerResults.html"))[[1]]
df <- df[1,]

stats <- cov %>%
  group_by(condition) %>%
  summarise(median=median(tags5),
            max=max(tags5),
            cent=max/2)

stats$width <- nchar(as.character(df$Motif))/4

num <- 1179

foot.neo <- 
  ggplot(cov,
       aes(Distance, tags5))+
  geom_tile(data=stats,
            aes(x=0, y=cent,
                width=width, height=max),
            color=NA, fill="grey") +
  geom_hline(data=stats,
             aes(yintercept=median),
             lty=2, color="dark grey") +
  geom_line(aes(color=condition), lwd=0.7) +
  scale_color_manual(values=pals$treatment,
                     name="Treatment", labels=function(x) toupper(x)) +
  annotate("text", 100, 0, label=num) +
  ylab("Normalized cleavage\nevents per bp") +
  ggtitle("Neo enhancers")

foot.neo
```


```{r, fig.height=3, fig.width=4.5}
load(file.path(out_dir, "FOOT_ISRE_primed_distal_cov.rda"))
cov$condition <- gsub("endoc_", "", gsub(".5Tag", "", cov$condition))

df <- XML::readHTMLTable(file.path(out_dir, "HOMER_IREs_endoc_primed_distal_mask/homerResults.html"))[[1]]
df <- df[1,]

stats <- cov %>%
  group_by(condition) %>%
  summarise(median=median(tags5),
            max=max(tags5),
            cent=max/2)

stats$width <- nchar(as.character(df$Motif))/4

num <- 403

foot.prim <- 
  ggplot(cov,
       aes(Distance, tags5))+
  geom_tile(data=stats,
            aes(x=0, y=cent,
                width=width, height=max),
            color=NA, fill="grey") +
  geom_hline(data=stats,
             aes(yintercept=median),
             lty=2, color="dark grey") +
  geom_line(aes(color=condition), lwd=0.7) +
  scale_color_manual(values=pals$treatment,
                     name="Treatment", labels=function(x) toupper(x)) +
  annotate("text", 100, 0, label=num) +
  ylab("Normalized cleavage\nevents per bp") +
  ggtitle("Primed enhancers")

foot.prim
```

## Overlap Primed IREs with islet regulome ChIP-seq TF data

- Overlap with peaks

```{r, eval=FALSE}
load("~/data/isletRegulome_TF.rda")

load("../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda")
re <- re[re$atac.annotation=="Distal",]
type <- c("Neo IRE", "Primed IRE", "SRE")
re <- re[re$subgroup2 %in% type,]

## Find overlaps
ols <- findOverlaps(re, tfs.islet)

tfs <- cbind(data.frame(mcols(re))[queryHits(ols),c(1,18)],
             data.frame(mcols(tfs.islet))[subjectHits(ols),])
tfs <- tfs[,-3]

no <- data.frame(mcols(re))[!(re$atac.GeneID %in% tfs$atac.GeneID),c(1,18)]
no$TF <- "None"

tfs <- rbind(tfs, no)

tfs$TF <- factor(tfs$TF,
                 levels=rev(unique(tfs$TF)[c(3,2,1,4,5,6)]))

save(tfs,
     file=file.path(out_dir, "IREs_fc1_padj0.05_isletSpecific_TFs_peaks.rda"))
```

```{r, fig.width=4, figh.height=4}
load(file.path(out_dir, "IREs_fc1_padj0.05_isletSpecific_TFs_peaks.rda"))

bar.tfs <- 
  ggplot(tfs,
       aes(subgroup2, ..count..)) +
  geom_bar(aes(fill=TF), position="fill",
           color="black") +
  scale_fill_manual(values = c("FOXA2"="plum3", "MAFB"="steelblue3", "NKX2.2"="olivedrab3",
                               "NKX6.1"="sienna3", "PDX1"="khaki3", 
                               "None" = "grey")) +
  coord_flip() +
  scale_y_continuous(labels=function(x) paste0(x*100, "%"),
                     name="Percentage of regions") +
  scale_x_discrete(labels=function(x) paste0(x, "s")) +
  theme(legend.position="top",
        axis.title.y=element_blank()) +
  guides(fill=guide_legend(nrow=2, reverse=TRUE))

bar.tfs
```

- Density of reads

```{r, eval=FALSE}
## Create SAF
load("../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda")
re <- re[re$atac.annotation=="Distal",]
type <- c("Neo IRE", "Primed IRE", "SRE")
re <- re[re$subgroup2 %in% type,]

re <- resize(re, width=2e3, fix="center")
re.bin <- unlist(tile(re, n=201))
pos <- seq(-1000, 1000, by=10)
re.bin$position <- rep(pos, length(re))
re.bin$atac.GeneID <- rep(re$atac.GeneID, each=length(pos))
re.bin$GeneID <- paste0(re.bin$atac.GeneID, "_", re.bin$position)

saf <- data.frame(re.bin)[,c(1:3,6:8)]
colnames(saf)[1:3] <- c("Chr", "Start", "End")
saf$Strand <- "+"

files <- list.files("~/data/TFsIslet_hg19/BAM",
                    pattern="*.bam$",
                    full.names=TRUE)
files <- files[!grepl("CTCF", files)]

counts <- Rsubread::featureCounts(files,
                                  annot.ext=saf,
                                  allowMultiOverlap = TRUE,
                                  nthreads=6)

names <- pipelineNGS::getNameFromPath(files, suffix=".bam")
colnames(counts$counts) <- names

mat <- preprocessCore::normalize.quantiles(counts$counts)
colnames(mat) <- names
rownames(mat) <- rownames(counts$counts)

##
df <- data.frame(mat)
df$GeneID <- rownames(mat)

df <- dplyr::left_join(df,
                       saf)
df <- dplyr::left_join(df,
                       data.frame(mcols(re))[,c(1,18)])

df <- reshape2::melt(df,
                     id.vars=6:13,
                     value.vars=1:5,
                     variable.name="TF",
                     value.name="counts")

df <- df[,c(5,8:10)]

tfs <- df %>%
  group_by(position, TF, subgroup2) %>%
  summarise(mean=mean(counts, na.rm=T))

save(tfs, file=file.path(out_dir, "IREs_fc1_padj0.05_isletSpecific_TFs_bam.rda"))
```


```{r, fig.width=4, fig.height=4}
load(file.path(out_dir, "IREs_fc1_padj0.05_isletSpecific_TFs_bam.rda"))

tfs.dens <- 
  ggplot(tfs,
       aes(position, mean)) +
  geom_smooth(aes(group=subgroup2, color=subgroup2)) +
  scale_color_manual(values=c("olivedrab3", "#80E8C0", "dark grey"),
                     name="") +
  scale_y_continuous(breaks=scales::pretty_breaks(),
                     name="Islet TFs read counts") +
  coord_cartesian(xlim=c(-500, 500)) + 
  xlab("Position relative to peak center (bp)") +
  ggtitle("Islet-specific TFs occupancy") +
  theme(legend.position=c(0.7, 0.9))

tfs.dens
```


## Co-localization of inflammatory with islet-specific TFs at Primed IREs

```
Rscript code/CYT_colocalizationMotifs.R
```

```{r, fig.width=6, fig.height=5}
load(file.path(out_dir, "MOT-COLOC_df.rda"))
mot.coloc <- mot.coloc[grep("primed", mot.coloc$type),]

infl <- c("ISRE", "STAT")
isl <- c("HNF", "PDX", "NEURO", "MAF", "NKX")

vert <- data.frame(Var1=unique(mot.coloc$Var1),
                   TF_type="Other",
                   stringsAsFactors = FALSE)
vert$TF_type[sapply(infl, grep, toupper(vert$Var1))] <- "Inflammatory"
vert$TF_type[sapply(isl, grep, toupper(vert$Var1))] <- "Islet"
vert <- dplyr::left_join(vert, unique(mot.coloc[mot.coloc$Var1==mot.coloc$Var2,c(1,4)]))

mot.coloc <- mot.coloc[order(mot.coloc$percentage),]

igraph <- graph_from_data_frame(mot.coloc,
                                vertices=vert)
cols <- c("hotpink3", "goldenrod3", "grey")

graphPrimed <-
  ggraph(igraph, layout = 'linear', circular = TRUE) + 
    geom_edge_arc(aes(edge_width=percentage, color=percentage), alpha=0.8) +
    scale_edge_width_continuous(guide=FALSE, limits=c(0,30)) +
    scale_edge_colour_gradient(low="white", high="grey15",
                               name="% Overlap", limits=c(0,30)) +
    geom_node_label(aes(label = name, fill=TF_type, size=overlaps)) +
    scale_fill_manual(values=cols, name="Motif Type") +
    scale_size_continuous(range=c(3,6), guide=FALSE) + 
    theme_void() +
    xlim(-1.3, 1.3) + ylim(-1.15, 1.15) +
    theme(legend.position="bottom") +
  ggtitle("Primed enhancers")

graphPrimed
```


```{r, fig.width=6, fig.height=5}
load(file.path(out_dir, "MOT-COLOC_df.rda"))
mot.coloc <- mot.coloc[grep("SRE", mot.coloc$type),]

infl <- c("ISRE", "STAT")
isl <- c("HNF", "PDX", "NEURO", "MAF", "NKX")

vert <- data.frame(Var1=unique(mot.coloc$Var1),
                   TF_type="Other",
                   stringsAsFactors = FALSE)
vert$TF_type[sapply(infl, grep, toupper(vert$Var1))] <- "Inflammatory"
vert$TF_type[sapply(isl, grep, toupper(vert$Var1))] <- "Islet"
vert <- dplyr::left_join(vert, unique(mot.coloc[mot.coloc$Var1==mot.coloc$Var2,c(1,4)]))

mot.coloc <- mot.coloc[order(mot.coloc$percentage),]

igraph <- graph_from_data_frame(mot.coloc,
                                vertices=vert)

cols <- c("hotpink3", "goldenrod3", "grey")

graphSRE <-
  ggraph(igraph, layout = 'linear', circular = TRUE) + 
    geom_edge_arc(aes(edge_width=percentage, color=percentage), alpha=0.8) +
    scale_edge_width_continuous(guide=FALSE, limits=c(0,30)) +
    scale_edge_colour_gradient(low="white", high="grey15",
                               name="% Overlap", limits=c(0,30)) +
    geom_node_label(aes(label = name, fill=TF_type, size=overlaps)) +
    scale_fill_manual(values=cols, name="Motif Type") +
    scale_size_continuous(range=c(3,6), guide=FALSE) + 
    theme_void() +
    xlim(-1.3, 1.3) + ylim(-1.15, 1.15) +
    theme(legend.position="bottom") +
  ggtitle("SREs enhancers")

graphSRE
```


- Fisher test

```{r, fig.width=4.5, fig.height=4.5}
load(file.path(out_dir, "MOT-COLOC_df_rmRedundant.rda"))

comb <- unique(mot.coloc[,c(1:2)])

## fisher test
comb$estimate <- NA
comb$p.value <- NA
comb$confIntMin <- NA
comb$confIntMax <- NA

for (i in 1:nrow(comb)) {
  sel <- mot.coloc[mot.coloc$Var1==comb$Var1[i] &
                     mot.coloc$Var2==comb$Var2[i],]
  
  overlapping.primed <- sel$overlaps[grepl("primed", sel$type)]
  overlapping.stable <- sel$overlaps[grepl("SREs", sel$type)]
  nonOverlapping.primed <- sel$total[grepl("primed", sel$type)] - overlapping.primed
  nonOverlapping.stable <- sel$total[grepl("SREs", sel$type)] - overlapping.stable
    
  mat.test <- matrix(c(overlapping.primed, overlapping.stable,
                       nonOverlapping.primed, nonOverlapping.stable),
                     byrow=T, ncol=2)
    
  test <- fisher.test(mat.test)
  
  comb$estimate[i] <- test$estimate
  comb$p.value[i] <- test$p.value
  comb$confIntMin[i] <- test$conf.int[1]
  comb$confIntMax[i] <- test$conf.int[2]
}

comb$padj <-  p.adjust(comb$p.value, method="fdr")

comb$name <- paste(comb$Var1, "+", comb$Var2)

comb.sel <- comb[comb$padj<=0.001,]
comb.sel <- comb.sel[order(comb.sel$estimate),]

comb.sel$name <- factor(comb.sel$name,
                        levels=unique(comb.sel$name))

comb.sel$TFtype1 <- "Other"
comb.sel$TFtype1[unlist(sapply(infl, grep, comb.sel$Var1))] <- "Inflammatory"
comb.sel$TFtype1[unlist(sapply(isl, grep, comb.sel$Var1))] <- "Islet"

comb.sel$TFtype2 <- "Other"
comb.sel$TFtype2[unlist(sapply(infl, grep, comb.sel$Var2))] <- "Inflammatory"
comb.sel$TFtype2[unlist(sapply(isl, grep, comb.sel$Var2))] <- "Islet"

comb.sel$combined <- FALSE
comb.sel$combined[comb.sel$TFtype1=="Inflammatory" & comb.sel$TFtype2=="Islet"] <- TRUE
comb.sel$combined[comb.sel$TFtype2=="Inflammatory" & comb.sel$TFtype1=="Islet"] <- TRUE

comb.sel$color <- "black"
comb.sel$color[comb.sel$combined] <- "violetred3"

fisher.coloc <-
  ggplot(comb.sel, aes(estimate, name)) + 
  geom_point(size=3, color=comb.sel$color) +
  geom_segment(aes(x=confIntMin, xend=confIntMax,
                   y=name, yend=name), color=comb.sel$color) +
  geom_vline(xintercept=1, color="dark red") +
  # scale_color_manual(values=pals$tfs,
  #                    name="Motif Pair Type",
  #                    labels=function(x) Hmisc::capitalize(x)) +
  # scale_x_log10() +
  xlab("Odds Ratio") +
  ylab("Motif Pairs") +
  theme(panel.grid.major.y=element_line(color="grey", linetype=2)) +
  guides(color=guide_legend(nrow=3))

fisher.coloc
```

# DNA methylation

```
Rscript code/CYT_runMethylationAnalysis.R
```

```{r process-methylation, eval=FALSE}
betas <- read.csv("../data/CYT/methylationEPIC/reports/differential_methylation_data/diffMethTable_site_cmp2.csv",
                  stringsAsFactors = F)

betas <- regioneR::toGRanges(betas[,c(3,4,4,2,6:18)])

## Overlap with IREs
load("../data/CYT/REs/REs_endoc_fc1_padj0.05_granges_subgroup.rda")
re <- re[grep("Distal", re$atac.annotation),]

ols <- findOverlaps(re,
                    betas)

betas.re <- cbind(data.frame(re)[queryHits(ols),c(6,21:23)],
                  data.frame(betas)[subjectHits(ols),c(6:19)])

betas.re$diffmeth.p.adj.fdr.2 <- p.adjust(betas.re$diffmeth.p.val, method="fdr")

length(unique(betas.re$cgid[betas.re$mean.diff>0.2 & betas.re$diffmeth.p.adj.fdr<=0.05]))
length(unique(betas.re$cgid[betas.re$mean.diff>0.2 & betas.re$diffmeth.p.adj.fdr.2<=0.05]))

betas.re$methType <- "stable"
betas.re$methType[betas.re$mean.diff>0.2 & betas.re$diffmeth.p.adj.fdr<=0.05] <- "lost"

save(betas.re,
     file=file.path(out_dir, "METH_betas_REs_endoc.rda"))

betas <- read.csv("../data//CYT/methylationEPIC/reports/differential_methylation_data/diffMethTable_site_cmp2.csv",
                  stringsAsFactors = F)

betas <- betas[,c(3,4,4,2,6:18)]

length(unique(betas$cgid[betas$mean.diff>0.2 & betas$diffmeth.p.adj.fdr<=0.05]))

betas$methType <- "stable"
betas$methType[betas$mean.diff>0.2 & betas$diffmeth.p.adj.fdr<=0.05] <- "lost"

save(betas,
     file=file.path(out_dir, "METH_betas_all_endoc.rda"))
```


```{r violin-plots, fig.width=4, fig.height=4}
load(file.path(out_dir, "METH_betas_REs_endoc.rda"))

betas.distr <- unique(betas.re[!grepl("Other", betas.re$subgroup2),c(1,4,5:7)])
betas.distr <- betas.distr[!grepl("SRE", betas.distr$subgroup2),]
betas.distr <- betas.distr[!is.na(betas.distr$subgroup2),]

betas.distr <- reshape2::melt(betas.distr,
                              id.vars=1:3,
                              value.vars=4:5,
                              value.name="beta",
                              variable.name="treatment")
betas.distr$treatment <- gsub("mean.", "", tolower(as.character(betas.distr$treatment)))

sum <- betas.distr %>%
  group_by(subgroup2, treatment) %>%
  summarise(mean=mean(beta),
            median=median(beta))

test <- betas.distr %>%
  group_by(subgroup2) %>%
  summarise(pval=wilcox.test(beta[grep("ctrl", treatment)],
                             beta[grep("cyt", treatment)],
                             paired=F)$p.value)
test$sym <- ""
test$sym[test$pval<0.05] <- "*"
test$sym[test$pval<0.01] <- "**"
test$sym[test$pval<0.001] <- "***"

violin <- 
  ggplot(betas.distr,
       aes(treatment, beta)) +
  geom_violin(aes(fill=treatment)) +
  geom_point(data=sum, 
             aes(treatment, median), size=3) +
  scale_fill_manual(values=pals$treatment,
                    name="Treatment", labels=function(x) toupper(x)) +
  geom_text(data=test,
            aes(x=1.5,
                y=0.9, label=sym), size=8) +
  facet_wrap(~gsub("IRE", "enhancers", subgroup2)) +
  scale_y_continuous(name=expression("DNA methylation "*beta*"-value"),
                     limits=c(0,1), expand=c(0,0)) +
  theme(legend.position="bottom",
        axis.text.x=element_blank(),
        axis.title.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.line.x=element_blank(),
        strip.background=element_rect(fill="white", color="black", linetype=1,
                                      size=0.5))

violin
```



```{r, fig.width=4, fig.height=4}
load(file.path(out_dir, "METH_betas_all_endoc.rda"))

volc <- ggplot(betas, 
       aes(-mean.diff, -log10(diffmeth.p.adj.fdr))) +
  geom_point(aes(color=methType)) +
  scale_color_manual(values=c(stable="grey", lost="tomato3")) +
  geom_vline(xintercept=c(-0.2, 0.2), lty=2, color="dark grey") +
  geom_hline(yintercept=-log10(0.05), lty=2, color="dark grey") +
  xlab(expression(beta*cyt-beta*ctrl)) +
  ylab(expression("DNA methlyation -log"[10]*" FDR")) +
  theme(legend.position="none")

volc
```


```{r, fig.height=4, fig.width=4}
load(file.path(out_dir, "METH_betas_REs_endoc.rda"))

bar <- 
  ggplot(betas.re,#[!grepl("Other", betas.re$subgroup2),],
       aes(methType, ..count..)) +
  geom_bar(aes(fill=subgroup2), position="fill", color="black") +
  scale_fill_manual(values=pals$re,
                    name="RE type",
                    labels=function(x) paste(gsub(" IRE", "", x), "enh")) +
  scale_y_continuous(name="Percentage of CpGs",
                     labels=function(x) paste0(x*100, "%")) +
  scale_x_discrete(name="DNA methylation change",
                   labels=c("Demethylated", "Stable")) +
  theme(axis.text.x=element_text(angle=20, hjust=1))

bar
```


